{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Similarity comparison "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,sys, tempfile, pandas as pd\n",
    "sys.path.append(\"src\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_dir =  \"tempv2\"\n",
    "os.makedirs(working_dir, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def scores_to_df(scores, details, data_set, data_type, task_type):\n",
    "    df = pd.DataFrame()\n",
    "    for k , v in scores.items():\n",
    "        df[k] = v\n",
    "        \n",
    "    for k , v in details.items():\n",
    "        df[k + \"_detail\"] = v\n",
    "    \n",
    "    df[\"data_set\"] = data_set + \" \" + data_type\n",
    "    df[\"task_type\"] = task_type\n",
    "         \n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "base_data_dir=\"tmp\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. BC2GM\n",
    "\n",
    "[Biocreative II gene mention](https://biocreative.bioinformatics.udel.edu/tasks/biocreative-ii/) overlap. Please download the test and train files for this task from the BioCreative Website.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "bc2gm_train_file = os.path.join(base_data_dir, \"train.in\")\n",
    "bc2gm_test_file = os.path.join(base_data_dir, \"test.in\")\n",
    "\n",
    "bc2gm_train_eval_file = os.path.join(base_data_dir, \"trainGENE.eval\")\n",
    "bc2gm_test_eval_file = os.path.join(base_data_dir, \"testGENE.eval\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exact matches Unigram, 39 / 5000\n",
      "Exact matches Bigram, 26 / 5000\n",
      "Exact matches Trigram, 26 / 5000\n",
      "Exact matches Unigram, 1996 / 6331\n",
      "Exact matches Bigram, 541 / 6331\n",
      "Exact matches Trigram, 134 / 6331\n"
     ]
    }
   ],
   "source": [
    "from bc2_gene_mention import BC2GeneMentionText \n",
    "\n",
    "\n",
    "bc2gmrun = BC2GeneMentionText()\n",
    "result_score, result_detail = bc2gmrun.run_similarity_comparer(\"text\", bc2gm_train_file,  bc2gm_test_file)\n",
    "df_bc2_gm_text = scores_to_df(result_score,result_detail, \"BC2GM\", \"text\", \"NER\")\n",
    "\n",
    "result_score, result_detail = bc2gmrun.run_similarity_comparer(\"eval\", bc2gm_train_eval_file,  bc2gm_test_eval_file)\n",
    "df_bc2_gm_eval = scores_to_df(result_score,result_detail, \"BC2GM\", \"anno\", \"NER\")\n",
    "\n",
    "df_bc2_gm = pd.concat([df_bc2_gm_text, df_bc2_gm_eval])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "      <th>Bigram_detail</th>\n",
       "      <th>Trigram_detail</th>\n",
       "      <th>data_set</th>\n",
       "      <th>task_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4229</th>\n",
       "      <td>44.721360</td>\n",
       "      <td>31.524416</td>\n",
       "      <td>18.257419</td>\n",
       "      <td>(This polypeptide includes the first three zin...</td>\n",
       "      <td>(This polypeptide includes the first three zin...</td>\n",
       "      <td>(This polypeptide includes the first three zin...</td>\n",
       "      <td>BC2GM text</td>\n",
       "      <td>NER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3052</th>\n",
       "      <td>21.821789</td>\n",
       "      <td>8.333333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>(Serum amyloid A (SAA) is a plasma protein whi...</td>\n",
       "      <td>(Serum amyloid A (SAA) is a plasma protein whi...</td>\n",
       "      <td>(Serum amyloid A (SAA) is a plasma protein whi...</td>\n",
       "      <td>BC2GM text</td>\n",
       "      <td>NER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unigram     Bigram    Trigram  \\\n",
       "4229  44.721360  31.524416  18.257419   \n",
       "3052  21.821789   8.333333   0.000000   \n",
       "\n",
       "                                         Unigram_detail  \\\n",
       "4229  (This polypeptide includes the first three zin...   \n",
       "3052  (Serum amyloid A (SAA) is a plasma protein whi...   \n",
       "\n",
       "                                          Bigram_detail  \\\n",
       "4229  (This polypeptide includes the first three zin...   \n",
       "3052  (Serum amyloid A (SAA) is a plasma protein whi...   \n",
       "\n",
       "                                         Trigram_detail    data_set task_type  \n",
       "4229  (This polypeptide includes the first three zin...  BC2GM text       NER  \n",
       "3052  (Serum amyloid A (SAA) is a plasma protein whi...  BC2GM text       NER  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc2_gm.sample(n=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>data_set</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">BC2GM anno</th>\n",
       "      <th>count</th>\n",
       "      <td>6331.000000</td>\n",
       "      <td>6331.000000</td>\n",
       "      <td>6331.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>70.766285</td>\n",
       "      <td>19.552406</td>\n",
       "      <td>5.411720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>35.975868</td>\n",
       "      <td>35.854864</td>\n",
       "      <td>20.895407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>57.735027</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>81.649658</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>25.819889</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">BC2GM text</th>\n",
       "      <th>count</th>\n",
       "      <td>5000.000000</td>\n",
       "      <td>5000.000000</td>\n",
       "      <td>5000.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>33.191640</td>\n",
       "      <td>13.122176</td>\n",
       "      <td>4.200074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>11.568668</td>\n",
       "      <td>10.636721</td>\n",
       "      <td>9.330504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>26.311741</td>\n",
       "      <td>8.058230</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>31.622777</td>\n",
       "      <td>12.171612</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>38.336508</td>\n",
       "      <td>17.320508</td>\n",
       "      <td>7.161149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Unigram       Bigram      Trigram\n",
       "data_set                                               \n",
       "BC2GM anno count  6331.000000  6331.000000  6331.000000\n",
       "           mean     70.766285    19.552406     5.411720\n",
       "           std      35.975868    35.854864    20.895407\n",
       "           min       0.000000     0.000000     0.000000\n",
       "           25%      57.735027     0.000000     0.000000\n",
       "           50%      81.649658     0.000000     0.000000\n",
       "           75%     100.000000    25.819889     0.000000\n",
       "           max     100.000000   100.000000   100.000000\n",
       "BC2GM text count  5000.000000  5000.000000  5000.000000\n",
       "           mean     33.191640    13.122176     4.200074\n",
       "           std      11.568668    10.636721     9.330504\n",
       "           min       0.000000     0.000000     0.000000\n",
       "           25%      26.311741     8.058230     0.000000\n",
       "           50%      31.622777    12.171612     0.000000\n",
       "           75%      38.336508    17.320508     7.161149\n",
       "           max     100.000000   100.000000   100.000000"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc2_gm.groupby([\"data_set\"]).describe().stack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2708</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(RNA polymerase II, RNA polymerase II)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4326</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(protein tyrosine kinase, protein tyrosine kin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6070</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(Src tyrosine kinases, src tyrosine kinases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6069</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(cyclin-dependent kinase 2, Cyclin-dependent k...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4648</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(signal recognition particle, signal recogniti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1204</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(type III collagen, type III collagen)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(hepatitis B surface antigen, hepatitis B surf...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2891</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(prostate-specific antigen, Prostate specific ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5712</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(nitric oxide synthase, nitric oxide synthase)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1987</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(bacterial chloramphenicol acetyltransferase, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2875</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(actin binding domain, actin-binding domain)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>551</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(immediate-early genes, immediate-early genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3444</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(protein tyrosine kinase, protein tyrosine kin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3804</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(Jun N-terminal kinase, Jun N-terminal kinase)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3457</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(Recombinant Human Erythropoietin, Recombinant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6093</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(Sp1-binding site, Sp1-binding site)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3458</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(Tumor necrosis factor, Tumor necrosis factor)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(protein-tyrosine kinase, protein tyrosine kin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(nuclear factor kappaB, nuclear factor-kappaB)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2850</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(E2F family members, E2F family members (1-5))</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unigram                                     Unigram_detail\n",
       "2708    100.0             (RNA polymerase II, RNA polymerase II)\n",
       "4326    100.0  (protein tyrosine kinase, protein tyrosine kin...\n",
       "6070    100.0       (Src tyrosine kinases, src tyrosine kinases)\n",
       "6069    100.0  (cyclin-dependent kinase 2, Cyclin-dependent k...\n",
       "4648    100.0  (signal recognition particle, signal recogniti...\n",
       "1204    100.0             (type III collagen, type III collagen)\n",
       "565     100.0  (hepatitis B surface antigen, hepatitis B surf...\n",
       "2891    100.0  (prostate-specific antigen, Prostate specific ...\n",
       "5712    100.0     (nitric oxide synthase, nitric oxide synthase)\n",
       "1987    100.0  (bacterial chloramphenicol acetyltransferase, ...\n",
       "2875    100.0       (actin binding domain, actin-binding domain)\n",
       "551     100.0     (immediate-early genes, immediate-early genes)\n",
       "3444    100.0  (protein tyrosine kinase, protein tyrosine kin...\n",
       "3804    100.0     (Jun N-terminal kinase, Jun N-terminal kinase)\n",
       "3457    100.0  (Recombinant Human Erythropoietin, Recombinant...\n",
       "6093    100.0               (Sp1-binding site, Sp1-binding site)\n",
       "3458    100.0     (Tumor necrosis factor, Tumor necrosis factor)\n",
       "543     100.0  (protein-tyrosine kinase, protein tyrosine kin...\n",
       "3462    100.0     (nuclear factor kappaB, nuclear factor-kappaB)\n",
       "2850    100.0     (E2F family members, E2F family members (1-5))"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc2_gm_eval.sort_values(by=[\"Unigram\"], ascending=False)[[\"Unigram\", \"Unigram_detail\"]].head(n=20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(bradykinin, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4743</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(UD3, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4747</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(thiolase, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4749</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(TBPH, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4751</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(TM4SF, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4753</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(maleless, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4755</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(CAM5.2, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4763</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(ICP22, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1572</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(FKB2, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1571</th>\n",
       "      <td>0.0</td>\n",
       "      <td>(GRP94, alkaline phosphatases)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unigram                       Unigram_detail\n",
       "0         0.0  (bradykinin, alkaline phosphatases)\n",
       "4743      0.0         (UD3, alkaline phosphatases)\n",
       "4747      0.0    (thiolase, alkaline phosphatases)\n",
       "4749      0.0        (TBPH, alkaline phosphatases)\n",
       "4751      0.0       (TM4SF, alkaline phosphatases)\n",
       "4753      0.0    (maleless, alkaline phosphatases)\n",
       "4755      0.0      (CAM5.2, alkaline phosphatases)\n",
       "4763      0.0       (ICP22, alkaline phosphatases)\n",
       "1572      0.0        (FKB2, alkaline phosphatases)\n",
       "1571      0.0       (GRP94, alkaline phosphatases)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc2_gm_eval.sort_values(by=[\"Unigram\"], ascending=True)[[\"Unigram\", \"Unigram_detail\"]].head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/aeg/envs/nlp-train-test-overlap-detector/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Train</th>\n",
       "      <th>Test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Biol.</td>\n",
       "      <td>Biol.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 2000 Academic Press.</td>\n",
       "      <td>Copyright 2000 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(1988) J.</td>\n",
       "      <td>1988).</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Coronary vasoconstriction caused by endothelin-1 is enhanced by ischemia-reperfusion and by norepinephrine present in concentrations typically observed after neonatal cardiopulmonary bypass.</td>\n",
       "      <td>Coronary vasoconstriction caused by endothelin-1 is enhanced by ischemia-reperfusion and by norepinephrine present in concentrations typically observed after neonatal cardiopulmonary bypass.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>We also show that in fusions with the DNA binding domain of GAL4, full activity requires the entire BHV-alpha TIF, although both amino and carboxyl termini display some activity on their own.</td>\n",
       "      <td>We also show that in fusions with the DNA binding domain of GAL4, full activity requires the entire BHV-alpha TIF, although both amino and carboxyl termini display some activity on their own.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Biol.</td>\n",
       "      <td>Biol.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Acad.</td>\n",
       "      <td>Acad.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>A heterologous promoter construct containing three repeats of a consensus Sp1 site, cloned upstream of a single copy of the ZII (CREB/ AP1) element from the BZLF1 promoter linked to the beta-globin TATA box, exhibited phorbol ester inducibility.</td>\n",
       "      <td>A heterologous promoter construct containing three repeats of a consensus Sp1 site, cloned upstream of a single copy of the ZII (CREB/ AP1) element from the BZLF1 promoter linked to the beta-globin TATA box, exhibited phorbol ester inducibility.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Virol.</td>\n",
       "      <td>Virol.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>748</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(1995) J.</td>\n",
       "      <td>1995.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>902</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1006</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Natl.</td>\n",
       "      <td>Natl.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1039</th>\n",
       "      <td>81.649658</td>\n",
       "      <td>Ischemic stroke due to protein C deficiency.</td>\n",
       "      <td>Free protein S deficiency in acute ischemic stroke.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1075</th>\n",
       "      <td>83.405766</td>\n",
       "      <td>3 cases</td>\n",
       "      <td>In stage I, histochemistry for copper was positive in 11 out of 21 cases: 6 cases were T+; 1 case R+ and 2 cases O+; 2 cases were T+, R+, O+.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1137</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(1992) J.</td>\n",
       "      <td>(1992) J.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1259</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>The activated glucocorticoid receptor forms a complex with Stat5 and enhances Stat5-mediated transcriptional induction.</td>\n",
       "      <td>The activated glucocorticoid receptor forms a complex with Stat5 and enhances Stat5-mediated transcriptional induction.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>The promoter for HMG-CoA synthase contains two binding sites for the sterol regulatory element-binding proteins (SREBPs).</td>\n",
       "      <td>The promoter for HMG-CoA synthase contains two binding sites for the sterol regulatory element-binding proteins (SREBPs).</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1652</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1709</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Chem.</td>\n",
       "      <td>Chem.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1724</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Appl.</td>\n",
       "      <td>Appl.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1898</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 1998 Academic Press.</td>\n",
       "      <td>Copyright 1998 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1929</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Slowly adapting type I mechanoreceptor discharge as a function of dynamic force versus dynamic displacement of glabrous skin of raccoon and squirrel monkey hand.</td>\n",
       "      <td>Slowly adapting type I mechanoreceptor discharge as a function of dynamic force versus dynamic displacement of glabrous skin of raccoon and squirrel monkey hand.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2094</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Sci.</td>\n",
       "      <td>Sci.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2098</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2182</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 1998 Academic Press.</td>\n",
       "      <td>Copyright 1998 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2270</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Mutational analysis of yeast CEG1 demonstrated that four of the five conserved motifs are essential for capping enzyme function in vivo.</td>\n",
       "      <td>Mutational analysis of yeast CEG1 demonstrated that four of the five conserved motifs are essential for capping enzyme function in vivo.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2298</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>The recruitment of constitutively phosphorylated p185(neu) and the activated mitogenic pathway proteins to this membrane-microfilament interaction site provides a physical model for integrating the assembly of the mitogenic pathway with the transmission of growth factor signal to the cytoskeleton.</td>\n",
       "      <td>The recruitment of constitutively phosphorylated p185(neu) and the activated mitogenic pathway proteins to this membrane-microfilament interaction site provides a physical model for integrating the assembly of the mitogenic pathway with the transmission of growth factor signal to the cytoskeleton.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2332</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Analysis of 1 Mb of published sequence from the region of conserved synteny on human chromosome 5q31-q33 identified 45 gene candidates, including 35 expressed genes in the human IL-4 cytokine gene cluster.</td>\n",
       "      <td>Analysis of 1 Mb of published sequence from the region of conserved synteny on human chromosome 5q31-q33 identified 45 gene candidates, including 35 expressed genes in the human IL-4 cytokine gene cluster.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2381</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>The human SHBG proximal promoter was analyzed by DNase I footprinting, and the functional significance of 6 footprinted regions (FP1-FP6) within the proximal promoter was studied in human HepG2 hepatoblastoma cells.</td>\n",
       "      <td>The human SHBG proximal promoter was analyzed by DNase I footprinting, and the functional significance of 6 footprinted regions (FP1-FP6) within the proximal promoter was studied in human HepG2 hepatoblastoma cells.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2486</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Recently we have performed a detailed analysis of specific neuronal populations affected by the mutation which shed new light on the role of Krox-20 in the segmentation and on the physiological consequences of its inactivation.</td>\n",
       "      <td>Recently we have performed a detailed analysis of specific neuronal populations affected by the mutation which shed new light on the role of Krox-20 in the segmentation and on the physiological consequences of its inactivation.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2866</th>\n",
       "      <td>86.602540</td>\n",
       "      <td>Dialyzable transfer factor.</td>\n",
       "      <td>Non-dialyzable transfer factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2933</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(LH P &lt; 0.05, LH/FSH P &lt; 0.01).</td>\n",
       "      <td>(LH P &lt; 0.05, LH/FSH P &lt; 0.01).</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2968</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>IV.</td>\n",
       "      <td>IV.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3144</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Urease activity, judged as the amount of ammonia production from urea, could be measured at 25 ng per tube (S/N = 1.5) with Jack bean meal urease.</td>\n",
       "      <td>Urease activity, judged as the amount of ammonia production from urea, could be measured at 25 ng per tube (S/N = 1.5) with Jack bean meal urease.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3342</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Although RAD17, RAD24 and MEC3 are not required for cell cycle arrest when S phase is inhibited by hydroxyurea (HU), they do contribute to the viability of yeast cells grown in the presence of HU, possibly because they are required for the repair of HU-induced DNA damage.</td>\n",
       "      <td>Although RAD17, RAD24 and MEC3 are not required for cell cycle arrest when S phase is inhibited by hydroxyurea (HU), they do contribute to the viability of yeast cells grown in the presence of HU, possibly because they are required for the repair of HU-induced DNA damage.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3365</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>IV.</td>\n",
       "      <td>IV.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3492</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>The reconstituted RNA polymerases containing the mutant alpha subunits were examined for their response to transcription activation by cAMP-CRP and the rrnBP1 UP element.</td>\n",
       "      <td>The reconstituted RNA polymerases containing the mutant alpha subunits were examined for their response to transcription activation by cAMP-CRP and the rrnBP1 UP element.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3716</th>\n",
       "      <td>77.459667</td>\n",
       "      <td>356, 93-98].</td>\n",
       "      <td>98, 93-98).</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3772</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Biol.</td>\n",
       "      <td>Biol.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3983</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4210</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 1998 Academic Press.</td>\n",
       "      <td>Copyright 1998 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4217</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Determinants of recurrent ischaemia and revascularisation procedures after thrombolysis with recombinant tissue plasminogen activator in primary coronary occlusion.</td>\n",
       "      <td>Determinants of recurrent ischaemia and revascularisation procedures after thrombolysis with recombinant tissue plasminogen activator in primary coronary occlusion.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4372</th>\n",
       "      <td>76.454959</td>\n",
       "      <td>The cases included 35 de novo diffuse aggressive lymphomas (DAL; 19 large-cell, 4 mixed-cell, and 12 large-cell immunoblastic), 52 transformed aggressive lymphomas derived from follicular lymphomas (TFL), 42 indolent follicular lymphomas (FL), 14 mantle cell lymphomas (MCL), and 27 small noncleaved cell lymphomas (SNCL).</td>\n",
       "      <td>Histological and immunophenotypic studies revealed 12 large cell lymphomas (11 B cell and one T cell), two small noncleaved cell lymphomas (B-cell phenotype), and five low grade B-cell lymphomas (two small lymphocytic and three follicular mixed lymphomas).</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4385</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>These results demonstrate a specific association of SIV and HIV-2 nef, but not HIV-1 nef, with TCRzeta.</td>\n",
       "      <td>These results demonstrate a specific association of SIV and HIV-2 nef, but not HIV-1 nef, with TCRzeta.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4594</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "      <td>Copyright 1999 Academic Press.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4764</th>\n",
       "      <td>86.602540</td>\n",
       "      <td>DESIGN: Retrospective study.</td>\n",
       "      <td>STUDY DESIGN: Retrospective review.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4819</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>Acad.</td>\n",
       "      <td>Acad.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Unigram  \\\n",
       "14    100.000000   \n",
       "15    100.000000   \n",
       "223   100.000000   \n",
       "255   100.000000   \n",
       "416   100.000000   \n",
       "429   100.000000   \n",
       "445   100.000000   \n",
       "568   100.000000   \n",
       "650   100.000000   \n",
       "748   100.000000   \n",
       "902   100.000000   \n",
       "1006  100.000000   \n",
       "1039  81.649658    \n",
       "1075  83.405766    \n",
       "1137  100.000000   \n",
       "1259  100.000000   \n",
       "1413  100.000000   \n",
       "1652  100.000000   \n",
       "1709  100.000000   \n",
       "1724  100.000000   \n",
       "1898  100.000000   \n",
       "1929  100.000000   \n",
       "2094  100.000000   \n",
       "2098  100.000000   \n",
       "2182  100.000000   \n",
       "2270  100.000000   \n",
       "2298  100.000000   \n",
       "2332  100.000000   \n",
       "2381  100.000000   \n",
       "2486  100.000000   \n",
       "2866  86.602540    \n",
       "2933  100.000000   \n",
       "2968  100.000000   \n",
       "3144  100.000000   \n",
       "3342  100.000000   \n",
       "3365  100.000000   \n",
       "3492  100.000000   \n",
       "3716  77.459667    \n",
       "3772  100.000000   \n",
       "3983  100.000000   \n",
       "4210  100.000000   \n",
       "4217  100.000000   \n",
       "4372  76.454959    \n",
       "4385  100.000000   \n",
       "4594  100.000000   \n",
       "4764  86.602540    \n",
       "4819  100.000000   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                   Train  \\\n",
       "14    Biol.                                                                                                                                                                                                                                                                                                                                \n",
       "15    Copyright 2000 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "223   (1988) J.                                                                                                                                                                                                                                                                                                                            \n",
       "255   Coronary vasoconstriction caused by endothelin-1 is enhanced by ischemia-reperfusion and by norepinephrine present in concentrations typically observed after neonatal cardiopulmonary bypass.                                                                                                                                       \n",
       "416   We also show that in fusions with the DNA binding domain of GAL4, full activity requires the entire BHV-alpha TIF, although both amino and carboxyl termini display some activity on their own.                                                                                                                                      \n",
       "429   Biol.                                                                                                                                                                                                                                                                                                                                \n",
       "445   Acad.                                                                                                                                                                                                                                                                                                                                \n",
       "568   A heterologous promoter construct containing three repeats of a consensus Sp1 site, cloned upstream of a single copy of the ZII (CREB/ AP1) element from the BZLF1 promoter linked to the beta-globin TATA box, exhibited phorbol ester inducibility.                                                                                \n",
       "650   Virol.                                                                                                                                                                                                                                                                                                                               \n",
       "748   (1995) J.                                                                                                                                                                                                                                                                                                                            \n",
       "902   Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "1006  Natl.                                                                                                                                                                                                                                                                                                                                \n",
       "1039  Ischemic stroke due to protein C deficiency.                                                                                                                                                                                                                                                                                         \n",
       "1075  3 cases                                                                                                                                                                                                                                                                                                                              \n",
       "1137  (1992) J.                                                                                                                                                                                                                                                                                                                            \n",
       "1259  The activated glucocorticoid receptor forms a complex with Stat5 and enhances Stat5-mediated transcriptional induction.                                                                                                                                                                                                              \n",
       "1413  The promoter for HMG-CoA synthase contains two binding sites for the sterol regulatory element-binding proteins (SREBPs).                                                                                                                                                                                                            \n",
       "1652  Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "1709  Chem.                                                                                                                                                                                                                                                                                                                                \n",
       "1724  Appl.                                                                                                                                                                                                                                                                                                                                \n",
       "1898  Copyright 1998 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "1929  Slowly adapting type I mechanoreceptor discharge as a function of dynamic force versus dynamic displacement of glabrous skin of raccoon and squirrel monkey hand.                                                                                                                                                                    \n",
       "2094  Sci.                                                                                                                                                                                                                                                                                                                                 \n",
       "2098  Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "2182  Copyright 1998 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "2270  Mutational analysis of yeast CEG1 demonstrated that four of the five conserved motifs are essential for capping enzyme function in vivo.                                                                                                                                                                                             \n",
       "2298  The recruitment of constitutively phosphorylated p185(neu) and the activated mitogenic pathway proteins to this membrane-microfilament interaction site provides a physical model for integrating the assembly of the mitogenic pathway with the transmission of growth factor signal to the cytoskeleton.                           \n",
       "2332  Analysis of 1 Mb of published sequence from the region of conserved synteny on human chromosome 5q31-q33 identified 45 gene candidates, including 35 expressed genes in the human IL-4 cytokine gene cluster.                                                                                                                        \n",
       "2381  The human SHBG proximal promoter was analyzed by DNase I footprinting, and the functional significance of 6 footprinted regions (FP1-FP6) within the proximal promoter was studied in human HepG2 hepatoblastoma cells.                                                                                                              \n",
       "2486  Recently we have performed a detailed analysis of specific neuronal populations affected by the mutation which shed new light on the role of Krox-20 in the segmentation and on the physiological consequences of its inactivation.                                                                                                  \n",
       "2866  Dialyzable transfer factor.                                                                                                                                                                                                                                                                                                          \n",
       "2933  (LH P < 0.05, LH/FSH P < 0.01).                                                                                                                                                                                                                                                                                                      \n",
       "2968  IV.                                                                                                                                                                                                                                                                                                                                  \n",
       "3144  Urease activity, judged as the amount of ammonia production from urea, could be measured at 25 ng per tube (S/N = 1.5) with Jack bean meal urease.                                                                                                                                                                                   \n",
       "3342  Although RAD17, RAD24 and MEC3 are not required for cell cycle arrest when S phase is inhibited by hydroxyurea (HU), they do contribute to the viability of yeast cells grown in the presence of HU, possibly because they are required for the repair of HU-induced DNA damage.                                                     \n",
       "3365  IV.                                                                                                                                                                                                                                                                                                                                  \n",
       "3492  The reconstituted RNA polymerases containing the mutant alpha subunits were examined for their response to transcription activation by cAMP-CRP and the rrnBP1 UP element.                                                                                                                                                           \n",
       "3716  356, 93-98].                                                                                                                                                                                                                                                                                                                         \n",
       "3772  Biol.                                                                                                                                                                                                                                                                                                                                \n",
       "3983  Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "4210  Copyright 1998 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "4217  Determinants of recurrent ischaemia and revascularisation procedures after thrombolysis with recombinant tissue plasminogen activator in primary coronary occlusion.                                                                                                                                                                 \n",
       "4372  The cases included 35 de novo diffuse aggressive lymphomas (DAL; 19 large-cell, 4 mixed-cell, and 12 large-cell immunoblastic), 52 transformed aggressive lymphomas derived from follicular lymphomas (TFL), 42 indolent follicular lymphomas (FL), 14 mantle cell lymphomas (MCL), and 27 small noncleaved cell lymphomas (SNCL).   \n",
       "4385  These results demonstrate a specific association of SIV and HIV-2 nef, but not HIV-1 nef, with TCRzeta.                                                                                                                                                                                                                              \n",
       "4594  Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                                                       \n",
       "4764  DESIGN: Retrospective study.                                                                                                                                                                                                                                                                                                         \n",
       "4819  Acad.                                                                                                                                                                                                                                                                                                                                \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                            Test  \n",
       "14    Biol.                                                                                                                                                                                                                                                                                                       \n",
       "15    Copyright 2000 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "223   1988).                                                                                                                                                                                                                                                                                                      \n",
       "255   Coronary vasoconstriction caused by endothelin-1 is enhanced by ischemia-reperfusion and by norepinephrine present in concentrations typically observed after neonatal cardiopulmonary bypass.                                                                                                              \n",
       "416   We also show that in fusions with the DNA binding domain of GAL4, full activity requires the entire BHV-alpha TIF, although both amino and carboxyl termini display some activity on their own.                                                                                                             \n",
       "429   Biol.                                                                                                                                                                                                                                                                                                       \n",
       "445   Acad.                                                                                                                                                                                                                                                                                                       \n",
       "568   A heterologous promoter construct containing three repeats of a consensus Sp1 site, cloned upstream of a single copy of the ZII (CREB/ AP1) element from the BZLF1 promoter linked to the beta-globin TATA box, exhibited phorbol ester inducibility.                                                       \n",
       "650   Virol.                                                                                                                                                                                                                                                                                                      \n",
       "748   1995.                                                                                                                                                                                                                                                                                                       \n",
       "902   Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "1006  Natl.                                                                                                                                                                                                                                                                                                       \n",
       "1039  Free protein S deficiency in acute ischemic stroke.                                                                                                                                                                                                                                                         \n",
       "1075  In stage I, histochemistry for copper was positive in 11 out of 21 cases: 6 cases were T+; 1 case R+ and 2 cases O+; 2 cases were T+, R+, O+.                                                                                                                                                               \n",
       "1137  (1992) J.                                                                                                                                                                                                                                                                                                   \n",
       "1259  The activated glucocorticoid receptor forms a complex with Stat5 and enhances Stat5-mediated transcriptional induction.                                                                                                                                                                                     \n",
       "1413  The promoter for HMG-CoA synthase contains two binding sites for the sterol regulatory element-binding proteins (SREBPs).                                                                                                                                                                                   \n",
       "1652  Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "1709  Chem.                                                                                                                                                                                                                                                                                                       \n",
       "1724  Appl.                                                                                                                                                                                                                                                                                                       \n",
       "1898  Copyright 1998 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "1929  Slowly adapting type I mechanoreceptor discharge as a function of dynamic force versus dynamic displacement of glabrous skin of raccoon and squirrel monkey hand.                                                                                                                                           \n",
       "2094  Sci.                                                                                                                                                                                                                                                                                                        \n",
       "2098  Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "2182  Copyright 1998 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "2270  Mutational analysis of yeast CEG1 demonstrated that four of the five conserved motifs are essential for capping enzyme function in vivo.                                                                                                                                                                    \n",
       "2298  The recruitment of constitutively phosphorylated p185(neu) and the activated mitogenic pathway proteins to this membrane-microfilament interaction site provides a physical model for integrating the assembly of the mitogenic pathway with the transmission of growth factor signal to the cytoskeleton.  \n",
       "2332  Analysis of 1 Mb of published sequence from the region of conserved synteny on human chromosome 5q31-q33 identified 45 gene candidates, including 35 expressed genes in the human IL-4 cytokine gene cluster.                                                                                               \n",
       "2381  The human SHBG proximal promoter was analyzed by DNase I footprinting, and the functional significance of 6 footprinted regions (FP1-FP6) within the proximal promoter was studied in human HepG2 hepatoblastoma cells.                                                                                     \n",
       "2486  Recently we have performed a detailed analysis of specific neuronal populations affected by the mutation which shed new light on the role of Krox-20 in the segmentation and on the physiological consequences of its inactivation.                                                                         \n",
       "2866  Non-dialyzable transfer factor                                                                                                                                                                                                                                                                              \n",
       "2933  (LH P < 0.05, LH/FSH P < 0.01).                                                                                                                                                                                                                                                                             \n",
       "2968  IV.                                                                                                                                                                                                                                                                                                         \n",
       "3144  Urease activity, judged as the amount of ammonia production from urea, could be measured at 25 ng per tube (S/N = 1.5) with Jack bean meal urease.                                                                                                                                                          \n",
       "3342  Although RAD17, RAD24 and MEC3 are not required for cell cycle arrest when S phase is inhibited by hydroxyurea (HU), they do contribute to the viability of yeast cells grown in the presence of HU, possibly because they are required for the repair of HU-induced DNA damage.                            \n",
       "3365  IV.                                                                                                                                                                                                                                                                                                         \n",
       "3492  The reconstituted RNA polymerases containing the mutant alpha subunits were examined for their response to transcription activation by cAMP-CRP and the rrnBP1 UP element.                                                                                                                                  \n",
       "3716  98, 93-98).                                                                                                                                                                                                                                                                                                 \n",
       "3772  Biol.                                                                                                                                                                                                                                                                                                       \n",
       "3983  Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "4210  Copyright 1998 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "4217  Determinants of recurrent ischaemia and revascularisation procedures after thrombolysis with recombinant tissue plasminogen activator in primary coronary occlusion.                                                                                                                                        \n",
       "4372  Histological and immunophenotypic studies revealed 12 large cell lymphomas (11 B cell and one T cell), two small noncleaved cell lymphomas (B-cell phenotype), and five low grade B-cell lymphomas (two small lymphocytic and three follicular mixed lymphomas).                                            \n",
       "4385  These results demonstrate a specific association of SIV and HIV-2 nef, but not HIV-1 nef, with TCRzeta.                                                                                                                                                                                                     \n",
       "4594  Copyright 1999 Academic Press.                                                                                                                                                                                                                                                                              \n",
       "4764  STUDY DESIGN: Retrospective review.                                                                                                                                                                                                                                                                         \n",
       "4819  Acad.                                                                                                                                                                                                                                                                                                       "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option('display.max_colwidth', -1)\n",
    "df_bc2_gm_text_high_sim = df_bc2_gm_text.query(\"Unigram > 75\")[[\"Unigram\", \"Unigram_detail\"]]\n",
    "df_bc2_gm_text_high_sim[\"Test\"] = df_bc2_gm_text_high_sim[\"Unigram_detail\"].apply(lambda x:x[0])\n",
    "df_bc2_gm_text_high_sim[\"Train\"] = df_bc2_gm_text_high_sim[\"Unigram_detail\"].apply(lambda x:x[1])\n",
    "df_bc2_gm_text_high_sim[[\"Unigram\", \"Train\", \"Test\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>47.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>97.705854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>6.231332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>76.454959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Unigram\n",
       "count  47.000000 \n",
       "mean   97.705854 \n",
       "std    6.231332  \n",
       "min    76.454959 \n",
       "25%    100.000000\n",
       "50%    100.000000\n",
       "75%    100.000000\n",
       "max    100.000000"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc2_gm_text_high_sim.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{rll}\n",
      "\\toprule\n",
      " Unigram &                                                                                                                                                                                                                                                                                                        Test &                                                                                                                                                                                                                                                                                                                               Train \\\\\n",
      "\\midrule\n",
      "76.45 &  Histological and immunophenotypic studies revealed 12 large cell lymphomas (11 B cell and one T cell), two small noncleaved cell lymphomas (B-cell phenotype), and five low grade B-cell lymphomas (two small lymphocytic and three follicular mixed lymphomas). &  The cases included 35 de novo diffuse aggressive lymphomas (DAL; 19 large-cell, 4 mixed-cell, and 12 large-cell immunoblastic), 52 transformed aggressive lymphomas derived from follicular lymphomas (TFL), 42 indolent follicular lymphomas (FL), 14 mantle cell lymphomas (MCL), and 27 small noncleaved cell lymphomas (SNCL). \\\\\n",
      "77.46 &  98, 93-98). &  356, 93-98]. \\\\\n",
      "81.65 &  Free protein S deficiency in acute ischemic stroke. &  Ischemic stroke due to protein C deficiency. \\\\\n",
      "83.41 &  In stage I, histochemistry for copper was positive in 11 out of 21 cases: 6 cases were T+; 1 case R+ and 2 cases O+; 2 cases were T+, R+, O+. &  3 cases \\\\\n",
      "86.60 &  STUDY DESIGN: Retrospective review. &  DESIGN: Retrospective study. \\\\\n",
      "86.60 &  Non-dialyzable transfer factor &  Dialyzable transfer factor. \\\\\n",
      "100.00 &  Recently we have performed a detailed analysis of specific neuronal populations affected by the mutation which shed new light on the role of Krox-20 in the segmentation and on the physiological consequences of its inactivation. &  Recently we have performed a detailed analysis of specific neuronal populations affected by the mutation which shed new light on the role of Krox-20 in the segmentation and on the physiological consequences of its inactivation. \\\\\n",
      "100.00 &  Slowly adapting type I mechanoreceptor discharge as a function of dynamic force versus dynamic displacement of glabrous skin of raccoon and squirrel monkey hand. &  Slowly adapting type I mechanoreceptor discharge as a function of dynamic force versus dynamic displacement of glabrous skin of raccoon and squirrel monkey hand. \\\\\n",
      "100.00 &  The recruitment of constitutively phosphorylated p185(neu) and the activated mitogenic pathway proteins to this membrane-microfilament interaction site provides a physical model for integrating the assembly of the mitogenic pathway with the transmission of growth factor signal to the cytoskeleton. &  The recruitment of constitutively phosphorylated p185(neu) and the activated mitogenic pathway proteins to this membrane-microfilament interaction site provides a physical model for integrating the assembly of the mitogenic pathway with the transmission of growth factor signal to the cytoskeleton. \\\\\n",
      "100.00 &  A heterologous promoter construct containing three repeats of a consensus Sp1 site, cloned upstream of a single copy of the ZII (CREB/ AP1) element from the BZLF1 promoter linked to the beta-globin TATA box, exhibited phorbol ester inducibility. &  A heterologous promoter construct containing three repeats of a consensus Sp1 site, cloned upstream of a single copy of the ZII (CREB/ AP1) element from the BZLF1 promoter linked to the beta-globin TATA box, exhibited phorbol ester inducibility. \\\\\n",
      "100.00 &  The reconstituted RNA polymerases containing the mutant alpha subunits were examined for their response to transcription activation by cAMP-CRP and the rrnBP1 UP element. &  The reconstituted RNA polymerases containing the mutant alpha subunits were examined for their response to transcription activation by cAMP-CRP and the rrnBP1 UP element. \\\\\n",
      "100.00 &  Analysis of 1 Mb of published sequence from the region of conserved synteny on human chromosome 5q31-q33 identified 45 gene candidates, including 35 expressed genes in the human IL-4 cytokine gene cluster. &  Analysis of 1 Mb of published sequence from the region of conserved synteny on human chromosome 5q31-q33 identified 45 gene candidates, including 35 expressed genes in the human IL-4 cytokine gene cluster. \\\\\n",
      "100.00 &  Although RAD17, RAD24 and MEC3 are not required for cell cycle arrest when S phase is inhibited by hydroxyurea (HU), they do contribute to the viability of yeast cells grown in the presence of HU, possibly because they are required for the repair of HU-induced DNA damage. &  Although RAD17, RAD24 and MEC3 are not required for cell cycle arrest when S phase is inhibited by hydroxyurea (HU), they do contribute to the viability of yeast cells grown in the presence of HU, possibly because they are required for the repair of HU-induced DNA damage. \\\\\n",
      "100.00 &  The promoter for HMG-CoA synthase contains two binding sites for the sterol regulatory element-binding proteins (SREBPs). &  The promoter for HMG-CoA synthase contains two binding sites for the sterol regulatory element-binding proteins (SREBPs). \\\\\n",
      "100.00 &  Coronary vasoconstriction caused by endothelin-1 is enhanced by ischemia-reperfusion and by norepinephrine present in concentrations typically observed after neonatal cardiopulmonary bypass. &  Coronary vasoconstriction caused by endothelin-1 is enhanced by ischemia-reperfusion and by norepinephrine present in concentrations typically observed after neonatal cardiopulmonary bypass. \\\\\n",
      "100.00 &  (LH P < 0.05, LH/FSH P < 0.01). &  (LH P < 0.05, LH/FSH P < 0.01). \\\\\n",
      "100.00 &  Determinants of recurrent ischaemia and revascularisation procedures after thrombolysis with recombinant tissue plasminogen activator in primary coronary occlusion. &  Determinants of recurrent ischaemia and revascularisation procedures after thrombolysis with recombinant tissue plasminogen activator in primary coronary occlusion. \\\\\n",
      "100.00 &  The human SHBG proximal promoter was analyzed by DNase I footprinting, and the functional significance of 6 footprinted regions (FP1-FP6) within the proximal promoter was studied in human HepG2 hepatoblastoma cells. &  The human SHBG proximal promoter was analyzed by DNase I footprinting, and the functional significance of 6 footprinted regions (FP1-FP6) within the proximal promoter was studied in human HepG2 hepatoblastoma cells. \\\\\n",
      "100.00 &  Biol. &  Biol. \\\\\n",
      "100.00 &  Copyright 1999 Academic Press. &  Copyright 1999 Academic Press. \\\\\n",
      "100.00 &  These results demonstrate a specific association of SIV and HIV-2 nef, but not HIV-1 nef, with TCRzeta. &  These results demonstrate a specific association of SIV and HIV-2 nef, but not HIV-1 nef, with TCRzeta. \\\\\n",
      "100.00 &  Urease activity, judged as the amount of ammonia production from urea, could be measured at 25 ng per tube (S/N = 1.5) with Jack bean meal urease. &  Urease activity, judged as the amount of ammonia production from urea, could be measured at 25 ng per tube (S/N = 1.5) with Jack bean meal urease. \\\\\n",
      "100.00 &  Copyright 1999 Academic Press. &  Copyright 1999 Academic Press. \\\\\n",
      "100.00 &  IV. &  IV. \\\\\n",
      "100.00 &  Copyright 1998 Academic Press. &  Copyright 1998 Academic Press. \\\\\n",
      "100.00 &  IV. &  IV. \\\\\n",
      "100.00 &  Biol. &  Biol. \\\\\n",
      "100.00 &  Copyright 1999 Academic Press. &  Copyright 1999 Academic Press. \\\\\n",
      "100.00 &  Copyright 1998 Academic Press. &  Copyright 1998 Academic Press. \\\\\n",
      "100.00 &  Copyright 2000 Academic Press. &  Copyright 2000 Academic Press. \\\\\n",
      "100.00 &  1988). &  (1988) J. \\\\\n",
      "100.00 &  Biol. &  Biol. \\\\\n",
      "100.00 &  Acad. &  Acad. \\\\\n",
      "100.00 &  Virol. &  Virol. \\\\\n",
      "100.00 &  1995. &  (1995) J. \\\\\n",
      "100.00 &  Natl. &  Natl. \\\\\n",
      "100.00 &  Copyright 1999 Academic Press. &  Copyright 1999 Academic Press. \\\\\n",
      "100.00 &  The activated glucocorticoid receptor forms a complex with Stat5 and enhances Stat5-mediated transcriptional induction. &  The activated glucocorticoid receptor forms a complex with Stat5 and enhances Stat5-mediated transcriptional induction. \\\\\n",
      "100.00 &  Copyright 1999 Academic Press. &  Copyright 1999 Academic Press. \\\\\n",
      "100.00 &  Chem. &  Chem. \\\\\n",
      "100.00 &  Appl. &  Appl. \\\\\n",
      "100.00 &  Copyright 1998 Academic Press. &  Copyright 1998 Academic Press. \\\\\n",
      "100.00 &  Sci. &  Sci. \\\\\n",
      "100.00 &  (1992) J. &  (1992) J. \\\\\n",
      "100.00 &  Acad. &  Acad. \\\\\n",
      "100.00 &  Mutational analysis of yeast CEG1 demonstrated that four of the five conserved motifs are essential for capping enzyme function in vivo. &  Mutational analysis of yeast CEG1 demonstrated that four of the five conserved motifs are essential for capping enzyme function in vivo. \\\\\n",
      "100.00 &  We also show that in fusions with the DNA binding domain of GAL4, full activity requires the entire BHV-alpha TIF, although both amino and carboxyl termini display some activity on their own. &  We also show that in fusions with the DNA binding domain of GAL4, full activity requires the entire BHV-alpha TIF, although both amino and carboxyl termini display some activity on their own. \\\\\n",
      "\\bottomrule\n",
      "\\end{tabular}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(df_bc2_gm_text_high_sim[[\"Unigram\",  \"Test\",\"Train\",]].sort_values(by=\"Unigram\").to_latex(index=False, float_format=lambda x: \"{:.2f}\".format(x)\n",
    "))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. AIMED (Random)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "aimed_file = os.path.join(base_data_dir, \"AIMedFull_preprocessed.json\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "from aimed_random import AIMedRandom\n",
    "\n",
    "result_score, result_detail = AIMedRandom().run_similarity_comparer(aimed_file)\n",
    "df_aimed_random = scores_to_df(result_score,result_detail, \"AIMED (R)\", \"\", \"REL\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>data_set</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">AIMED (R)</th>\n",
       "      <th>count</th>\n",
       "      <td>584.000000</td>\n",
       "      <td>584.000000</td>\n",
       "      <td>584.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>96.947936</td>\n",
       "      <td>82.294414</td>\n",
       "      <td>73.146194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>11.921198</td>\n",
       "      <td>18.519621</td>\n",
       "      <td>21.285885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>27.498597</td>\n",
       "      <td>8.247861</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>77.777778</td>\n",
       "      <td>66.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>86.666667</td>\n",
       "      <td>77.777778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>93.801712</td>\n",
       "      <td>87.053315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Unigram      Bigram     Trigram\n",
       "data_set                                            \n",
       "AIMED (R)  count  584.000000  584.000000  584.000000\n",
       "           mean   96.947936   82.294414   73.146194 \n",
       "           std    11.921198   18.519621   21.285885 \n",
       "           min    27.498597   8.247861    0.000000  \n",
       "           25%    100.000000  77.777778   66.666667 \n",
       "           50%    100.000000  86.666667   77.777778 \n",
       "           75%    100.000000  93.801712   87.053315 \n",
       "           max    100.000000  100.000000  100.000000"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_aimed_random.groupby([\"data_set\"]).describe().stack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(Using the yeast two-hybrid system, a genetic assay for studying protein-protein interactions, we have examined and compared the interaction of the PROTEIN1 receptor ( PROTEIN2 ) and the PROTEIN ( PROTEIN ) with their two known substrates PROTEIN and the PROTEIN substrate-1 ( PROTEIN )., Using the yeast two-hybrid system, a genetic assay for studying protein-protein interactions, we have examined and compared the interaction of the PROTEIN1 receptor ( PROTEIN ) and the PROTEIN2 ( PROTEIN ) with their two known substrates PROTEIN and the PROTEIN substrate-1 ( PROTEIN ).)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>478</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(These data demonstrate that PROTEIN1 but not PROTEIN2 or PROTEIN is directly recruited to the ligand-activated PROTEIN by binding to specific but redundant receptor intracellular domain sequences containing phosphotyrosine., These data demonstrate that PROTEIN1 but not PROTEIN or PROTEIN is directly recruited to the ligand-activated PROTEIN2 by binding to specific but redundant receptor intracellular domain sequences containing phosphotyrosine.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(Previously, a mammalian protein that directly associates with PROTEIN1 -rapamycin has been identified and its encoding gene has been cloned from both human (designated PROTEIN ) [Brown, E.J., Albers, M.W., Shin, T.B., Ichikawa, K., Keith, C.T., Lane, W.S. &amp; Schreiber, S.L. (1994) Nature (London) 369, 756-758] and rat (designated PROTEIN2 ) [Sabatini, D.M., Erdjument-Bromage, H., Lui, M., Tempst, P. &amp; Snyder, S.H. (1994) Cell 78, 35-43]., Previously, a mammalian protein that directly associates with PROTEIN1 -rapamycin has been identified and its encoding gene has been cloned from both human (designated PROTEIN2 ) [Brown, E.J., Albers, M.W., Shin, T.B., Ichikawa, K., Keith, C.T., Lane, W.S. &amp; Schreiber, S.L. (1994) Nature (London) 369, 756-758] and rat (designated PROTEIN ) [Sabatini, D.M., Erdjument-Bromage, H., Lui, M., Tempst, P. &amp; Snyder, S.H. (1994) Cell 78, 35-43].)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(Thus, during PROTEIN1 -mediated suppression of cell proliferation, PROTEIN and PROTEIN2 may be important for coordinating cell-cycle progression, DNA replication and repair of damaged DNA., Thus, during PROTEIN -mediated suppression of cell proliferation, PROTEIN1 and PROTEIN2 may be important for coordinating cell-cycle progression, DNA replication and repair of damaged DNA.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>100.0</td>\n",
       "      <td>(We have found a COOH-terminal 37-kD fragment of PROTEIN1 sufficient to interact with translated PROTEIN and its homologues, suggesting that the PROTEIN2 binding site on PROTEIN occurs on a region that is conserved among the three syntrophin homologues., We have found a COOH-terminal 37-kD fragment of PROTEIN sufficient to interact with translated PROTEIN and its homologues, suggesting that the PROTEIN1 binding site on PROTEIN2 occurs on a region that is conserved among the three syntrophin homologues.)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unigram  \\\n",
       "212  100.0     \n",
       "478  100.0     \n",
       "209  100.0     \n",
       "313  100.0     \n",
       "300  100.0     \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Unigram_detail  \n",
       "212  (Using the yeast two-hybrid system, a genetic assay for studying protein-protein interactions, we have examined and compared the interaction of the PROTEIN1 receptor ( PROTEIN2 ) and the PROTEIN ( PROTEIN ) with their two known substrates PROTEIN and the PROTEIN substrate-1 ( PROTEIN )., Using the yeast two-hybrid system, a genetic assay for studying protein-protein interactions, we have examined and compared the interaction of the PROTEIN1 receptor ( PROTEIN ) and the PROTEIN2 ( PROTEIN ) with their two known substrates PROTEIN and the PROTEIN substrate-1 ( PROTEIN ).)                                                                                                                                                                                                                                                                                                                      \n",
       "478  (These data demonstrate that PROTEIN1 but not PROTEIN2 or PROTEIN is directly recruited to the ligand-activated PROTEIN by binding to specific but redundant receptor intracellular domain sequences containing phosphotyrosine., These data demonstrate that PROTEIN1 but not PROTEIN or PROTEIN is directly recruited to the ligand-activated PROTEIN2 by binding to specific but redundant receptor intracellular domain sequences containing phosphotyrosine.)                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
       "209  (Previously, a mammalian protein that directly associates with PROTEIN1 -rapamycin has been identified and its encoding gene has been cloned from both human (designated PROTEIN ) [Brown, E.J., Albers, M.W., Shin, T.B., Ichikawa, K., Keith, C.T., Lane, W.S. & Schreiber, S.L. (1994) Nature (London) 369, 756-758] and rat (designated PROTEIN2 ) [Sabatini, D.M., Erdjument-Bromage, H., Lui, M., Tempst, P. & Snyder, S.H. (1994) Cell 78, 35-43]., Previously, a mammalian protein that directly associates with PROTEIN1 -rapamycin has been identified and its encoding gene has been cloned from both human (designated PROTEIN2 ) [Brown, E.J., Albers, M.W., Shin, T.B., Ichikawa, K., Keith, C.T., Lane, W.S. & Schreiber, S.L. (1994) Nature (London) 369, 756-758] and rat (designated PROTEIN ) [Sabatini, D.M., Erdjument-Bromage, H., Lui, M., Tempst, P. & Snyder, S.H. (1994) Cell 78, 35-43].)  \n",
       "313  (Thus, during PROTEIN1 -mediated suppression of cell proliferation, PROTEIN and PROTEIN2 may be important for coordinating cell-cycle progression, DNA replication and repair of damaged DNA., Thus, during PROTEIN -mediated suppression of cell proliferation, PROTEIN1 and PROTEIN2 may be important for coordinating cell-cycle progression, DNA replication and repair of damaged DNA.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "300  (We have found a COOH-terminal 37-kD fragment of PROTEIN1 sufficient to interact with translated PROTEIN and its homologues, suggesting that the PROTEIN2 binding site on PROTEIN occurs on a region that is conserved among the three syntrophin homologues., We have found a COOH-terminal 37-kD fragment of PROTEIN sufficient to interact with translated PROTEIN and its homologues, suggesting that the PROTEIN1 binding site on PROTEIN2 occurs on a region that is conserved among the three syntrophin homologues.)                                                                                                                                                                                                                                                                                                                                                                                          "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_aimed_random.sort_values(by=[\"Unigram\"], ascending=False)[[\"Unigram\", \"Unigram_detail\"]].head(n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Unigram           27.4986                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "Unigram_detail    (While the circulating PROTEIN1 of renal origin is acutely activated to contribute to the accomplishment of short term homeostatic reactions, tissue PROTEIN2 , in paracrine or autocrine ways, exerts local-regional adaptive actions of long duration., In the end we refer to the relations between circulating PROTEIN of renal origin and the extrarenal PROTEIN1 specific to tissue underlying the fact that the two PROTEIN2 components cooperate and constitute a unitary hormonal system, regulating the major functions of the organism and maintaining the main homeostatic equilibria.)\n",
       "Name: 34, dtype: object"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_aimed_random.sort_values(by=[\"Unigram\"], ascending=True).head(n=1)[[\"Unigram\", \"Unigram_detail\"]].iloc[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. AIMED (Unqiue Document)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "from aimed_uniquedoc import AIMedUniqueDoc\n",
    "\n",
    "result_score, result_detail = AIMedUniqueDoc().run_similarity_comparer(aimed_file)\n",
    "df_aimed_unique = scores_to_df(result_score,result_detail, \"AIMED (U)\", \"\", \"REL\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>data_set</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">AIMED (U)</th>\n",
       "      <th>count</th>\n",
       "      <td>498.000000</td>\n",
       "      <td>498.000000</td>\n",
       "      <td>498.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>67.137026</td>\n",
       "      <td>36.067096</td>\n",
       "      <td>20.770509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>15.317615</td>\n",
       "      <td>19.457675</td>\n",
       "      <td>18.481003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>28.426762</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>53.935989</td>\n",
       "      <td>21.629523</td>\n",
       "      <td>8.679295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>68.640647</td>\n",
       "      <td>31.622777</td>\n",
       "      <td>14.629372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>78.920519</td>\n",
       "      <td>44.721360</td>\n",
       "      <td>26.726124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>87.017654</td>\n",
       "      <td>80.489529</td>\n",
       "      <td>71.632286</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Unigram      Bigram     Trigram\n",
       "data_set                                            \n",
       "AIMED (U)  count  498.000000  498.000000  498.000000\n",
       "           mean   67.137026   36.067096   20.770509 \n",
       "           std    15.317615   19.457675   18.481003 \n",
       "           min    28.426762   0.000000    0.000000  \n",
       "           25%    53.935989   21.629523   8.679295  \n",
       "           50%    68.640647   31.622777   14.629372 \n",
       "           75%    78.920519   44.721360   26.726124 \n",
       "           max    87.017654   80.489529   71.632286 "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_aimed_unique.groupby([\"data_set\"]).describe().stack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('Using antibody-mediated immunofluorescence copatching of epitope-tagged receptors, we provide evidence in live cells for preexisting heteromeric ( PROTEIN / PROTEIN and PROTEIN / PROTEIN1 ) and homomeric ( PROTEIN / PROTEIN , PROTEIN / PROTEIN , PROTEIN / PROTEIN , and also PROTEIN2 / PROTEIN ) oligomers in the absence of ligand.',\n",
       " ' PROTEIN1 is a potent, tight-binding inhibitor of Cdks and can inhibit the phosphorylation of PROTEIN by PROTEIN - PROTEIN , PROTEIN - PROTEIN , PROTEIN2 - PROTEIN , and PROTEIN - PROTEIN complexes.')"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_aimed_unique.sort_values(by=[\"Unigram\"], ascending=False).head(n=1)[\"Unigram_detail\"].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('In contrast, in normal nasal tissues, PROTEIN1 labeling was only found in the vascular wall, and the expression was weaker--a finding demonstrating that PROTEIN2 is upregulated in nasal polyps.',\n",
       " 'We found PROTEIN1 mRNA in several peripheral tissues, and detected PROTEIN2 protein on cultured vascular endothelial cells.')"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_aimed_unique.sort_values(by=[\"Unigram\"], ascending=True).head(n=1)[\"Unigram_detail\"].iloc[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. SST2 Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "sst2_train_file = os.path.join(base_data_dir, \"train.tsv\")\n",
    "sst2_test_file = os.path.join(base_data_dir, \"test.tsv\")\n",
    "sst2_sentiment_labels_file  =  os.path.join(base_data_dir, \"sentiment_labels.txt\")\n",
    "sst2_dictionary_file  =  os.path.join(base_data_dir, \"dictionary.txt\") \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sst2_dataset import SST2Dataset\n",
    "result_score, result_detail =  SST2Dataset().run_similarity_comparer(sst2_train_file,sst2_test_file)\n",
    "df_sst2 = scores_to_df(result_score,result_detail, \"SST2\", \"\", \"CLS\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>data_set</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">SST2</th>\n",
       "      <th>count</th>\n",
       "      <td>1821.000000</td>\n",
       "      <td>1821.000000</td>\n",
       "      <td>1821.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>46.055815</td>\n",
       "      <td>17.378379</td>\n",
       "      <td>1.388049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>14.067934</td>\n",
       "      <td>18.009431</td>\n",
       "      <td>6.733712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>36.514837</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>43.643578</td>\n",
       "      <td>16.666667</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>53.452248</td>\n",
       "      <td>30.151134</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>70.710678</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Unigram       Bigram      Trigram\n",
       "data_set                                             \n",
       "SST2     count  1821.000000  1821.000000  1821.000000\n",
       "         mean   46.055815    17.378379    1.388049   \n",
       "         std    14.067934    18.009431    6.733712   \n",
       "         min    0.000000     0.000000     0.000000   \n",
       "         25%    36.514837    0.000000     0.000000   \n",
       "         50%    43.643578    16.666667    0.000000   \n",
       "         75%    53.452248    30.151134    0.000000   \n",
       "         max    100.000000   100.000000   70.710678  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sst2.groupby([\"data_set\"]).describe().stack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>462</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(now , if it only had a brain ., the brain )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1392</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(go see it and enjoy ., enjoy this )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1502</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(is n't it great ?, great )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(how did it ever get made ?, did they not )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(i loved it !, loved )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>703</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(still , i thought it could have been more ., well-thought )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(but he somehow pulls it off ., pulls off enough )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>972</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(it 's painful ., 's that painful . )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>937</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(it 's laughing at us ., laughing )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>776</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(she must have a very strong back ., strong as always )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>785</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(almost everything else is wan ., wan )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>641</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(this is one for the ages ., of all ages )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1279</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(nothing about it fits ., fits the bill )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(ridiculous ., ridiculous )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(does n't amount to much of anything ., that does not move )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(it 's astonishing ., is astonishing )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(brilliant ! ', most brilliant )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(so what is the point ?, i do n't see the point )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1374</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(a very bad sign ., a bad sign )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(one of the best of the year ., is one of the year 's best )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(well worth the time ., is worth your time )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1099</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(it 's still worth a look ., worth the look )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>(good movie ., , it 's still not a good movie )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>856</th>\n",
       "      <td>86.602540</td>\n",
       "      <td>(a great idea becomes a not-great movie ., great idea )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>86.602540</td>\n",
       "      <td>(i just did n't care as much for the story ., just did n't care )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>86.602540</td>\n",
       "      <td>(the acting alone is worth the price of admission ., worth the price of admission )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1305</th>\n",
       "      <td>86.602540</td>\n",
       "      <td>(( an ) hilarious romantic comedy ., , often hilarious romantic jealousy comedy )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1407</th>\n",
       "      <td>86.602540</td>\n",
       "      <td>(the movie makes absolutely no sense ., makes absolutely no sense )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>722</th>\n",
       "      <td>84.223489</td>\n",
       "      <td>(divine secrets of the ya-ya sisterhood may not be exactly divine , but it 's definitely -- defiantly -- ya ya , what with all of those terrific songs and spirited performances ., 's ` divine secrets of the ya-ya sisterhood )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>84.016805</td>\n",
       "      <td>(even by dumb action-movie standards , ballistic : ecks vs. sever is a dumb action movie ., dumb action movie )</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Unigram  \\\n",
       "462   100.000000   \n",
       "1392  100.000000   \n",
       "1502  100.000000   \n",
       "47    100.000000   \n",
       "138   100.000000   \n",
       "703   100.000000   \n",
       "391   100.000000   \n",
       "972   100.000000   \n",
       "937   100.000000   \n",
       "776   100.000000   \n",
       "785   100.000000   \n",
       "641   100.000000   \n",
       "1279  100.000000   \n",
       "835   100.000000   \n",
       "339   100.000000   \n",
       "307   100.000000   \n",
       "169   100.000000   \n",
       "342   100.000000   \n",
       "1374  100.000000   \n",
       "137   100.000000   \n",
       "212   100.000000   \n",
       "1099  100.000000   \n",
       "23    100.000000   \n",
       "856   86.602540    \n",
       "380   86.602540    \n",
       "996   86.602540    \n",
       "1305  86.602540    \n",
       "1407  86.602540    \n",
       "722   84.223489    \n",
       "561   84.016805    \n",
       "\n",
       "                                                                                                                                                                                                                         Unigram_detail  \n",
       "462   (now , if it only had a brain ., the brain )                                                                                                                                                                                       \n",
       "1392  (go see it and enjoy ., enjoy this )                                                                                                                                                                                               \n",
       "1502  (is n't it great ?, great )                                                                                                                                                                                                        \n",
       "47    (how did it ever get made ?, did they not )                                                                                                                                                                                        \n",
       "138   (i loved it !, loved )                                                                                                                                                                                                             \n",
       "703   (still , i thought it could have been more ., well-thought )                                                                                                                                                                       \n",
       "391   (but he somehow pulls it off ., pulls off enough )                                                                                                                                                                                 \n",
       "972   (it 's painful ., 's that painful . )                                                                                                                                                                                              \n",
       "937   (it 's laughing at us ., laughing )                                                                                                                                                                                                \n",
       "776   (she must have a very strong back ., strong as always )                                                                                                                                                                            \n",
       "785   (almost everything else is wan ., wan )                                                                                                                                                                                            \n",
       "641   (this is one for the ages ., of all ages )                                                                                                                                                                                         \n",
       "1279  (nothing about it fits ., fits the bill )                                                                                                                                                                                          \n",
       "835   (ridiculous ., ridiculous )                                                                                                                                                                                                        \n",
       "339   (does n't amount to much of anything ., that does not move )                                                                                                                                                                       \n",
       "307   (it 's astonishing ., is astonishing )                                                                                                                                                                                             \n",
       "169   (brilliant ! ', most brilliant )                                                                                                                                                                                                   \n",
       "342   (so what is the point ?, i do n't see the point )                                                                                                                                                                                  \n",
       "1374  (a very bad sign ., a bad sign )                                                                                                                                                                                                   \n",
       "137   (one of the best of the year ., is one of the year 's best )                                                                                                                                                                       \n",
       "212   (well worth the time ., is worth your time )                                                                                                                                                                                       \n",
       "1099  (it 's still worth a look ., worth the look )                                                                                                                                                                                      \n",
       "23    (good movie ., , it 's still not a good movie )                                                                                                                                                                                    \n",
       "856   (a great idea becomes a not-great movie ., great idea )                                                                                                                                                                            \n",
       "380   (i just did n't care as much for the story ., just did n't care )                                                                                                                                                                  \n",
       "996   (the acting alone is worth the price of admission ., worth the price of admission )                                                                                                                                                \n",
       "1305  (( an ) hilarious romantic comedy ., , often hilarious romantic jealousy comedy )                                                                                                                                                  \n",
       "1407  (the movie makes absolutely no sense ., makes absolutely no sense )                                                                                                                                                                \n",
       "722   (divine secrets of the ya-ya sisterhood may not be exactly divine , but it 's definitely -- defiantly -- ya ya , what with all of those terrific songs and spirited performances ., 's ` divine secrets of the ya-ya sisterhood )  \n",
       "561   (even by dumb action-movie standards , ballistic : ecks vs. sever is a dumb action movie ., dumb action movie )                                                                                                                    "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sst2.sort_values(by=[\"Unigram\"], ascending=False)[[\"Unigram\", \"Unigram_detail\"]].head(n=30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21.82178902359924\n",
      "\n",
      "(\"herzog is obviously looking for a moral to his fable , but the notion that a strong , unified showing among germany and eastern european jews might have changed 20th-century history is undermined by ahola 's inadequate performance .\", 'of the unsung heroes of 20th century ')\n"
     ]
    }
   ],
   "source": [
    "print(*df_sst2.sort_values(by=[\"Unigram\"], ascending=True)[[\"Unigram\",\"Unigram_detail\"]].iloc[9], sep='\\n\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. BC3 Article classification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "bc3_act_train_file = os.path.join(base_data_dir, \"bc3_act_all_records.tsv\")\n",
    "bc3_act_test_file = os.path.join(base_data_dir, \"bc3_act_all_records_test.tsv\")\n",
    "\n",
    "bc3_act_test_eval_file = os.path.join(base_data_dir, \"bc3_act_gold_standard_test.tsv\")\n",
    "\n",
    "bc3_act_train_eval_file = os.path.join(base_data_dir, \"bc3_act_gold_standard.tsv\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "from bc3_article_classification import BC3ArticleClassification\n",
    "bc3actrun = BC3ArticleClassification()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "result_score, result_detail = bc3actrun.run_similarity_comparer(bc3_act_train_file,  bc3_act_test_file)\n",
    "df_bc3_act = scores_to_df(result_score,result_detail, \"BC3ACT\", \"\", \"CLS\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>6000.000000</td>\n",
       "      <td>6000.000000</td>\n",
       "      <td>6000.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>26.760271</td>\n",
       "      <td>6.907863</td>\n",
       "      <td>1.807134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>9.325118</td>\n",
       "      <td>5.474373</td>\n",
       "      <td>1.728708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>6.277610</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>20.094071</td>\n",
       "      <td>3.474651</td>\n",
       "      <td>0.835709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>25.677218</td>\n",
       "      <td>5.298225</td>\n",
       "      <td>1.476668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>31.914209</td>\n",
       "      <td>8.530364</td>\n",
       "      <td>2.352489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>75.011317</td>\n",
       "      <td>51.197969</td>\n",
       "      <td>18.973666</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Unigram       Bigram      Trigram\n",
       "count  6000.000000  6000.000000  6000.000000\n",
       "mean   26.760271    6.907863     1.807134   \n",
       "std    9.325118     5.474373     1.728708   \n",
       "min    6.277610     0.000000     0.000000   \n",
       "25%    20.094071    3.474651     0.835709   \n",
       "50%    25.677218    5.298225     1.476668   \n",
       "75%    31.914209    8.530364     2.352489   \n",
       "max    75.011317    51.197969    18.973666  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc3_act.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1819</th>\n",
       "      <td>6.277610</td>\n",
       "      <td>(Pollination services underpin sustainability of restored ecosystems. Yet, outside of agri-environments, effective restoration of pollinator services in ecological restoration has received little attention. This deficiency in the knowledge needed to restore pollinator capability represents a major liability in restoration programs, particularly in regions where specialist invertebrate and vertebrate pollinators exist, such as global biodiversity hotspots. When compounded with the likely negative impacts of climate change on pollination services, the need to understand and manage pollinator services in restoration becomes paramount., Fhit protein is lost in most cancers, its restoration suppresses tumorigenicity, and virus-mediated FHIT gene therapy induces apoptosis and suppresses tumors in preclinical models. We have used protein cross-linking and proteomics methods to characterize a Fhit protein complex involved in triggering Fhit-mediated apoptosis. The complex includes Hsp60 and Hsp10 that mediate Fhit stability and may affect import into mitochondria, where it interacts with ferredoxin reductase, responsible for transferring electrons from NADPH to cytochrome P450 via ferredoxin. Viral-mediated Fhit restoration increases production of intracellular reactive oxygen species, followed by increased apoptosis of lung cancer cells under oxidative stress conditions; conversely, Fhit-negative cells escape apoptosis, carrying serious oxidative DNA damage that may contribute to an increased mutation rate. Characterization of Fhit interacting proteins has identified direct effectors of the Fhit-mediated apoptotic pathway that is lost in most cancers through loss of Fhit.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5631</th>\n",
       "      <td>6.507914</td>\n",
       "      <td>(The Waorani may have the highest rate of homicide of any society known to anthropology. We interviewed 121 Waorani elders of both sexes to obtain genealogical information and recollections of raids in which they and their relatives participated. We also obtained complete raiding histories of 95 warriors. An analysis of the raiding histories, marital trajectories, and reproductive histories of these men reveals that more aggressive warriors have lower indices of reproductive success than their milder brethren. This result contrasts the findings of Chagnon [Chagnon N (1988) Science 239:985-992] for the Yanomamo. We suggest that the spacing of revenge raids may be involved in the explanation of why the consequences of aggressiveness differ between these 2 warlike lowland South American peoples., Until now, Rho proteins were known as GTPases involved in cell polarity and morphogenesis. In a recent issue of Cell, Espinosa and coworkers show that RhoBTB3, a member of this family, is an ATPase involved in endosome-to-Golgi transport.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3184</th>\n",
       "      <td>6.929785</td>\n",
       "      <td>(We compile over 270 wildlife counts of Kenya's wildlife populations conducted over the last 30 years to compare trends in national parks and reserves with adjacent ecosystems and country-wide trends. The study shows the importance of discriminating human-induced changes from natural population oscillations related to rainfall and ecological factors. National park and reserve populations have declined sharply over the last 30 years, at a rate similar to non-protected areas and country-wide trends. The protected area losses reflect in part their poor coverage of seasonal ungulate migrations. The losses vary among parks. The largest parks, Tsavo East, Tsavo West and Meru, account for a disproportionate share of the losses due to habitat change and the difficulty of protecting large remote parks. The losses in Kenya's parks add to growing evidence for wildlife declines inside as well as outside African parks. The losses point to the need to quantify the performance of conservation policies and promote integrated landscape practices that combine parks with private and community-based measures., The recent identification of copy-number variation in the human genome has opened up new avenues for the discovery of positional candidate genes underlying complex genetic disorders, especially in the field of psychiatric disease. One major challenge that remains is pinpointing the susceptibility genes in the multitude of disease-associated loci. This challenge may be tackled by reconstruction of functional gene-networks from the genes residing in these loci. We applied this approach to autism spectrum disorder (ASD), and identified the copy-number changes in the DNA of 105 ASD patients and 267 healthy individuals with Illumina Humanhap300 Beadchips. Subsequently, we used a human reconstructed gene-network, Prioritizer, to rank candidate genes in the segmental gains and losses in our autism cohort. This analysis highlighted several candidate genes already known to be mutated in cognitive and neuropsychiatric disorders, including RAI1, BRD1, and LARGE. In addition, the LARGE gene was part of a sub-network of seven genes functioning in glycobiology, present in seven copy-number changes specifically identified in autism patients with limited co-morbidity. Three of these seven copy-number changes were de novo in the patients. In autism patients with a complex phenotype and healthy controls no such sub-network was identified. An independent systematic analysis of 13 published autism susceptibility loci supports the involvement of genes related to glycobiology as we also identified the same or similar genes from those loci. Our findings suggest that the occurrence of genomic gains and losses of genes associated with glycobiology are important contributors to the development of ASD.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4840</th>\n",
       "      <td>7.267012</td>\n",
       "      <td>(The president of the Venezuelan Academy of Physical, Mathematical, and Natural Sciences outlines actions of the Venezuelan government that are interfering with independent scientific research in the country., Memory suppression is investigated with the no-think paradigm, which produces forgetting following repeated practice of not thinking about a memory [Anderson MC, Green C (2001) Nature 410:366-369]. Because the forgotten item is not retrieved even when tested with an independent, semantically related cue, it has been assumed that this forgetting is due to an inhibition process. However, this conclusion is based on a single stage to recall, whereas global memory models, which produce forgetting through a process of interference, include both a sampling and a recovery stage to recall. By assuming that interference exists during recovery, these models can explain cue-independent forgetting. We tested several predictions of this interference explanation of cue-independent forgetting by modifying the think/no-think paradigm. We added a condition where participants quickly pressed enter rather than not thinking. We also manipulated initial memory strength and tested recognition memory. Most importantly, learning to quickly press enter produced as much cue-independent forgetting as no-think instructions. Demonstrating the adequacy of two-stage recall, a simple computational model (SAM-RI) simultaneously captured the original cue, independent cue, and recognition results.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5787</th>\n",
       "      <td>7.674278</td>\n",
       "      <td>(Several cellular chaperones have been shown to affect the propagation of the yeast prions [PSI(+)], [PIN(+)] and [URE3]. Ssa1 and Ssa2 are Hsp70 family chaperones that generally cause pro-[PSI(+)] effects, since dominant-negative mutants of Ssa1 or Ssa2 cure [PSI(+)], and overexpression of Ssa1 enhances de novo [PSI(+)] appearance and prevents curing by excess Hsp104. In contrast, Ssa1 was shown to have anti-[URE3] effects, since overexpression of Ssa1 cures [URE3]. Here we show that excess Ssa1 or Ssa2 can also cure [PSI(+)]. This curing is enhanced in the presence of [PIN(+)]. During curing, Sup35-GFP fluorescent aggregates get bigger and fewer in number, which leads to their being diluted out during cell division, a phenotype that was also observed during the curing of [PSI(+)] by certain variants of [PIN(+)]. The sizes of the detergent-resistant [PSI(+)] prion oligomers increase during [PSI(+)] curing by excess Ssa1. Excess Ssa1 likewise leads to an increase in oligomer sizes of low, medium and very high [PIN(+)] variants. While these phenotypes are also caused by inhibition of Hsp104 or Sis1, the overexpression of Ssa1 did not cause any change in Hsp104 or Sis1 levels., High NaCl rapidly activates p38 MAPK by phosphorylating it, the phosphorylation presumably being regulated by a balance of kinases and phosphatases. Kinases are known, but the phosphatases are uncertain. Our initial purpose was to identify the phosphatases. We find that in HEK293 cells transient overexpression of MAPK phosphatase-1 (MKP-1), a dual-specificity phosphatase, inhibits high NaCl-induced phosphorylation of p38, and that overexpression of a dominant negative mutant of MKP-1 does the opposite. High NaCl lowers MKP-1 activity by increasing reactive oxygen species, which directly inhibit MKP-1, and by reducing binding of MKP-1 to p38. Because inhibition of p38 is reported to reduce hypertonicity-induced activation of the osmoprotective transcription factor, TonEBP/OREBP, we anticipated that MKP-1 expression might also. However, overexpression of MKP-1 has no significant effect on Ton EBP/OREBP activity. This paradox is explained by opposing effects of p38alpha and p38delta, both of which are activated by high NaCl and inhibited by MKP-1. Thus, we find that overexpression of p38alpha increases high NaCl-induced TonEBP/OREBP activity, but overexpression of p38delta reduces it. Also, siRNA-mediated knockdown of p38delta enhances the activation of TonEBP/OREBP. We conclude that high NaCl inhibits MKP-1, which contributes to the activation of p38. However, opposing actions of p38alpha and p38delta negate any effect on TonEBP/OREBP activity. Thus, activation of p38 isoforms by hypertonicity does not contribute to activation of TonEBP/OREBP because of opposing effects of p38alpha and p38delta, and effects of inhibitors of p38 depend on which isoform is affected, which can be misleading.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2382</th>\n",
       "      <td>7.840151</td>\n",
       "      <td>(BACKGROUND AND AIMS: Celiac sprue is a life-long disease characterized by an intestinal inflammatory response to dietary gluten. A gluten-free diet is an effective treatment for most patients, but accidental ingestion of gluten is common, leading to incomplete recovery or relapse. Food-grade proteases capable of detoxifying moderate quantities of dietary gluten could mitigate this problem. METHODS: We evaluated the gluten detoxification properties of two food-grade enzymes, aspergillopepsin (ASP) from Aspergillus niger and dipeptidyl peptidase IV (DPPIV) from Aspergillus oryzae. The ability of each enzyme to hydrolyze gluten was tested against synthetic gluten peptides, a recombinant gluten protein, and simulated gastric digests of whole gluten and whole-wheat bread. Reaction products were analyzed by mass spectrometry, HPLC, ELISA with a monoclonal antibody that recognizes an immunodominant gluten epitope, and a T cell proliferation assay. RESULTS: ASP markedly enhanced gluten digestion relative to pepsin, and cleaved recombinant alpha2-gliadin at multiple sites in a non-specific manner. When used alone, neither ASP nor DPPIV efficiently cleaved synthetic immunotoxic gluten peptides. This lack of specificity for gluten was especially evident in the presence of casein, a competing dietary protein. However, supplementation of ASP with DPPIV enabled detoxification of moderate amounts of gluten in the presence of excess casein and in whole-wheat bread. ASP was also effective at enhancing the gluten-detoxifying efficacy of cysteine endoprotease EP-B2 under simulated gastric conditions. CONCLUSIONS: Clinical studies are warranted to evaluate whether a fixed dose ratio combination of ASP and DPPIV can provide near-term relief for celiac patients suffering from inadvertent gluten exposure. Due to its markedly greater hydrolytic activity against gluten than endogenous pepsin, food-grade ASP may also augment the activity of therapeutically relevant doses of glutenases such as EP-B2 and certain prolyl endopeptidases., AIM: To determine the prevalence of a new set of anti-glycan and anti-outer membrane protein (anti-OMP) antibodies in a Hungarian cohort of adult Celiac disease (CD) patients. METHODS: 190 consecutive CD patients [M/F: 71/119, age:39.9 (SD:14.1) years], 100 healthy, and 48 gastrointestinal controls were tested for glycan anti-Saccharomyces cerevisiae (gASCA), anti-laminaribioside (ALCA), anti-chitobioside, anti-mannobioside, anti-OMP antibodies and major NOD2/CARD15 mutations. Thirty out of 82 CD patients enrolled at the time of diagnosis were re-evaluated for the same antibodies after longstanding gluten-free diet (GFD). RESULTS: 65.9% of the CD patients were positive for at least one of the tested antibodies at the time of the diagnosis. Except anti-OMP and ALCA, anti-microbial antibodies were exclusively seen in untreated CD; however, the overall sensitivity was low. Any glycan positivity (LR+: 3.13; 95% CI: 2.08-4.73) was associated with an increased likelihood ratio for diagnosing CD. Significant correlation was found between the levels of anti-glycan and anti-endomysial or anti-transglutaminase antibodies. Anti-glycan positivity was lost after longstanding GFD. Anti-glycan antibody titers were associated with symptoms at presentation, but not the presence of NOD2/CARD15 mutations. Patients with severe malabsorption more frequently had multiple antibodies at diagnosis (P = 0.019). CONCLUSION: The presence of anti-glycan antibodies in CD seems to be secondary to the impaired small bowel mucosa which can lead to increased antigen presentation. Furthermore, anti-glycan positivity may be considered an additional marker of CD and dietary adherence.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2941</th>\n",
       "      <td>7.986700</td>\n",
       "      <td>(The synthesis and photophysical evaluation of two enatiomerially pure dimetallic lanthanide luminescent triple-stranded helicates is described. The two systems, formed from the chiral (R,R) ligand 1 and (S,S) ligand 2, were produced as single species in solution, where the excitation of either the naphthalene antennae or the pyridiyl units gave rise to Eu(III) emission in a variety of solvents. Excitation of the antennae also gave rise to circularly polarized Eu(III) luminescence emissions for Eu(2):1(3) and Eu(2):2(3) that were of equal intensity and opposite sign, confirming their enantiomeric nature in solution providing a basis upon which we were able to assign the absolute configurations of Eu(2):1(3) and Eu(2):2(3)., Toll-like receptors (TLRs) initiate immune responses by recognizing pathogen-associated molecules, but the molecular basis for recognition is poorly understood. In particular, it is unclear how receptor-ligand interactions lead to the initiation of downstream signaling. Here, we describe the mechanism by which TLR3 recognizes its ligand, double-stranded RNA (dsRNA), and forms an active signaling complex. We show that dsRNA binds saturably, specifically, and reversibly to a defined ligand-binding site (or sites) on the TLR3 ectodomain (TLR3ecd). Binding affinities increase with both buffer acidity and ligand size. Purified TLR3ecd protein is exclusively monomeric in solution, but through a highly cooperative process, it forms dimers when bound to dsRNA, and multiple TLR3ecd dimers bind to long dsRNA strands. The smallest dsRNA oligonucleotides that form stable complexes with TLR3ecd (40-50 bp) each bind one TLR3ecd dimer, and these are also the smallest oligonucleotides that efficiently activate TLR3 in cells. We conclude that TLR3 assembles on dsRNA as stable dimers and that the minimal signaling unit is one TLR3 dimer.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3775</th>\n",
       "      <td>8.125283</td>\n",
       "      <td>(The Mori-Zwanzig formalism is an effective tool to derive differential equations describing the evolution of a small number of resolved variables. In this paper we present its application to the derivation of generalized Langevin equations and generalized non-Markovian Fokker-Planck equations. We show how long time scales rates and metastable basins can be extracted from these equations. Numerical algorithms are proposed to discretize these equations. An important aspect is the numerical solution of the orthogonal dynamics equation which is a partial differential equation in a high dimensional space. We propose efficient numerical methods to solve this orthogonal dynamics equation. In addition, we present a projection formalism of the Mori-Zwanzig type that is applicable to discrete maps. Numerical applications are presented from the field of Hamiltonian systems., A new approach for the acquisition of static, wideline (14)N NMR powder patterns is outlined. The method involves the use of frequency-swept pulses which serve two simultaneous functions: (1) broad-band excitation of magnetization and (2) signal enhancement via population transfer. The signal enhancement mechanism is described using numerical simulations and confirmed experimentally. This approach, which we call DEISM (Direct Enhancement of Integer Spin Magnetization), allows high-quality (14)N spectra to be acquired at intermediate field strengths in an uncomplicated way and in a fraction of the time required for previously reported methods.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2948</th>\n",
       "      <td>8.193083</td>\n",
       "      <td>(A cavernous hemangioma of the cecum is a rare vascular malformation but is clinically important because of the possibility of massive bleeding. We report a case of a large cavernous hemangioma with pericolic infiltration in the cecum which was removed successfully using minimally invasive surgery., Hypertension, a major cardiovascular risk factor and cause of mortality worldwide, is thought to arise from primary renal abnormalities. However, the etiology of most cases of hypertension remains unexplained. Vascular tone, an important determinant of blood pressure, is regulated by nitric oxide, which causes vascular relaxation by increasing intracellular cGMP and activating cGMP-dependent protein kinase I (PKGI). Here we show that mice with a selective mutation in the N-terminal protein interaction domain of PKGIalpha display inherited vascular smooth muscle cell abnormalities of contraction, abnormal relaxation of large and resistance blood vessels, and increased systemic blood pressure. Renal function studies and responses to changes in dietary sodium in the PKGIalpha mutant mice are normal. These data reveal that PKGIalpha is required for normal VSMC physiology and support the idea that high blood pressure can arise from a primary abnormality of vascular smooth muscle cell contractile regulation, suggesting a new approach to the diagnosis and therapy of hypertension and cardiovascular diseases.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1036</th>\n",
       "      <td>8.277087</td>\n",
       "      <td>(BACKGROUND: Lévy flights are random walks, the step lengths of which come from probability distributions with heavy power-law tails, such that clusters of short steps are connected by rare long steps. Lévy walks maximise search efficiency of mobile foragers. Recently, several studies raised some concerns about the reliability of the statistical analysis used in previous analyses. Further, it is unclear whether Lévy walks represent adaptive strategies or emergent properties determined by the interaction between foragers and resource distribution. Thus two fundamental questions still need to be addressed: the presence of Lévy walks in the wild and whether or not they represent a form of adaptive behaviour. METHODOLOGY/PRINCIPAL FINDINGS: We studied 235 paths of solitary and clustered (i.e. foraging in group) fallow deer (Dama dama), exploiting the same pasture. We used maximum likelihood estimation for discriminating between a power-tailed distribution and the exponential alternative and rank/frequency plots to discriminate between Lévy walks and composite Brownian walks. We showed that solitary deer perform Lévy searches, while clustered animals did not adopt that strategy. CONCLUSION/SIGNIFICANCE: Our demonstration of the presence of Lévy walks is, at our knowledge, the first available which adopts up-to-date statistical methodologies in a terrestrial mammal. Comparing solitary and clustered deer, we concluded that the Lévy walks of solitary deer represent an adaptation maximising encounter rates with forage resources and not an epiphenomenon induced by a peculiar food distribution., BACKGROUND: A better understanding of the size and abundance of open reading frames (ORFS) in whole genomes may shed light on the factors that control genome complexity. Here we examine the statistical distributions of open reading frames (i.e. distribution of start and stop codons) in the fully sequenced genomes of 297 prokaryotes, and 14 eukaryotes. METHODOLOGY/PRINCIPAL FINDINGS: By fitting mixture models to data from whole genome sequences we show that the size-frequency distributions for ORFS are strikingly similar across prokaryotic and eukaryotic genomes. Moreover, we show that i) a large fraction (60-80%) of ORF size-frequency distributions can be predicted a priori with a stochastic assembly model based on GC content, and that (ii) size-frequency distributions of the remaining &amp;quot;non-random&amp;quot; ORFs are well-fitted by log-normal or gamma distributions, and similar to the size distributions of annotated proteins. CONCLUSIONS/SIGNIFICANCE: Our findings suggest stochastic processes have played a primary role in the evolution of genome complexity, and that common processes govern the conservation and loss of functional genomics units in both prokaryotes and eukaryotes.)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unigram  \\\n",
       "1819  6.277610   \n",
       "5631  6.507914   \n",
       "3184  6.929785   \n",
       "4840  7.267012   \n",
       "5787  7.674278   \n",
       "2382  7.840151   \n",
       "2941  7.986700   \n",
       "3775  8.125283   \n",
       "2948  8.193083   \n",
       "1036  8.277087   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Unigram_detail  \n",
       "1819  (Pollination services underpin sustainability of restored ecosystems. Yet, outside of agri-environments, effective restoration of pollinator services in ecological restoration has received little attention. This deficiency in the knowledge needed to restore pollinator capability represents a major liability in restoration programs, particularly in regions where specialist invertebrate and vertebrate pollinators exist, such as global biodiversity hotspots. When compounded with the likely negative impacts of climate change on pollination services, the need to understand and manage pollinator services in restoration becomes paramount., Fhit protein is lost in most cancers, its restoration suppresses tumorigenicity, and virus-mediated FHIT gene therapy induces apoptosis and suppresses tumors in preclinical models. We have used protein cross-linking and proteomics methods to characterize a Fhit protein complex involved in triggering Fhit-mediated apoptosis. The complex includes Hsp60 and Hsp10 that mediate Fhit stability and may affect import into mitochondria, where it interacts with ferredoxin reductase, responsible for transferring electrons from NADPH to cytochrome P450 via ferredoxin. Viral-mediated Fhit restoration increases production of intracellular reactive oxygen species, followed by increased apoptosis of lung cancer cells under oxidative stress conditions; conversely, Fhit-negative cells escape apoptosis, carrying serious oxidative DNA damage that may contribute to an increased mutation rate. Characterization of Fhit interacting proteins has identified direct effectors of the Fhit-mediated apoptotic pathway that is lost in most cancers through loss of Fhit.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
       "5631  (The Waorani may have the highest rate of homicide of any society known to anthropology. We interviewed 121 Waorani elders of both sexes to obtain genealogical information and recollections of raids in which they and their relatives participated. We also obtained complete raiding histories of 95 warriors. An analysis of the raiding histories, marital trajectories, and reproductive histories of these men reveals that more aggressive warriors have lower indices of reproductive success than their milder brethren. This result contrasts the findings of Chagnon [Chagnon N (1988) Science 239:985-992] for the Yanomamo. We suggest that the spacing of revenge raids may be involved in the explanation of why the consequences of aggressiveness differ between these 2 warlike lowland South American peoples., Until now, Rho proteins were known as GTPases involved in cell polarity and morphogenesis. In a recent issue of Cell, Espinosa and coworkers show that RhoBTB3, a member of this family, is an ATPase involved in endosome-to-Golgi transport.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "3184  (We compile over 270 wildlife counts of Kenya's wildlife populations conducted over the last 30 years to compare trends in national parks and reserves with adjacent ecosystems and country-wide trends. The study shows the importance of discriminating human-induced changes from natural population oscillations related to rainfall and ecological factors. National park and reserve populations have declined sharply over the last 30 years, at a rate similar to non-protected areas and country-wide trends. The protected area losses reflect in part their poor coverage of seasonal ungulate migrations. The losses vary among parks. The largest parks, Tsavo East, Tsavo West and Meru, account for a disproportionate share of the losses due to habitat change and the difficulty of protecting large remote parks. The losses in Kenya's parks add to growing evidence for wildlife declines inside as well as outside African parks. The losses point to the need to quantify the performance of conservation policies and promote integrated landscape practices that combine parks with private and community-based measures., The recent identification of copy-number variation in the human genome has opened up new avenues for the discovery of positional candidate genes underlying complex genetic disorders, especially in the field of psychiatric disease. One major challenge that remains is pinpointing the susceptibility genes in the multitude of disease-associated loci. This challenge may be tackled by reconstruction of functional gene-networks from the genes residing in these loci. We applied this approach to autism spectrum disorder (ASD), and identified the copy-number changes in the DNA of 105 ASD patients and 267 healthy individuals with Illumina Humanhap300 Beadchips. Subsequently, we used a human reconstructed gene-network, Prioritizer, to rank candidate genes in the segmental gains and losses in our autism cohort. This analysis highlighted several candidate genes already known to be mutated in cognitive and neuropsychiatric disorders, including RAI1, BRD1, and LARGE. In addition, the LARGE gene was part of a sub-network of seven genes functioning in glycobiology, present in seven copy-number changes specifically identified in autism patients with limited co-morbidity. Three of these seven copy-number changes were de novo in the patients. In autism patients with a complex phenotype and healthy controls no such sub-network was identified. An independent systematic analysis of 13 published autism susceptibility loci supports the involvement of genes related to glycobiology as we also identified the same or similar genes from those loci. Our findings suggest that the occurrence of genomic gains and losses of genes associated with glycobiology are important contributors to the development of ASD.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "4840  (The president of the Venezuelan Academy of Physical, Mathematical, and Natural Sciences outlines actions of the Venezuelan government that are interfering with independent scientific research in the country., Memory suppression is investigated with the no-think paradigm, which produces forgetting following repeated practice of not thinking about a memory [Anderson MC, Green C (2001) Nature 410:366-369]. Because the forgotten item is not retrieved even when tested with an independent, semantically related cue, it has been assumed that this forgetting is due to an inhibition process. However, this conclusion is based on a single stage to recall, whereas global memory models, which produce forgetting through a process of interference, include both a sampling and a recovery stage to recall. By assuming that interference exists during recovery, these models can explain cue-independent forgetting. We tested several predictions of this interference explanation of cue-independent forgetting by modifying the think/no-think paradigm. We added a condition where participants quickly pressed enter rather than not thinking. We also manipulated initial memory strength and tested recognition memory. Most importantly, learning to quickly press enter produced as much cue-independent forgetting as no-think instructions. Demonstrating the adequacy of two-stage recall, a simple computational model (SAM-RI) simultaneously captured the original cue, independent cue, and recognition results.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "5787  (Several cellular chaperones have been shown to affect the propagation of the yeast prions [PSI(+)], [PIN(+)] and [URE3]. Ssa1 and Ssa2 are Hsp70 family chaperones that generally cause pro-[PSI(+)] effects, since dominant-negative mutants of Ssa1 or Ssa2 cure [PSI(+)], and overexpression of Ssa1 enhances de novo [PSI(+)] appearance and prevents curing by excess Hsp104. In contrast, Ssa1 was shown to have anti-[URE3] effects, since overexpression of Ssa1 cures [URE3]. Here we show that excess Ssa1 or Ssa2 can also cure [PSI(+)]. This curing is enhanced in the presence of [PIN(+)]. During curing, Sup35-GFP fluorescent aggregates get bigger and fewer in number, which leads to their being diluted out during cell division, a phenotype that was also observed during the curing of [PSI(+)] by certain variants of [PIN(+)]. The sizes of the detergent-resistant [PSI(+)] prion oligomers increase during [PSI(+)] curing by excess Ssa1. Excess Ssa1 likewise leads to an increase in oligomer sizes of low, medium and very high [PIN(+)] variants. While these phenotypes are also caused by inhibition of Hsp104 or Sis1, the overexpression of Ssa1 did not cause any change in Hsp104 or Sis1 levels., High NaCl rapidly activates p38 MAPK by phosphorylating it, the phosphorylation presumably being regulated by a balance of kinases and phosphatases. Kinases are known, but the phosphatases are uncertain. Our initial purpose was to identify the phosphatases. We find that in HEK293 cells transient overexpression of MAPK phosphatase-1 (MKP-1), a dual-specificity phosphatase, inhibits high NaCl-induced phosphorylation of p38, and that overexpression of a dominant negative mutant of MKP-1 does the opposite. High NaCl lowers MKP-1 activity by increasing reactive oxygen species, which directly inhibit MKP-1, and by reducing binding of MKP-1 to p38. Because inhibition of p38 is reported to reduce hypertonicity-induced activation of the osmoprotective transcription factor, TonEBP/OREBP, we anticipated that MKP-1 expression might also. However, overexpression of MKP-1 has no significant effect on Ton EBP/OREBP activity. This paradox is explained by opposing effects of p38alpha and p38delta, both of which are activated by high NaCl and inhibited by MKP-1. Thus, we find that overexpression of p38alpha increases high NaCl-induced TonEBP/OREBP activity, but overexpression of p38delta reduces it. Also, siRNA-mediated knockdown of p38delta enhances the activation of TonEBP/OREBP. We conclude that high NaCl inhibits MKP-1, which contributes to the activation of p38. However, opposing actions of p38alpha and p38delta negate any effect on TonEBP/OREBP activity. Thus, activation of p38 isoforms by hypertonicity does not contribute to activation of TonEBP/OREBP because of opposing effects of p38alpha and p38delta, and effects of inhibitors of p38 depend on which isoform is affected, which can be misleading.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
       "2382  (BACKGROUND AND AIMS: Celiac sprue is a life-long disease characterized by an intestinal inflammatory response to dietary gluten. A gluten-free diet is an effective treatment for most patients, but accidental ingestion of gluten is common, leading to incomplete recovery or relapse. Food-grade proteases capable of detoxifying moderate quantities of dietary gluten could mitigate this problem. METHODS: We evaluated the gluten detoxification properties of two food-grade enzymes, aspergillopepsin (ASP) from Aspergillus niger and dipeptidyl peptidase IV (DPPIV) from Aspergillus oryzae. The ability of each enzyme to hydrolyze gluten was tested against synthetic gluten peptides, a recombinant gluten protein, and simulated gastric digests of whole gluten and whole-wheat bread. Reaction products were analyzed by mass spectrometry, HPLC, ELISA with a monoclonal antibody that recognizes an immunodominant gluten epitope, and a T cell proliferation assay. RESULTS: ASP markedly enhanced gluten digestion relative to pepsin, and cleaved recombinant alpha2-gliadin at multiple sites in a non-specific manner. When used alone, neither ASP nor DPPIV efficiently cleaved synthetic immunotoxic gluten peptides. This lack of specificity for gluten was especially evident in the presence of casein, a competing dietary protein. However, supplementation of ASP with DPPIV enabled detoxification of moderate amounts of gluten in the presence of excess casein and in whole-wheat bread. ASP was also effective at enhancing the gluten-detoxifying efficacy of cysteine endoprotease EP-B2 under simulated gastric conditions. CONCLUSIONS: Clinical studies are warranted to evaluate whether a fixed dose ratio combination of ASP and DPPIV can provide near-term relief for celiac patients suffering from inadvertent gluten exposure. Due to its markedly greater hydrolytic activity against gluten than endogenous pepsin, food-grade ASP may also augment the activity of therapeutically relevant doses of glutenases such as EP-B2 and certain prolyl endopeptidases., AIM: To determine the prevalence of a new set of anti-glycan and anti-outer membrane protein (anti-OMP) antibodies in a Hungarian cohort of adult Celiac disease (CD) patients. METHODS: 190 consecutive CD patients [M/F: 71/119, age:39.9 (SD:14.1) years], 100 healthy, and 48 gastrointestinal controls were tested for glycan anti-Saccharomyces cerevisiae (gASCA), anti-laminaribioside (ALCA), anti-chitobioside, anti-mannobioside, anti-OMP antibodies and major NOD2/CARD15 mutations. Thirty out of 82 CD patients enrolled at the time of diagnosis were re-evaluated for the same antibodies after longstanding gluten-free diet (GFD). RESULTS: 65.9% of the CD patients were positive for at least one of the tested antibodies at the time of the diagnosis. Except anti-OMP and ALCA, anti-microbial antibodies were exclusively seen in untreated CD; however, the overall sensitivity was low. Any glycan positivity (LR+: 3.13; 95% CI: 2.08-4.73) was associated with an increased likelihood ratio for diagnosing CD. Significant correlation was found between the levels of anti-glycan and anti-endomysial or anti-transglutaminase antibodies. Anti-glycan positivity was lost after longstanding GFD. Anti-glycan antibody titers were associated with symptoms at presentation, but not the presence of NOD2/CARD15 mutations. Patients with severe malabsorption more frequently had multiple antibodies at diagnosis (P = 0.019). CONCLUSION: The presence of anti-glycan antibodies in CD seems to be secondary to the impaired small bowel mucosa which can lead to increased antigen presentation. Furthermore, anti-glycan positivity may be considered an additional marker of CD and dietary adherence.)  \n",
       "2941  (The synthesis and photophysical evaluation of two enatiomerially pure dimetallic lanthanide luminescent triple-stranded helicates is described. The two systems, formed from the chiral (R,R) ligand 1 and (S,S) ligand 2, were produced as single species in solution, where the excitation of either the naphthalene antennae or the pyridiyl units gave rise to Eu(III) emission in a variety of solvents. Excitation of the antennae also gave rise to circularly polarized Eu(III) luminescence emissions for Eu(2):1(3) and Eu(2):2(3) that were of equal intensity and opposite sign, confirming their enantiomeric nature in solution providing a basis upon which we were able to assign the absolute configurations of Eu(2):1(3) and Eu(2):2(3)., Toll-like receptors (TLRs) initiate immune responses by recognizing pathogen-associated molecules, but the molecular basis for recognition is poorly understood. In particular, it is unclear how receptor-ligand interactions lead to the initiation of downstream signaling. Here, we describe the mechanism by which TLR3 recognizes its ligand, double-stranded RNA (dsRNA), and forms an active signaling complex. We show that dsRNA binds saturably, specifically, and reversibly to a defined ligand-binding site (or sites) on the TLR3 ectodomain (TLR3ecd). Binding affinities increase with both buffer acidity and ligand size. Purified TLR3ecd protein is exclusively monomeric in solution, but through a highly cooperative process, it forms dimers when bound to dsRNA, and multiple TLR3ecd dimers bind to long dsRNA strands. The smallest dsRNA oligonucleotides that form stable complexes with TLR3ecd (40-50 bp) each bind one TLR3ecd dimer, and these are also the smallest oligonucleotides that efficiently activate TLR3 in cells. We conclude that TLR3 assembles on dsRNA as stable dimers and that the minimal signaling unit is one TLR3 dimer.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "3775  (The Mori-Zwanzig formalism is an effective tool to derive differential equations describing the evolution of a small number of resolved variables. In this paper we present its application to the derivation of generalized Langevin equations and generalized non-Markovian Fokker-Planck equations. We show how long time scales rates and metastable basins can be extracted from these equations. Numerical algorithms are proposed to discretize these equations. An important aspect is the numerical solution of the orthogonal dynamics equation which is a partial differential equation in a high dimensional space. We propose efficient numerical methods to solve this orthogonal dynamics equation. In addition, we present a projection formalism of the Mori-Zwanzig type that is applicable to discrete maps. Numerical applications are presented from the field of Hamiltonian systems., A new approach for the acquisition of static, wideline (14)N NMR powder patterns is outlined. The method involves the use of frequency-swept pulses which serve two simultaneous functions: (1) broad-band excitation of magnetization and (2) signal enhancement via population transfer. The signal enhancement mechanism is described using numerical simulations and confirmed experimentally. This approach, which we call DEISM (Direct Enhancement of Integer Spin Magnetization), allows high-quality (14)N spectra to be acquired at intermediate field strengths in an uncomplicated way and in a fraction of the time required for previously reported methods.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "2948  (A cavernous hemangioma of the cecum is a rare vascular malformation but is clinically important because of the possibility of massive bleeding. We report a case of a large cavernous hemangioma with pericolic infiltration in the cecum which was removed successfully using minimally invasive surgery., Hypertension, a major cardiovascular risk factor and cause of mortality worldwide, is thought to arise from primary renal abnormalities. However, the etiology of most cases of hypertension remains unexplained. Vascular tone, an important determinant of blood pressure, is regulated by nitric oxide, which causes vascular relaxation by increasing intracellular cGMP and activating cGMP-dependent protein kinase I (PKGI). Here we show that mice with a selective mutation in the N-terminal protein interaction domain of PKGIalpha display inherited vascular smooth muscle cell abnormalities of contraction, abnormal relaxation of large and resistance blood vessels, and increased systemic blood pressure. Renal function studies and responses to changes in dietary sodium in the PKGIalpha mutant mice are normal. These data reveal that PKGIalpha is required for normal VSMC physiology and support the idea that high blood pressure can arise from a primary abnormality of vascular smooth muscle cell contractile regulation, suggesting a new approach to the diagnosis and therapy of hypertension and cardiovascular diseases.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
       "1036  (BACKGROUND: Lévy flights are random walks, the step lengths of which come from probability distributions with heavy power-law tails, such that clusters of short steps are connected by rare long steps. Lévy walks maximise search efficiency of mobile foragers. Recently, several studies raised some concerns about the reliability of the statistical analysis used in previous analyses. Further, it is unclear whether Lévy walks represent adaptive strategies or emergent properties determined by the interaction between foragers and resource distribution. Thus two fundamental questions still need to be addressed: the presence of Lévy walks in the wild and whether or not they represent a form of adaptive behaviour. METHODOLOGY/PRINCIPAL FINDINGS: We studied 235 paths of solitary and clustered (i.e. foraging in group) fallow deer (Dama dama), exploiting the same pasture. We used maximum likelihood estimation for discriminating between a power-tailed distribution and the exponential alternative and rank/frequency plots to discriminate between Lévy walks and composite Brownian walks. We showed that solitary deer perform Lévy searches, while clustered animals did not adopt that strategy. CONCLUSION/SIGNIFICANCE: Our demonstration of the presence of Lévy walks is, at our knowledge, the first available which adopts up-to-date statistical methodologies in a terrestrial mammal. Comparing solitary and clustered deer, we concluded that the Lévy walks of solitary deer represent an adaptation maximising encounter rates with forage resources and not an epiphenomenon induced by a peculiar food distribution., BACKGROUND: A better understanding of the size and abundance of open reading frames (ORFS) in whole genomes may shed light on the factors that control genome complexity. Here we examine the statistical distributions of open reading frames (i.e. distribution of start and stop codons) in the fully sequenced genomes of 297 prokaryotes, and 14 eukaryotes. METHODOLOGY/PRINCIPAL FINDINGS: By fitting mixture models to data from whole genome sequences we show that the size-frequency distributions for ORFS are strikingly similar across prokaryotic and eukaryotic genomes. Moreover, we show that i) a large fraction (60-80%) of ORF size-frequency distributions can be predicted a priori with a stochastic assembly model based on GC content, and that (ii) size-frequency distributions of the remaining &quot;non-random&quot; ORFs are well-fitted by log-normal or gamma distributions, and similar to the size distributions of annotated proteins. CONCLUSIONS/SIGNIFICANCE: Our findings suggest stochastic processes have played a primary role in the evolution of genome complexity, and that common processes govern the conservation and loss of functional genomics units in both prokaryotes and eukaryotes.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc3_act.sort_values(by=[\"Unigram\"], ascending=True)[[\"Unigram\", \"Unigram_detail\"]].head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5630</th>\n",
       "      <td>75.011317</td>\n",
       "      <td>(The development of T helper (T(H))17 and regulatory T (T(reg)) cells is reciprocally regulated by cytokines. Transforming growth factor (TGF)-beta alone induces FoxP3(+) T(reg) cells, but together with IL-6 or IL-21 induces T(H)17 cells. Here we demonstrate that IL-9 is a key molecule that affects differentiation of T(H)17 cells and T(reg) function. IL-9 predominantly produced by T(H)17 cells, synergizes with TGF-beta1 to differentiate naïve CD4(+) T cells into T(H)17 cells, while IL-9 secretion by T(H)17 cells is regulated by IL-23. Interestingly, IL-9 enhances the suppressive functions of FoxP3(+) CD4(+) T(reg) cells in vitro, and absence of IL-9 signaling weakens the suppressive activity of nT(regs) in vivo, leading to an increase in effector cells and worsening of experimental autoimmune encephalomyelitis. The mechanism of IL-9 effects on T(H)17 and T(regs) is through activation of STAT3 and STAT5 signaling. Our findings highlight a role of IL-9 as a regulator of pathogenic versus protective mechanisms of immune responses., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3873</th>\n",
       "      <td>74.535183</td>\n",
       "      <td>(CD1d-restricted invariant NKT (iNKT) cells play crucial roles in various types of immune responses, including autoimmune diseases, infectious diseases and tumor surveillance. The mechanisms underlying their adjuvant functions are well understood. Nevertheless, although IL-4 and IL-10 production characterize iNKT cells able to prevent or ameliorate some autoimmune diseases and inflammatory conditions, the precise mechanisms by which iNKT cells exert immune regulatory function remain elusive. This study demonstrates that the activation of human iNKT cells by their specific ligand alpha-galactosylceramide enhances IL-12p70 while inhibiting the IL-23 production by monocyte-derived dendritic cells, and in turn down-regulating the IL-17 production by memory CD4(+) Th cells. The ability of the iNKT cells to regulate the differential production of IL-12p70/IL-23 is mainly mediated by a remarkable hallmark of their function to produce both Th1 and Th2 cytokines. In particular, the down-regulation of IL-23 is markedly associated with a production of IL-4 and IL-10 from iNKT cells. Moreover, Th2 cytokines, such as IL-4 and IL-13 play a crucial role in defining the biased production of IL-12p70/IL-23 by enhancement of IL-12p70 in synergy with IFN-gamma, whereas inhibition of the IFN-gamma-promoted IL-23 production. Collectively, the results suggest that iNKT cells modify the IL-12p70/IL-23 balance to enhance the IL-12p70-induced cell-mediated immunity and suppress the IL-23-dependent inflammatory pathologies. These results may account for the long-appreciated contrasting beneficial and adverse consequence of ligand activation of iNKT cells., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2623</th>\n",
       "      <td>74.011540</td>\n",
       "      <td>(IL (interleukin)-4 and IL-13 are key cytokines in the pathogenesis of allergic inflammatory disease. IL-4 and IL-13 share many functional properties as a result of their utilization of a common receptor complex comprising IL-13Ralpha1 (IL-13 receptor alpha-chain 1) and IL-4Ralpha. The second IL-13R (IL-13 receptor) has been identified, namely IL-13Ralpha2. This has been thought to be a decoy receptor due to its short cytoplasmic tail and its high binding affinity for IL-13 but not IL-4. IL-13Ralpha2 exists on the cell membrane, intracellularly and in a soluble form. Recent reports revealed that membrane IL-13Ralpha2 may have some signalling capabilities, and a soluble form of IL-13Ralpha2 can be generated in the presence of environmental allergens such as DerP. Interestingly, IL-13Ralpha2 has also been shown to regulate both IL-13 and IL-4 response in primary airway cells, despite the fact that IL-13Ralpha2 does not bind IL-4. The regulator mechanism is still unclear but the physical association of IL-13Ralpha2 with IL-4Ralpha appears to be a key regulatory step. These results suggest that the cytoplasmic tail of IL-13Ralpha2 may interfere with the association or activation of signalling molecules, such as JAK1 (Janus kinase 1), on IL-4Ralpha and thus prevents downstream signal cascade. The receptor has more complicated functions than a simple decoy receptor. In this review, we discuss newly revealed functions of IL-13Ralpha2., Lack of the IL-1 receptor accessory protein (IL-1RAcP) abrogates responses to IL-33 and IL-1 in the mouse thymoma clone EL-4 D6/76 cells. Reconstitution with full-length IL-1RAcP is sufficient to restore responsiveness to IL-33 and IL-1. IL-33 activates IL-1 receptor-associated kinase-1, cJun-N-terminal kinase, and the NF-kappaB pathway in an IL-1RAcP-dependent manner and results in IL-2 release. IL-33 is able to induce the release of proinflammatory cytokines in bone marrow-derived (BMD) mast cells, indicating that IL-33 may have a proinflammatory potential like its relatives IL-1 and IL-18, in addition to its Th2-skewing properties in the adaptive response described previously. Blocking of murine IL-1RAcP with the neutralizing antibody 4C5 inhibits response of mouse thymoma cells and BMD mast cells to IL-33. The interaction of either membrane-bound or soluble forms of IL-1RAcP and IL-33Ralpha-chain depends on the presence of IL-33, as demonstrated by coimmunoprecipitation assays. These data demonstrate that IL-1RAcP is indispensable for IL-33 signaling. Furthermore, they suggest that IL-1RAcP is used by more than one alpha-chain of the IL-1 receptor family and thus may resemble a common beta-chain of that family.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4000</th>\n",
       "      <td>71.268446</td>\n",
       "      <td>(MC2 (ACTH) receptors require MC2 receptor accessory protein (MRAP) to reach the cell surface. In this study, we show that MRAP has the opposite effect on the closely related MC5 receptor. In enzyme-linked immunosorbent assay and microscopy experiments, MC2 receptor was retained in the endoplasmic reticulum in the absence of MRAP and targeted to the plasma membrane with MRAP. MC5 receptor was at the plasma membrane in the absence of MRAP, but trapped intracellularly when expressed with MRAP. Using bimolecular fluorescence complementation, where one fragment of yellow fluorescent protein (YFP) was fused to receptors and another to MRAP, we showed that MC2 receptor-MRAP dimers were present at the plasma membrane, whereas MC5 receptor-MRAP dimers were intracellular. Both MC2 and MC5 receptors co-precipitated with MRAP. MRAP did not alter expression of beta2-adrenergic receptors or co-precipitate with them. To determine if MRAP affects formation of receptor oligomers, we co-expressed MC2 receptors fused to YFP fragments in the presence or absence of MRAP. YFP fluorescence, reporting MC2 receptor homodimers, was readily detectable with or without MRAP. In contrast, MC5 receptor homodimers were visible in the absence of MRAP, but little fluorescence was observed by microscopic analysis when MRAP was co-expressed. Co-precipitation of differentially tagged receptors confirmed that MRAP blocks MC5 receptor dimerization. The regions of MRAP required for its effects on MC2 and MC5 receptors differed. These results establish that MRAP forms stable complexes with two different melanocortin receptors, facilitating surface expression of MC2 receptor but disrupting dimerization and surface localization of MC5 receptor., The melanocortin-2 (MC2) receptor accessory protein (MRAP) is required for trafficking of the G protein-coupled MC2 receptor to the plasma membrane. The mechanism of action and structure of MRAP, which has a single transmembrane domain, are unknown. Here, we show that MRAP displays a previously uncharacterized topology. Epitopes on both the N- and C-terminal ends of MRAP were localized on the external face of CHO cells at comparable levels. Using antibodies raised against N- and C-terminal MRAP peptides, we demonstrated that both ends of endogenous MRAP face the outside in adrenal cells. Nearly half of MRAP was glycosylated at the single endogenous N-terminal glycosylation site, and over half was glycosylated when the natural glycosylation site was replaced by one in the C-terminal domain. A mutant MRAP with potential glycosylation sites on both sides of the membrane was singly but not doubly glycosylated, suggesting that MRAP is not monotopic. Coimmunoprecipitation of differentially tagged MRAPs established that MRAP is a dimer. By selectively immunoprecipitating cell surface MRAP in one or the other orientation, we showed that MRAP homodimers are antiparallel and form a stable complex with MC2 receptor. In the absence of MRAP, MC2 receptor was trapped in the endoplasmic reticulum, but with MRAP, the MC2 receptor was glycosylated and localized on the plasma membrane, where it signaled in response to ACTH. MRAP acted specifically, because it did not increase surface expression of other melanocortin, beta2-adrenergic, or TSH-releasing hormone receptors. MRAP is the first eukaryotic membrane protein identified with an antiparallel homodimeric structure.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4370</th>\n",
       "      <td>69.897546</td>\n",
       "      <td>(Using biopsy specimens from patients with B-cell non-Hodgkin's lymphoma, we observed a significantly low frequency of T(H)17 cells, including several samples with no detectable amount of interleukin (IL)-17-producing cells present in the tumor microenvironment. We found that, in the absence of lymphoma B cells, treatment with IL-1beta/IL-6 or lipopolysaccharide (LPS) enhanced IL-17 expression in CD4(+) T cells and this enhancement was attenuated when CD4(+) T cells were cocultured with lymphoma B cells. Blockade of CD27-CD70 or CD28-CD80/86 interactions by anti-CD70 or anti-CD80/86 antibodies restored LPS-mediated induction of IL-17 expression in CD4(+) T cells cocultured with lymphoma B cells. Because a subset of lymphoma B cells express IL-2 and given that IL-2 signaling is critically important in the development of regulatory T (T(reg)) cells, we tested the role of IL-2 signaling in T(H)17 cell development. We found that treatment with anti-IL-2 antibody to interrupt IL-2 signaling significantly inhibited Foxp3 expression in CD4(+) T cells. In contrast, interruption of IL-2 signaling up-regulated IL-17 expression in CD4(+) T cells and restored lymphoma-mediated down-regulation of IL-17-producing cells. Furthermore, the reversal of T(reg) cell activity by LPS or CpG-A resulted in an enhancement of IL-17-producing cells. Taken together, our study indicated that lymphoma B cells play an important role in skewing the balance between T(reg) and T(H)17 cells resulting in the establishment of a profoundly inhibitory tumor microenvironment., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5392</th>\n",
       "      <td>68.417062</td>\n",
       "      <td>(Interleukin 23 (IL-23) and IL-17 have been linked to the pathogenesis of several chronic inflammatory disorders, including inflammatory bowel disease. Yet as an important function for IL-23 is emerging, the function of IL-17 in inflammatory bowel disease remains unclear. Here we demonstrate IL-17A-mediated protection in the CD45RBhi transfer model of colitis. An accelerated wasting disease elicited by T cells deficient in IL-17A correlated with higher expression of genes encoding T helper type 1-type cytokines in colon tissue. IL-17A also modulated T helper type 1 polarization in vitro. Furthermore, T cells deficient in the IL-17 receptor elicited an accelerated, aggressive wasting disease relative to that elicited by wild-type T cells in recipient mice. Our data demonstrate a protective function for IL-17 and identify T cells as not only the source but also a target of IL-17 in vivo., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1683</th>\n",
       "      <td>66.717460</td>\n",
       "      <td>(Lymphocyte recruitment and activation have been implicated in the progression of cerebral ischemia-reperfusion (I/R) injury, but the roles of specific lymphocyte subpopulations and cytokines during stroke remain to be clarified. Here we demonstrate that the infiltration of T cells into the brain, as well as the cytokines interleukin-23 (IL-23) and IL-17, have pivotal roles in the evolution of brain infarction and accompanying neurological deficits. Blockade of T cell infiltration into the brain by the immunosuppressant FTY720 reduced I/R-induced brain damage. The expression of IL-23, which was derived mostly from infiltrated macrophages, increased on day 1 after I/R, whereas IL-17 levels were elevated after day 3, and this induction of IL-17 was dependent on IL-23. These data, together with analysis of mice genetically disrupted for IL-17 and IL-23, suggest that IL-23 functions in the immediate stage of I/R brain injury, whereas IL-17 has an important role in the delayed phase of I/R injury during which apoptotic neuronal death occurs in the penumbra. Intracellular cytokine staining revealed that gammadeltaT lymphocytes, but not CD4(+) helper T cells, were a major source of IL-17. Moreover, depletion of gammadeltaT lymphocytes ameliorated the I/R injury. We propose that T lymphocytes, including gammadeltaT lymphocytes, could be a therapeutic target for mitigating the inflammatory events that amplify the initial damage in cerebral ischemia., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2997</th>\n",
       "      <td>66.225136</td>\n",
       "      <td>(We report that like other T cells cultured in the presence of transforming growth factor (TGF) beta, Th17 cells also produce interleukin (IL) 9. Th17 cells generated in vitro with IL-6 and TGF-beta as well as purified ex vivo Th17 cells both produced IL-9. To determine if IL-9 has functional consequences in Th17-mediated inflammatory disease, we evaluated the role of IL-9 in the development and progression of experimental autoimmune encephalomyelitis, a mouse model of multiple sclerosis. The data show that IL-9 neutralization and IL-9 receptor deficiency attenuates disease, and this correlates with decreases in Th17 cells and IL-6-producing macrophages in the central nervous system, as well as mast cell numbers in the regional lymph nodes. Collectively, these data implicate IL-9 as a Th17-derived cytokine that can contribute to inflammatory disease., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128</th>\n",
       "      <td>65.559946</td>\n",
       "      <td>(B-cell survival depends on signals induced by B-cell activating factor (BAFF) binding to its receptor (BAFF-R). In mice, mutations in BAFF or BAFF-R cause B-cell lymphopenia and antibody deficiency. Analyzing BAFF-R expression and BAFF-binding to B cells in common variable immunodeficiency (CVID) patients, we identified two siblings carrying a homozygous deletion in the BAFF-R gene. Removing most of the BAFF-R transmembrane part, the deletion precludes BAFF-R expression. Without BAFF-R, B-cell development is arrested at the stage of transitional B cells and the numbers of all subsequent B-cell stages are severely reduced. Both siblings have lower IgG and IgM serum levels but, unlike most CVID patients, normal IgA concentrations. They also did not mount a T-independent immune response against pneumococcal cell wall polysaccharides but only one BAFF-R-deficient sibling developed recurrent infections. Therefore, deletion of the BAFF-R gene in humans causes a characteristic immunological phenotype but it does not necessarily lead to a clinically manifest immunodeficiency., Defects in the expression of either BAFF (B cell activating factor) or BAFF-R impairs B cell development beyond the immature, transitional type-1 stage and thus, prevents the formation of follicular and marginal zone B cells, whereas B-1 B cells remain unaffected. The expression of BAFF-R on all mature B cells might suggest a role for BAFF-R signaling also for their in vivo maintenance. Here, we show that, 14 days following a single injection of an anti-BAFF-R mAb that prevents BAFF binding, both follicular and marginal zone B cell numbers are drastically reduced, whereas B-1 cells are not affected. Injection of control, isotype-matched but non-blocking anti-BAFF-R mAbs does not result in B cell depletion. We also show that this depletion is neither due to antibody-dependent cellular cytotoxicity nor to complement-mediated lysis. Moreover, prevention of BAFF binding leads to a decrease in the size of the B cell follicles, an impairment of a T cell dependent humoral immune response and a reduction in the formation of memory B cells. Collectively, these results establish a central role for BAFF-BAFF-R signaling in the in vivo survival and maintenance of both follicular and marginal zone B cell pools.)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3153</th>\n",
       "      <td>63.458674</td>\n",
       "      <td>(IL-1 is a potent cytokine that can induce bone erosion in inflammatory sites such as rheumatoid joint regions via activation of osteoclasts. Not only is IL-1 capable of activating osteoclasts, but it is also a key cytokine involved in the differentiation, multinucleation, and survival of osteoclasts. Herein, we show that IL-1 has the potential to drive osteoclast differentiation via a receptor activator of NF-kappaB ligand (RANKL)/RANK-independent mechanism. Although IL-1 has a synergistic effect on RANKL-induced osteoclast formation, IL-1 alone cannot induce osteoclast differentiation from osteoclast precursors (bone marrow-derived macrophages (BMMs)) due to a lack of IL-1 signaling potential in these cells. However, we demonstrate that overexpression of the IL-1RI receptor in BMMs or induction of IL-1RI by c-Fos overexpression enables IL-1 alone to induce the formation of authentic osteoclasts by a RANKL/RANK-independent mechanism. The expression of IL-1RI is up-regulated by RANKL via c-Fos and NFATc1. Furthermore, the addition of IL-1 to IL-1RI overexpressing BMMs (IL-1/IL-1RI) strongly activates NF-kappaB, JNK, p38, and ERK which is a hallmark gene activation profile of osteoclastogenesis. Interestingly, IL-1/IL-1RI does not induce expression of c-Fos or NFATc1 during osteoclast differentiation, although basal levels of c-Fos and NFATc1 seem to be required. Rather, IL-1/IL-1RI strongly activates MITF, which subsequently induces osteoclast-specific genes such as osteoclast-associated receptor and tartrate-resistant acid phosphatase. Together, these results reveal that IL-1 has the potential to induce osteoclast differentiation via activation of microphthalmia transcription factor under specific microenvironmental conditions., Lack of the IL-1 receptor accessory protein (IL-1RAcP) abrogates responses to IL-33 and IL-1 in the mouse thymoma clone EL-4 D6/76 cells. Reconstitution with full-length IL-1RAcP is sufficient to restore responsiveness to IL-33 and IL-1. IL-33 activates IL-1 receptor-associated kinase-1, cJun-N-terminal kinase, and the NF-kappaB pathway in an IL-1RAcP-dependent manner and results in IL-2 release. IL-33 is able to induce the release of proinflammatory cytokines in bone marrow-derived (BMD) mast cells, indicating that IL-33 may have a proinflammatory potential like its relatives IL-1 and IL-18, in addition to its Th2-skewing properties in the adaptive response described previously. Blocking of murine IL-1RAcP with the neutralizing antibody 4C5 inhibits response of mouse thymoma cells and BMD mast cells to IL-33. The interaction of either membrane-bound or soluble forms of IL-1RAcP and IL-33Ralpha-chain depends on the presence of IL-33, as demonstrated by coimmunoprecipitation assays. These data demonstrate that IL-1RAcP is indispensable for IL-33 signaling. Furthermore, they suggest that IL-1RAcP is used by more than one alpha-chain of the IL-1 receptor family and thus may resemble a common beta-chain of that family.)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unigram  \\\n",
       "5630  75.011317   \n",
       "3873  74.535183   \n",
       "2623  74.011540   \n",
       "4000  71.268446   \n",
       "4370  69.897546   \n",
       "5392  68.417062   \n",
       "1683  66.717460   \n",
       "2997  66.225136   \n",
       "1128  65.559946   \n",
       "3153  63.458674   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Unigram_detail  \n",
       "5630  (The development of T helper (T(H))17 and regulatory T (T(reg)) cells is reciprocally regulated by cytokines. Transforming growth factor (TGF)-beta alone induces FoxP3(+) T(reg) cells, but together with IL-6 or IL-21 induces T(H)17 cells. Here we demonstrate that IL-9 is a key molecule that affects differentiation of T(H)17 cells and T(reg) function. IL-9 predominantly produced by T(H)17 cells, synergizes with TGF-beta1 to differentiate naïve CD4(+) T cells into T(H)17 cells, while IL-9 secretion by T(H)17 cells is regulated by IL-23. Interestingly, IL-9 enhances the suppressive functions of FoxP3(+) CD4(+) T(reg) cells in vitro, and absence of IL-9 signaling weakens the suppressive activity of nT(regs) in vivo, leading to an increase in effector cells and worsening of experimental autoimmune encephalomyelitis. The mechanism of IL-9 effects on T(H)17 and T(regs) is through activation of STAT3 and STAT5 signaling. Our findings highlight a role of IL-9 as a regulator of pathogenic versus protective mechanisms of immune responses., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "3873  (CD1d-restricted invariant NKT (iNKT) cells play crucial roles in various types of immune responses, including autoimmune diseases, infectious diseases and tumor surveillance. The mechanisms underlying their adjuvant functions are well understood. Nevertheless, although IL-4 and IL-10 production characterize iNKT cells able to prevent or ameliorate some autoimmune diseases and inflammatory conditions, the precise mechanisms by which iNKT cells exert immune regulatory function remain elusive. This study demonstrates that the activation of human iNKT cells by their specific ligand alpha-galactosylceramide enhances IL-12p70 while inhibiting the IL-23 production by monocyte-derived dendritic cells, and in turn down-regulating the IL-17 production by memory CD4(+) Th cells. The ability of the iNKT cells to regulate the differential production of IL-12p70/IL-23 is mainly mediated by a remarkable hallmark of their function to produce both Th1 and Th2 cytokines. In particular, the down-regulation of IL-23 is markedly associated with a production of IL-4 and IL-10 from iNKT cells. Moreover, Th2 cytokines, such as IL-4 and IL-13 play a crucial role in defining the biased production of IL-12p70/IL-23 by enhancement of IL-12p70 in synergy with IFN-gamma, whereas inhibition of the IFN-gamma-promoted IL-23 production. Collectively, the results suggest that iNKT cells modify the IL-12p70/IL-23 balance to enhance the IL-12p70-induced cell-mediated immunity and suppress the IL-23-dependent inflammatory pathologies. These results may account for the long-appreciated contrasting beneficial and adverse consequence of ligand activation of iNKT cells., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
       "2623  (IL (interleukin)-4 and IL-13 are key cytokines in the pathogenesis of allergic inflammatory disease. IL-4 and IL-13 share many functional properties as a result of their utilization of a common receptor complex comprising IL-13Ralpha1 (IL-13 receptor alpha-chain 1) and IL-4Ralpha. The second IL-13R (IL-13 receptor) has been identified, namely IL-13Ralpha2. This has been thought to be a decoy receptor due to its short cytoplasmic tail and its high binding affinity for IL-13 but not IL-4. IL-13Ralpha2 exists on the cell membrane, intracellularly and in a soluble form. Recent reports revealed that membrane IL-13Ralpha2 may have some signalling capabilities, and a soluble form of IL-13Ralpha2 can be generated in the presence of environmental allergens such as DerP. Interestingly, IL-13Ralpha2 has also been shown to regulate both IL-13 and IL-4 response in primary airway cells, despite the fact that IL-13Ralpha2 does not bind IL-4. The regulator mechanism is still unclear but the physical association of IL-13Ralpha2 with IL-4Ralpha appears to be a key regulatory step. These results suggest that the cytoplasmic tail of IL-13Ralpha2 may interfere with the association or activation of signalling molecules, such as JAK1 (Janus kinase 1), on IL-4Ralpha and thus prevents downstream signal cascade. The receptor has more complicated functions than a simple decoy receptor. In this review, we discuss newly revealed functions of IL-13Ralpha2., Lack of the IL-1 receptor accessory protein (IL-1RAcP) abrogates responses to IL-33 and IL-1 in the mouse thymoma clone EL-4 D6/76 cells. Reconstitution with full-length IL-1RAcP is sufficient to restore responsiveness to IL-33 and IL-1. IL-33 activates IL-1 receptor-associated kinase-1, cJun-N-terminal kinase, and the NF-kappaB pathway in an IL-1RAcP-dependent manner and results in IL-2 release. IL-33 is able to induce the release of proinflammatory cytokines in bone marrow-derived (BMD) mast cells, indicating that IL-33 may have a proinflammatory potential like its relatives IL-1 and IL-18, in addition to its Th2-skewing properties in the adaptive response described previously. Blocking of murine IL-1RAcP with the neutralizing antibody 4C5 inhibits response of mouse thymoma cells and BMD mast cells to IL-33. The interaction of either membrane-bound or soluble forms of IL-1RAcP and IL-33Ralpha-chain depends on the presence of IL-33, as demonstrated by coimmunoprecipitation assays. These data demonstrate that IL-1RAcP is indispensable for IL-33 signaling. Furthermore, they suggest that IL-1RAcP is used by more than one alpha-chain of the IL-1 receptor family and thus may resemble a common beta-chain of that family.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "4000  (MC2 (ACTH) receptors require MC2 receptor accessory protein (MRAP) to reach the cell surface. In this study, we show that MRAP has the opposite effect on the closely related MC5 receptor. In enzyme-linked immunosorbent assay and microscopy experiments, MC2 receptor was retained in the endoplasmic reticulum in the absence of MRAP and targeted to the plasma membrane with MRAP. MC5 receptor was at the plasma membrane in the absence of MRAP, but trapped intracellularly when expressed with MRAP. Using bimolecular fluorescence complementation, where one fragment of yellow fluorescent protein (YFP) was fused to receptors and another to MRAP, we showed that MC2 receptor-MRAP dimers were present at the plasma membrane, whereas MC5 receptor-MRAP dimers were intracellular. Both MC2 and MC5 receptors co-precipitated with MRAP. MRAP did not alter expression of beta2-adrenergic receptors or co-precipitate with them. To determine if MRAP affects formation of receptor oligomers, we co-expressed MC2 receptors fused to YFP fragments in the presence or absence of MRAP. YFP fluorescence, reporting MC2 receptor homodimers, was readily detectable with or without MRAP. In contrast, MC5 receptor homodimers were visible in the absence of MRAP, but little fluorescence was observed by microscopic analysis when MRAP was co-expressed. Co-precipitation of differentially tagged receptors confirmed that MRAP blocks MC5 receptor dimerization. The regions of MRAP required for its effects on MC2 and MC5 receptors differed. These results establish that MRAP forms stable complexes with two different melanocortin receptors, facilitating surface expression of MC2 receptor but disrupting dimerization and surface localization of MC5 receptor., The melanocortin-2 (MC2) receptor accessory protein (MRAP) is required for trafficking of the G protein-coupled MC2 receptor to the plasma membrane. The mechanism of action and structure of MRAP, which has a single transmembrane domain, are unknown. Here, we show that MRAP displays a previously uncharacterized topology. Epitopes on both the N- and C-terminal ends of MRAP were localized on the external face of CHO cells at comparable levels. Using antibodies raised against N- and C-terminal MRAP peptides, we demonstrated that both ends of endogenous MRAP face the outside in adrenal cells. Nearly half of MRAP was glycosylated at the single endogenous N-terminal glycosylation site, and over half was glycosylated when the natural glycosylation site was replaced by one in the C-terminal domain. A mutant MRAP with potential glycosylation sites on both sides of the membrane was singly but not doubly glycosylated, suggesting that MRAP is not monotopic. Coimmunoprecipitation of differentially tagged MRAPs established that MRAP is a dimer. By selectively immunoprecipitating cell surface MRAP in one or the other orientation, we showed that MRAP homodimers are antiparallel and form a stable complex with MC2 receptor. In the absence of MRAP, MC2 receptor was trapped in the endoplasmic reticulum, but with MRAP, the MC2 receptor was glycosylated and localized on the plasma membrane, where it signaled in response to ACTH. MRAP acted specifically, because it did not increase surface expression of other melanocortin, beta2-adrenergic, or TSH-releasing hormone receptors. MRAP is the first eukaryotic membrane protein identified with an antiparallel homodimeric structure.)  \n",
       "4370  (Using biopsy specimens from patients with B-cell non-Hodgkin's lymphoma, we observed a significantly low frequency of T(H)17 cells, including several samples with no detectable amount of interleukin (IL)-17-producing cells present in the tumor microenvironment. We found that, in the absence of lymphoma B cells, treatment with IL-1beta/IL-6 or lipopolysaccharide (LPS) enhanced IL-17 expression in CD4(+) T cells and this enhancement was attenuated when CD4(+) T cells were cocultured with lymphoma B cells. Blockade of CD27-CD70 or CD28-CD80/86 interactions by anti-CD70 or anti-CD80/86 antibodies restored LPS-mediated induction of IL-17 expression in CD4(+) T cells cocultured with lymphoma B cells. Because a subset of lymphoma B cells express IL-2 and given that IL-2 signaling is critically important in the development of regulatory T (T(reg)) cells, we tested the role of IL-2 signaling in T(H)17 cell development. We found that treatment with anti-IL-2 antibody to interrupt IL-2 signaling significantly inhibited Foxp3 expression in CD4(+) T cells. In contrast, interruption of IL-2 signaling up-regulated IL-17 expression in CD4(+) T cells and restored lymphoma-mediated down-regulation of IL-17-producing cells. Furthermore, the reversal of T(reg) cell activity by LPS or CpG-A resulted in an enhancement of IL-17-producing cells. Taken together, our study indicated that lymphoma B cells play an important role in skewing the balance between T(reg) and T(H)17 cells resulting in the establishment of a profoundly inhibitory tumor microenvironment., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "5392  (Interleukin 23 (IL-23) and IL-17 have been linked to the pathogenesis of several chronic inflammatory disorders, including inflammatory bowel disease. Yet as an important function for IL-23 is emerging, the function of IL-17 in inflammatory bowel disease remains unclear. Here we demonstrate IL-17A-mediated protection in the CD45RBhi transfer model of colitis. An accelerated wasting disease elicited by T cells deficient in IL-17A correlated with higher expression of genes encoding T helper type 1-type cytokines in colon tissue. IL-17A also modulated T helper type 1 polarization in vitro. Furthermore, T cells deficient in the IL-17 receptor elicited an accelerated, aggressive wasting disease relative to that elicited by wild-type T cells in recipient mice. Our data demonstrate a protective function for IL-17 and identify T cells as not only the source but also a target of IL-17 in vivo., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "1683  (Lymphocyte recruitment and activation have been implicated in the progression of cerebral ischemia-reperfusion (I/R) injury, but the roles of specific lymphocyte subpopulations and cytokines during stroke remain to be clarified. Here we demonstrate that the infiltration of T cells into the brain, as well as the cytokines interleukin-23 (IL-23) and IL-17, have pivotal roles in the evolution of brain infarction and accompanying neurological deficits. Blockade of T cell infiltration into the brain by the immunosuppressant FTY720 reduced I/R-induced brain damage. The expression of IL-23, which was derived mostly from infiltrated macrophages, increased on day 1 after I/R, whereas IL-17 levels were elevated after day 3, and this induction of IL-17 was dependent on IL-23. These data, together with analysis of mice genetically disrupted for IL-17 and IL-23, suggest that IL-23 functions in the immediate stage of I/R brain injury, whereas IL-17 has an important role in the delayed phase of I/R injury during which apoptotic neuronal death occurs in the penumbra. Intracellular cytokine staining revealed that gammadeltaT lymphocytes, but not CD4(+) helper T cells, were a major source of IL-17. Moreover, depletion of gammadeltaT lymphocytes ameliorated the I/R injury. We propose that T lymphocytes, including gammadeltaT lymphocytes, could be a therapeutic target for mitigating the inflammatory events that amplify the initial damage in cerebral ischemia., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
       "2997  (We report that like other T cells cultured in the presence of transforming growth factor (TGF) beta, Th17 cells also produce interleukin (IL) 9. Th17 cells generated in vitro with IL-6 and TGF-beta as well as purified ex vivo Th17 cells both produced IL-9. To determine if IL-9 has functional consequences in Th17-mediated inflammatory disease, we evaluated the role of IL-9 in the development and progression of experimental autoimmune encephalomyelitis, a mouse model of multiple sclerosis. The data show that IL-9 neutralization and IL-9 receptor deficiency attenuates disease, and this correlates with decreases in Th17 cells and IL-6-producing macrophages in the central nervous system, as well as mast cell numbers in the regional lymph nodes. Collectively, these data implicate IL-9 as a Th17-derived cytokine that can contribute to inflammatory disease., Th17 cells, CD4(+) T cells that secrete interleukin-17 (IL-17), are pathogenic in autoimmune diseases and their development and expansion is driven by the cytokines IL-6, TGF-beta, IL-21, IL-1, and IL-23. However, there are also innate sources of IL-17. Here, we show that gammadelta T cells express IL-23R and the transcription factor RORgammat and produce IL-17, IL-21, and IL-22 in response to IL-1beta and IL-23, without T cell receptor engagement. IL-17-producing gammadelta T cells were found at high frequency in the brain of mice with experimental autoimmune encephalomyelitis (EAE). gammadelta T cells activated by IL-1beta and IL-23 promoted IL-17 production by CD4(+) T cells and increased susceptibility to EAE, suggesting that gammadelta T cells act in an amplification loop for IL-17 production by Th17 cells. Our findings demonstrate that gammadelta T cells activated by IL-1beta and IL-23 are an important source of innate IL-17 and IL-21 and provide an alternative mechanism whereby IL-1 and IL-23 may mediate autoimmune inflammation.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "1128  (B-cell survival depends on signals induced by B-cell activating factor (BAFF) binding to its receptor (BAFF-R). In mice, mutations in BAFF or BAFF-R cause B-cell lymphopenia and antibody deficiency. Analyzing BAFF-R expression and BAFF-binding to B cells in common variable immunodeficiency (CVID) patients, we identified two siblings carrying a homozygous deletion in the BAFF-R gene. Removing most of the BAFF-R transmembrane part, the deletion precludes BAFF-R expression. Without BAFF-R, B-cell development is arrested at the stage of transitional B cells and the numbers of all subsequent B-cell stages are severely reduced. Both siblings have lower IgG and IgM serum levels but, unlike most CVID patients, normal IgA concentrations. They also did not mount a T-independent immune response against pneumococcal cell wall polysaccharides but only one BAFF-R-deficient sibling developed recurrent infections. Therefore, deletion of the BAFF-R gene in humans causes a characteristic immunological phenotype but it does not necessarily lead to a clinically manifest immunodeficiency., Defects in the expression of either BAFF (B cell activating factor) or BAFF-R impairs B cell development beyond the immature, transitional type-1 stage and thus, prevents the formation of follicular and marginal zone B cells, whereas B-1 B cells remain unaffected. The expression of BAFF-R on all mature B cells might suggest a role for BAFF-R signaling also for their in vivo maintenance. Here, we show that, 14 days following a single injection of an anti-BAFF-R mAb that prevents BAFF binding, both follicular and marginal zone B cell numbers are drastically reduced, whereas B-1 cells are not affected. Injection of control, isotype-matched but non-blocking anti-BAFF-R mAbs does not result in B cell depletion. We also show that this depletion is neither due to antibody-dependent cellular cytotoxicity nor to complement-mediated lysis. Moreover, prevention of BAFF binding leads to a decrease in the size of the B cell follicles, an impairment of a T cell dependent humoral immune response and a reduction in the formation of memory B cells. Collectively, these results establish a central role for BAFF-BAFF-R signaling in the in vivo survival and maintenance of both follicular and marginal zone B cell pools.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "3153  (IL-1 is a potent cytokine that can induce bone erosion in inflammatory sites such as rheumatoid joint regions via activation of osteoclasts. Not only is IL-1 capable of activating osteoclasts, but it is also a key cytokine involved in the differentiation, multinucleation, and survival of osteoclasts. Herein, we show that IL-1 has the potential to drive osteoclast differentiation via a receptor activator of NF-kappaB ligand (RANKL)/RANK-independent mechanism. Although IL-1 has a synergistic effect on RANKL-induced osteoclast formation, IL-1 alone cannot induce osteoclast differentiation from osteoclast precursors (bone marrow-derived macrophages (BMMs)) due to a lack of IL-1 signaling potential in these cells. However, we demonstrate that overexpression of the IL-1RI receptor in BMMs or induction of IL-1RI by c-Fos overexpression enables IL-1 alone to induce the formation of authentic osteoclasts by a RANKL/RANK-independent mechanism. The expression of IL-1RI is up-regulated by RANKL via c-Fos and NFATc1. Furthermore, the addition of IL-1 to IL-1RI overexpressing BMMs (IL-1/IL-1RI) strongly activates NF-kappaB, JNK, p38, and ERK which is a hallmark gene activation profile of osteoclastogenesis. Interestingly, IL-1/IL-1RI does not induce expression of c-Fos or NFATc1 during osteoclast differentiation, although basal levels of c-Fos and NFATc1 seem to be required. Rather, IL-1/IL-1RI strongly activates MITF, which subsequently induces osteoclast-specific genes such as osteoclast-associated receptor and tartrate-resistant acid phosphatase. Together, these results reveal that IL-1 has the potential to induce osteoclast differentiation via activation of microphthalmia transcription factor under specific microenvironmental conditions., Lack of the IL-1 receptor accessory protein (IL-1RAcP) abrogates responses to IL-33 and IL-1 in the mouse thymoma clone EL-4 D6/76 cells. Reconstitution with full-length IL-1RAcP is sufficient to restore responsiveness to IL-33 and IL-1. IL-33 activates IL-1 receptor-associated kinase-1, cJun-N-terminal kinase, and the NF-kappaB pathway in an IL-1RAcP-dependent manner and results in IL-2 release. IL-33 is able to induce the release of proinflammatory cytokines in bone marrow-derived (BMD) mast cells, indicating that IL-33 may have a proinflammatory potential like its relatives IL-1 and IL-18, in addition to its Th2-skewing properties in the adaptive response described previously. Blocking of murine IL-1RAcP with the neutralizing antibody 4C5 inhibits response of mouse thymoma cells and BMD mast cells to IL-33. The interaction of either membrane-bound or soluble forms of IL-1RAcP and IL-33Ralpha-chain depends on the presence of IL-33, as demonstrated by coimmunoprecipitation assays. These data demonstrate that IL-1RAcP is indispensable for IL-33 signaling. Furthermore, they suggest that IL-1RAcP is used by more than one alpha-chain of the IL-1 receptor family and thus may resemble a common beta-chain of that family.)                                                                                                                                                                                                                                                                                                                                                                                                                                     "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc3_act.sort_values(by=[\"Unigram\"], ascending=False)[[\"Unigram\", \"Unigram_detail\"]].head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "      <th>Bigram_detail</th>\n",
       "      <th>Trigram_detail</th>\n",
       "      <th>data_set</th>\n",
       "      <th>task_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>24.053512</td>\n",
       "      <td>3.221897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>(A genetic tool to introduce marker-free deletions is essential for multiple manipulations of genomes. We report a simple and efficient method to create marker-free deletion mutants of Bacillus subtilis through transformation with recombinant PCR products, using the Escherichia coli mazF gene encoding an endoribonuclease that cleaves free mRNAs as a counter-selection tool. Our method will be applicable to any bacterium in which introduction of the mazF cassette into the genome by double crossover homologous recombination is possible., The next-generation sequencing technology coupled with the growing number of genome sequences opens the opportunity to redesign genotyping strategies for more effective genetic mapping and genome analysis. We have developed a high-throughput method for genotyping recombinant populations utilizing whole-genome resequencing data generated by the Illumina Genome Analyzer. A sliding window approach is designed to collectively examine genome-wide single nucleotide polymorphisms for genotype calling and recombination breakpoint determination. Using this method, we constructed a genetic map for 150 rice recombinant inbred lines with an expected genotype calling accuracy of 99.94% and a resolution of recombination breakpoints within an average of 40 kb. In comparison to the genetic map constructed with 287 PCR-based markers for the rice population, the sequencing-based method was approximately 20x faster in data collection and 35x more precise in recombination breakpoint determination. Using the sequencing-based genetic map, we located a quantitative trait locus of large effect on plant height in a 100-kb region containing the rice &amp;quot;green revolution&amp;quot; gene. Through computer simulation, we demonstrate that the method is robust for different types of mapping populations derived from organisms with variable quality of genome sequences and is feasible for organisms with large genome sizes and low polymorphisms. With continuous advances in sequencing technologies, this genome-based method may replace the conventional marker-based genotyping approach to provide a powerful tool for large-scale gene discovery and for addressing a wide range of biological questions.)</td>\n",
       "      <td>(A genetic tool to introduce marker-free deletions is essential for multiple manipulations of genomes. We report a simple and efficient method to create marker-free deletion mutants of Bacillus subtilis through transformation with recombinant PCR products, using the Escherichia coli mazF gene encoding an endoribonuclease that cleaves free mRNAs as a counter-selection tool. Our method will be applicable to any bacterium in which introduction of the mazF cassette into the genome by double crossover homologous recombination is possible., Rickettsia prowazekii, the causative agent of epidemic typhus, is an obligately intracytoplasmic bacterium, a lifestyle that imposes significant barriers to genetic manipulation. The key to understanding how this unique bacterium evades host immunity is the mutagenesis of selected genes hypothesized to be involved in virulence. The R. prowazekii pld gene, encoding a protein with phospholipase D activity, has been associated with phagosomal escape. To demonstrate the feasibility of site-directed knockout mutagenesis of rickettsial genes and to generate a nonrevertible vaccine strain, we utilized homologous recombination to generate a pld mutant of the virulent R. prowazekii strain Madrid Evir. Using linear DNA for transformation, a double-crossover event resulted in the replacement of the rickettsial wild-type gene with a partially deleted pld gene. Linear DNA was used to prevent potentially revertible single-crossover events resulting in plasmid insertion. Southern blot and PCR analyses were used to confirm the presence of the desired mutation and to demonstrate clonality. While no phenotypic differences were observed between the mutant and wild-type strains when grown in tissue culture, the pld mutant exhibited attenuated virulence in the guinea pig model. In addition, animals immunized with the mutant strain were protected against subsequent challenge with the virulent Breinl strain, suggesting that this transformant could serve as a nonrevertible, attenuated vaccine strain. This study demonstrates the feasibility of generating site-directed rickettsial gene mutants, providing a new tool for understanding rickettsial biology and furthering advances in the prevention of epidemic typhus.)</td>\n",
       "      <td>(A genetic tool to introduce marker-free deletions is essential for multiple manipulations of genomes. We report a simple and efficient method to create marker-free deletion mutants of Bacillus subtilis through transformation with recombinant PCR products, using the Escherichia coli mazF gene encoding an endoribonuclease that cleaves free mRNAs as a counter-selection tool. Our method will be applicable to any bacterium in which introduction of the mazF cassette into the genome by double crossover homologous recombination is possible., Programmed cell death (PCD) is a genetically-controlled disassembly of the cell. In animal systems, the central core execution switch for apoptotic PCD is the activation of caspases (Cysteine-containing Aspartate-specific proteases). Accumulating evidence in recent years suggests the existence of caspase-like activity in plants and its functional involvement in various types of plant PCD, although no functional homologs of animal caspases were identified in plant genome. In this mini-review, we will cover the recent results on the existence of plant caspase-like proteases and introduce major technologies used in detecting the activation of caspase-like proteases during plant PCD.)</td>\n",
       "      <td>BC3ACT</td>\n",
       "      <td>CLS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>22.905071</td>\n",
       "      <td>3.087832</td>\n",
       "      <td>1.346595</td>\n",
       "      <td>(Mitochondrial (mt) heteroplasmy in the control region (CR) of the black-faced spoonbill was investigated using LA-PCR. To avoid amplification of transpositioned nuclear genome fragment from mtDNA (numt), PCR product of the almost-complete mitochondrial genome was amplified using primers designed to anneal on the COIII gene. Then nested LA-PCR product was amplified between the cyt b and 12S rRNA genes using the almost-complete mitochondrial genome PCR product as a template. Nucleotide sequencing revealed tandem duplication composed of two units. The first contains cyt b-1, tRNA(Thr)-1, tRNA(Pro)-1, ND6-1, tRNA(Glu)-1 and CR1, and the second consists of cyt b-2, tRNA(Thr)-2, tRNA(Pro)-2, ND6-2, tRNA(Glu)-2 and CR2, followed by tRNA(Phe) and 12S rRNA. The duplicated cyt b-2 sequence coincided with 499 bp at the 3' end of cyt b-1. With the exception of the CR, the other genes in the duplicated sequence were identical to the original corresponding gene. Even though both CR1 and CR2 contain functional blocks, such as a poly-C site, a goose hairpin and a TAS structure in Domain I, the 3' end of CR1 was followed by a 112 bp sequence (non-coding region) that was not found in CR2 or in sequence homology analysis of similar genes. Meanwhile, CR2 ended in a complicated repeat sequence. The 5' franking region in the Domain I (Region A) and the 3' franking region in the Domain I (Region B) of the two CRs evolve in quite different manners: Region A was highly variable between CR1 and CR2 in the same individuals, while Region B was almost identical between them, which indicates concerted evolution., Selenocysteine is the only genetically encoded amino acid in humans whose biosynthesis occurs on its cognate transfer RNA (tRNA). O-Phosphoseryl-tRNA:selenocysteinyl-tRNA synthase (SepSecS) catalyzes the final step of selenocysteine formation by a poorly understood tRNA-dependent mechanism. The crystal structure of human tRNA(Sec) in complex with SepSecS, phosphoserine, and thiophosphate, together with in vivo and in vitro enzyme assays, supports a pyridoxal phosphate-dependent mechanism of Sec-tRNA(Sec) formation. Two tRNA(Sec) molecules, with a fold distinct from other canonical tRNAs, bind to each SepSecS tetramer through their 13-base pair acceptor-TPsiC arm (where Psi indicates pseudouridine). The tRNA binding is likely to induce a conformational change in the enzyme's active site that allows a phosphoserine covalently attached to tRNA(Sec), but not free phosphoserine, to be oriented properly for the reaction to occur.)</td>\n",
       "      <td>(Mitochondrial (mt) heteroplasmy in the control region (CR) of the black-faced spoonbill was investigated using LA-PCR. To avoid amplification of transpositioned nuclear genome fragment from mtDNA (numt), PCR product of the almost-complete mitochondrial genome was amplified using primers designed to anneal on the COIII gene. Then nested LA-PCR product was amplified between the cyt b and 12S rRNA genes using the almost-complete mitochondrial genome PCR product as a template. Nucleotide sequencing revealed tandem duplication composed of two units. The first contains cyt b-1, tRNA(Thr)-1, tRNA(Pro)-1, ND6-1, tRNA(Glu)-1 and CR1, and the second consists of cyt b-2, tRNA(Thr)-2, tRNA(Pro)-2, ND6-2, tRNA(Glu)-2 and CR2, followed by tRNA(Phe) and 12S rRNA. The duplicated cyt b-2 sequence coincided with 499 bp at the 3' end of cyt b-1. With the exception of the CR, the other genes in the duplicated sequence were identical to the original corresponding gene. Even though both CR1 and CR2 contain functional blocks, such as a poly-C site, a goose hairpin and a TAS structure in Domain I, the 3' end of CR1 was followed by a 112 bp sequence (non-coding region) that was not found in CR2 or in sequence homology analysis of similar genes. Meanwhile, CR2 ended in a complicated repeat sequence. The 5' franking region in the Domain I (Region A) and the 3' franking region in the Domain I (Region B) of the two CRs evolve in quite different manners: Region A was highly variable between CR1 and CR2 in the same individuals, while Region B was almost identical between them, which indicates concerted evolution., Plasmodium falciparum is the major human malaria agent responsible for 200 to 300 million infections and one to three million deaths annually, mainly among African infants. The origin and evolution of this pathogen within the human lineage is still unresolved. A single species, P. reichenowi, which infects chimpanzees, is known to be a close sister lineage of P. falciparum. Here we report the discovery of a new Plasmodium species infecting Hominids. This new species has been isolated in two chimpanzees (Pan troglodytes) kept as pets by villagers in Gabon (Africa). Analysis of its complete mitochondrial genome (5529 nucleotides including Cyt b, Cox I and Cox III genes) reveals an older divergence of this lineage from the clade that includes P. falciparum and P. reichenowi (approximately 21+/-9 Myrs ago using Bayesian methods and considering that the divergence between P. falciparum and P. reichenowi occurred 4 to 7 million years ago as generally considered in the literature). This time frame would be congruent with the radiation of hominoids, suggesting that this Plasmodium lineage might have been present in early hominoids and that they may both have experienced a simultaneous diversification. Investigation of the nuclear genome of this new species will further the understanding of the genetic adaptations of P. falciparum to humans. The risk of transfer and emergence of this new species in humans must be now seriously considered given that it was found in two chimpanzees living in contact with humans and its close relatedness to the most virulent agent of malaria.)</td>\n",
       "      <td>(Mitochondrial (mt) heteroplasmy in the control region (CR) of the black-faced spoonbill was investigated using LA-PCR. To avoid amplification of transpositioned nuclear genome fragment from mtDNA (numt), PCR product of the almost-complete mitochondrial genome was amplified using primers designed to anneal on the COIII gene. Then nested LA-PCR product was amplified between the cyt b and 12S rRNA genes using the almost-complete mitochondrial genome PCR product as a template. Nucleotide sequencing revealed tandem duplication composed of two units. The first contains cyt b-1, tRNA(Thr)-1, tRNA(Pro)-1, ND6-1, tRNA(Glu)-1 and CR1, and the second consists of cyt b-2, tRNA(Thr)-2, tRNA(Pro)-2, ND6-2, tRNA(Glu)-2 and CR2, followed by tRNA(Phe) and 12S rRNA. The duplicated cyt b-2 sequence coincided with 499 bp at the 3' end of cyt b-1. With the exception of the CR, the other genes in the duplicated sequence were identical to the original corresponding gene. Even though both CR1 and CR2 contain functional blocks, such as a poly-C site, a goose hairpin and a TAS structure in Domain I, the 3' end of CR1 was followed by a 112 bp sequence (non-coding region) that was not found in CR2 or in sequence homology analysis of similar genes. Meanwhile, CR2 ended in a complicated repeat sequence. The 5' franking region in the Domain I (Region A) and the 3' franking region in the Domain I (Region B) of the two CRs evolve in quite different manners: Region A was highly variable between CR1 and CR2 in the same individuals, while Region B was almost identical between them, which indicates concerted evolution., Plasmodium falciparum is the major human malaria agent responsible for 200 to 300 million infections and one to three million deaths annually, mainly among African infants. The origin and evolution of this pathogen within the human lineage is still unresolved. A single species, P. reichenowi, which infects chimpanzees, is known to be a close sister lineage of P. falciparum. Here we report the discovery of a new Plasmodium species infecting Hominids. This new species has been isolated in two chimpanzees (Pan troglodytes) kept as pets by villagers in Gabon (Africa). Analysis of its complete mitochondrial genome (5529 nucleotides including Cyt b, Cox I and Cox III genes) reveals an older divergence of this lineage from the clade that includes P. falciparum and P. reichenowi (approximately 21+/-9 Myrs ago using Bayesian methods and considering that the divergence between P. falciparum and P. reichenowi occurred 4 to 7 million years ago as generally considered in the literature). This time frame would be congruent with the radiation of hominoids, suggesting that this Plasmodium lineage might have been present in early hominoids and that they may both have experienced a simultaneous diversification. Investigation of the nuclear genome of this new species will further the understanding of the genetic adaptations of P. falciparum to humans. The risk of transfer and emergence of this new species in humans must be now seriously considered given that it was found in two chimpanzees living in contact with humans and its close relatedness to the most virulent agent of malaria.)</td>\n",
       "      <td>BC3ACT</td>\n",
       "      <td>CLS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>27.548932</td>\n",
       "      <td>4.184284</td>\n",
       "      <td>0.924027</td>\n",
       "      <td>(In mangrove species the past geomorphic changes in coastal regions and reproductive systems are important factors of their distribution and genetic structure of populations. However, very little is known about the levels of genetic variation of Rhiozophora species in Southeast Asia. In this study, we surveyed levels and patterns of genetic variation as well as population structure of two sympatric mangrove species, Rhizophora apiculata and R. mucronata in Thailand, using five nuclear genes and two cpDNA regions. In all investigated DNA regions, nucleotide variation within species was low, while nucleotide divergence between the two species was considerable. The nuclear genes evolved 10 times faster than the cpDNA regions. In both R. apiculata and R. mucronata, significant positive F(IS) values were found, indicating deviation from Hardy-Weinberg proportions and a deficiency of heterozygotes. In both species, we found significant genetic differentiation between populations. However, the pattern of population differentiation (F(ST)) of R. apiculata differed from that of R. mucronata. Our results suggest that the two investigated species have different demographic history, even though they are sympatric and have similar reproductive systems., Whether or not bacteria form coherent evolutionary groups via means of genetic exchange and, hence, elicit distinct species boundaries remains an unsettled issue. A recent report implied that not only may the former be true but also, in fact, the clearly distinct Campylobacter jejuni and Campylobacter coli species may be converging as a consequence of increased interspecies gene flow fostered, presumably, by the recent invasion of an overlapping ecological niche (S. K. Sheppard, N. D. McCarthy, D. Falush, and M. C. Maiden, Science 320:237-239, 2008). We have reanalyzed the Campylobacter multilocus sequence typing database used in the previous study and found that the number of interspecies gene transfer events may actually be too infrequent to account, unequivocally, for species convergence. For instance, only 1 to 2% of the 4,507 Campylobacter isolates examined appeared to have imported gene alleles from another Campylobacter species. Furthermore, by analyzing the available Campylobacter genomic sequences, we show that although there seems to be a slightly higher number of exchanged genes between C. jejuni and C. coli relative to other comparable species ( approximately 10% versus 2 to 3% of the total genes in the genome, respectively), the function and spatial distribution in the genome of the exchanged genes are far from random, and hence, inconsistent with the species convergence hypothesis. In fact, the exchanged genes appear to be limited to a few environmentally selected cellular functions. Accordingly, these genes may represent important pathogenic determinants of pathogenic Campylobacter, and convergence of (any) two bacterial species remains to be seen.)</td>\n",
       "      <td>(In mangrove species the past geomorphic changes in coastal regions and reproductive systems are important factors of their distribution and genetic structure of populations. However, very little is known about the levels of genetic variation of Rhiozophora species in Southeast Asia. In this study, we surveyed levels and patterns of genetic variation as well as population structure of two sympatric mangrove species, Rhizophora apiculata and R. mucronata in Thailand, using five nuclear genes and two cpDNA regions. In all investigated DNA regions, nucleotide variation within species was low, while nucleotide divergence between the two species was considerable. The nuclear genes evolved 10 times faster than the cpDNA regions. In both R. apiculata and R. mucronata, significant positive F(IS) values were found, indicating deviation from Hardy-Weinberg proportions and a deficiency of heterozygotes. In both species, we found significant genetic differentiation between populations. However, the pattern of population differentiation (F(ST)) of R. apiculata differed from that of R. mucronata. Our results suggest that the two investigated species have different demographic history, even though they are sympatric and have similar reproductive systems., Mexico is developing the basis for genomic medicine to improve healthcare of its population. The extensive study of genetic diversity and linkage disequilibrium structure of different populations has made it possible to develop tagging and imputation strategies to comprehensively analyze common genetic variation in association studies of complex diseases. We assessed the benefit of a Mexican haplotype map to improve identification of genes related to common diseases in the Mexican population. We evaluated genetic diversity, linkage disequilibrium patterns, and extent of haplotype sharing using genomewide data from Mexican Mestizos from regions with different histories of admixture and particular population dynamics. Ancestry was evaluated by including 1 Mexican Amerindian group and data from the HapMap. Our results provide evidence of genetic differences between Mexican subpopulations that should be considered in the design and analysis of association studies of complex diseases. In addition, these results support the notion that a haplotype map of the Mexican Mestizo population can reduce the number of tag SNPs required to characterize common genetic variation in this population. This is one of the first genomewide genotyping efforts of a recently admixed population in Latin America.)</td>\n",
       "      <td>(In mangrove species the past geomorphic changes in coastal regions and reproductive systems are important factors of their distribution and genetic structure of populations. However, very little is known about the levels of genetic variation of Rhiozophora species in Southeast Asia. In this study, we surveyed levels and patterns of genetic variation as well as population structure of two sympatric mangrove species, Rhizophora apiculata and R. mucronata in Thailand, using five nuclear genes and two cpDNA regions. In all investigated DNA regions, nucleotide variation within species was low, while nucleotide divergence between the two species was considerable. The nuclear genes evolved 10 times faster than the cpDNA regions. In both R. apiculata and R. mucronata, significant positive F(IS) values were found, indicating deviation from Hardy-Weinberg proportions and a deficiency of heterozygotes. In both species, we found significant genetic differentiation between populations. However, the pattern of population differentiation (F(ST)) of R. apiculata differed from that of R. mucronata. Our results suggest that the two investigated species have different demographic history, even though they are sympatric and have similar reproductive systems., Mexico is developing the basis for genomic medicine to improve healthcare of its population. The extensive study of genetic diversity and linkage disequilibrium structure of different populations has made it possible to develop tagging and imputation strategies to comprehensively analyze common genetic variation in association studies of complex diseases. We assessed the benefit of a Mexican haplotype map to improve identification of genes related to common diseases in the Mexican population. We evaluated genetic diversity, linkage disequilibrium patterns, and extent of haplotype sharing using genomewide data from Mexican Mestizos from regions with different histories of admixture and particular population dynamics. Ancestry was evaluated by including 1 Mexican Amerindian group and data from the HapMap. Our results provide evidence of genetic differences between Mexican subpopulations that should be considered in the design and analysis of association studies of complex diseases. In addition, these results support the notion that a haplotype map of the Mexican Mestizo population can reduce the number of tag SNPs required to characterize common genetic variation in this population. This is one of the first genomewide genotyping efforts of a recently admixed population in Latin America.)</td>\n",
       "      <td>BC3ACT</td>\n",
       "      <td>CLS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>18.624808</td>\n",
       "      <td>7.035975</td>\n",
       "      <td>2.722033</td>\n",
       "      <td>(The cell envelope is the target for many antibiotics. In Gram-positive bacteria, membrane alterations and dysfunction caused by antibiotics are sensed mainly by two classes of signal transduction systems: the ECF sigma factors and the two-component signal transduction systems (TCSs). Enduracidin is an antibiotic that inhibits the transglycosylation step of peptidoglycan biosynthesis, and is an attractive target for further antibiotic development studies. We assessed transcriptional responses to enduracidin in Bacillus subtilis cells using a high-density tiling chip, and compared the results with responses to bacitracin, which inhibits the lipid II cycle of peptidoglycan synthesis. We exploited the quantitative advantage of the tiling chip to introduce a new criterion, an increase in transcriptional level, in addition to the conventional induction ratio, in order to distinguish genes of biological significance from those with lower induction ratios. Our results indicate that introduction of the new criterion led to unambiguous identification of core transcriptional responses to antibiotics, with a reduction in the number of possible background genes, compared to previous results obtained using gene arrays. We identified 129 genes that were significantly upregulated by enduracidin and/or bacitracin. Notably, we found that inactivation of the LiaRS TCS, which was the system most strongly induced by the two antibiotics, resulted in increased sensitivity to enduracidin, probably through a failure to induce LiaIH proteins. We noted that 33 genes belonging to the SigM regulon were induced by both antibiotics. Consistent with stronger induction of the SigM regulon in enduracidin-treated cells, inactivation of sigM resulted in increased sensitivity to enduracidin. In addition, and for the first time, we found that the Spx regulon was induced in cells challenged by enduracidin and bacitracin, suggesting that thiol-oxidative stress occurred in cells treated with antibiotics. These findings contribute to further our understanding of the molecular nature of genetic systems involved in antibiotic resistance., Increased expression and activity of proteins driving cell cycle progression as well as inactivation of endogenous inhibitors of cyclin-dependent kinases (CDKs) enhance the proliferative potential of cells. Escape of cells during malignant transformation from the proper cell cycle control rendering them independent from growth factors provides rationale for therapeutic targeting of CDKs. Exposure of rapidly growing human MCF-7 breast cancer and HeLa cervix cancer cells to roscovitine (ROSC), a selective inhibitor of CDKs, inhibits their proliferation by induction of cell cycle arrest and/or apoptosis. The outcome strongly depends on the intrinsic traits of the tumor cells, on their cell cycle status prior to the onset of treatment and also on ROSC concentration. At lower dose ROSC primarily inhibits the cell cycle-related CDKs resulting in a strong cell cycle arrest. Interestingly, ROSC arrests asynchronously growing cells at the G(2)/M transition irrespective of the status of their restriction checkpoint. However, the exposure of cancer cells synchronized after serum starvation in the late G(1) phase results in a transient G(1) arrest only in cells displaying the intact G(1)/S checkpoint. At higher dosage ROSC triggers apoptosis. In HeLa cells inhibition of the activity of CDK7 and, in consequence, that of RNA polymerase II is a major event that facilitates the initiation of caspase-dependent apoptosis. In contrast, in the caspase-3-deficient MCF-7 breast cancer cells ROSC induces apoptosis by a p53-dependent pathway. HIPK2-mediated activation of the p53 transcription factor by phosphorylation at Ser46 results in upregulation of p53AIP1 protein. This protein after de novo synthesis and translocation into the mitochondria promotes depolarization of the mitochondrial membrane.)</td>\n",
       "      <td>(The cell envelope is the target for many antibiotics. In Gram-positive bacteria, membrane alterations and dysfunction caused by antibiotics are sensed mainly by two classes of signal transduction systems: the ECF sigma factors and the two-component signal transduction systems (TCSs). Enduracidin is an antibiotic that inhibits the transglycosylation step of peptidoglycan biosynthesis, and is an attractive target for further antibiotic development studies. We assessed transcriptional responses to enduracidin in Bacillus subtilis cells using a high-density tiling chip, and compared the results with responses to bacitracin, which inhibits the lipid II cycle of peptidoglycan synthesis. We exploited the quantitative advantage of the tiling chip to introduce a new criterion, an increase in transcriptional level, in addition to the conventional induction ratio, in order to distinguish genes of biological significance from those with lower induction ratios. Our results indicate that introduction of the new criterion led to unambiguous identification of core transcriptional responses to antibiotics, with a reduction in the number of possible background genes, compared to previous results obtained using gene arrays. We identified 129 genes that were significantly upregulated by enduracidin and/or bacitracin. Notably, we found that inactivation of the LiaRS TCS, which was the system most strongly induced by the two antibiotics, resulted in increased sensitivity to enduracidin, probably through a failure to induce LiaIH proteins. We noted that 33 genes belonging to the SigM regulon were induced by both antibiotics. Consistent with stronger induction of the SigM regulon in enduracidin-treated cells, inactivation of sigM resulted in increased sensitivity to enduracidin. In addition, and for the first time, we found that the Spx regulon was induced in cells challenged by enduracidin and bacitracin, suggesting that thiol-oxidative stress occurred in cells treated with antibiotics. These findings contribute to further our understanding of the molecular nature of genetic systems involved in antibiotic resistance., Two-component signal-transduction systems (TCSs) of bacteria are considered to form an intricate signal network to cope with various environmental stresses. One example of such a network in Escherichia coli is the signal transduction cascade from the EvgS/EvgA system to the PhoQ/PhoP system, where activation of the EvgS/EvgA system promotes expression of PhoP-activated genes. As a factor connecting this signal transduction cascade, we have identified a small inner membrane protein (65 aa), B1500. Expression of the b1500 gene is directly regulated by the EvgS/EvgA system, and b1500 expression from a heterologous promoter simultaneously activated the expression of mgtA and other PhoP regulon genes. This activation was PhoQ/PhoP-dependent and EvgS/EvgA-independent. Furthermore, deletion of b1500 from an EvgS-activated strain suppressed mgtA expression. B1500 is localized in the inner membrane, and bacterial two-hybrid data showed that B1500 formed a complex with the sensor PhoQ. These results indicate that the small membrane protein, B1500, connected the signal transduction between EvgS/EvgA and PhoQ/PhoP systems by directly interacting with PhoQ, thus activating the PhoQ/PhoP system.)</td>\n",
       "      <td>(The cell envelope is the target for many antibiotics. In Gram-positive bacteria, membrane alterations and dysfunction caused by antibiotics are sensed mainly by two classes of signal transduction systems: the ECF sigma factors and the two-component signal transduction systems (TCSs). Enduracidin is an antibiotic that inhibits the transglycosylation step of peptidoglycan biosynthesis, and is an attractive target for further antibiotic development studies. We assessed transcriptional responses to enduracidin in Bacillus subtilis cells using a high-density tiling chip, and compared the results with responses to bacitracin, which inhibits the lipid II cycle of peptidoglycan synthesis. We exploited the quantitative advantage of the tiling chip to introduce a new criterion, an increase in transcriptional level, in addition to the conventional induction ratio, in order to distinguish genes of biological significance from those with lower induction ratios. Our results indicate that introduction of the new criterion led to unambiguous identification of core transcriptional responses to antibiotics, with a reduction in the number of possible background genes, compared to previous results obtained using gene arrays. We identified 129 genes that were significantly upregulated by enduracidin and/or bacitracin. Notably, we found that inactivation of the LiaRS TCS, which was the system most strongly induced by the two antibiotics, resulted in increased sensitivity to enduracidin, probably through a failure to induce LiaIH proteins. We noted that 33 genes belonging to the SigM regulon were induced by both antibiotics. Consistent with stronger induction of the SigM regulon in enduracidin-treated cells, inactivation of sigM resulted in increased sensitivity to enduracidin. In addition, and for the first time, we found that the Spx regulon was induced in cells challenged by enduracidin and bacitracin, suggesting that thiol-oxidative stress occurred in cells treated with antibiotics. These findings contribute to further our understanding of the molecular nature of genetic systems involved in antibiotic resistance., Two-component signal-transduction systems (TCSs) of bacteria are considered to form an intricate signal network to cope with various environmental stresses. One example of such a network in Escherichia coli is the signal transduction cascade from the EvgS/EvgA system to the PhoQ/PhoP system, where activation of the EvgS/EvgA system promotes expression of PhoP-activated genes. As a factor connecting this signal transduction cascade, we have identified a small inner membrane protein (65 aa), B1500. Expression of the b1500 gene is directly regulated by the EvgS/EvgA system, and b1500 expression from a heterologous promoter simultaneously activated the expression of mgtA and other PhoP regulon genes. This activation was PhoQ/PhoP-dependent and EvgS/EvgA-independent. Furthermore, deletion of b1500 from an EvgS-activated strain suppressed mgtA expression. B1500 is localized in the inner membrane, and bacterial two-hybrid data showed that B1500 formed a complex with the sensor PhoQ. These results indicate that the small membrane protein, B1500, connected the signal transduction between EvgS/EvgA and PhoQ/PhoP systems by directly interacting with PhoQ, thus activating the PhoQ/PhoP system.)</td>\n",
       "      <td>BC3ACT</td>\n",
       "      <td>CLS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>30.773328</td>\n",
       "      <td>3.112864</td>\n",
       "      <td>0.590549</td>\n",
       "      <td>(Microscale technologies have emerged as a powerful tool for studying and manipulating biological systems and miniaturizing experiments. However, the lack of software complementing these techniques has made it difficult to apply them for many high-throughput experiments. This work establishes Arraycount, an approach to automatically count cells in microwell arrays. The procedure consists of fluorescent microscope imaging of cells that are seeded in microwells of a microarray system and then analyzing images via computer to recognize the array and count cells inside each microwell. To start counting, green and red fluorescent images (representing live and dead cells, respectively) are extracted from the original image and processed separately. A template-matching algorithm is proposed in which pre-defined well and cell templates are matched against the red and green images to locate microwells and cells. Subsequently, local maxima in the correlation maps are determined and local maxima maps are thresholded. At the end, the software records the cell counts for each detected microwell on the original image in high-throughput. The automated counting was shown to be accurate compared with manual counting, with a difference of approximately 1-2 cells per microwell: based on cell concentration, the absolute difference between manual and automatic counting measurements was 2.5-13%., Elimination of peripheral tumors by adoptively transferred tumor-specific T cells may require killing of cancer cells and tumor stromal cells. Tumor Ags are cross-presented on stromal cells, resulting in direct cytotoxic T cell (CTL) killing of both Ag-expressing cancer cells and stromal cells. Indirect killing of Ag loss variant cells also occurs. We show here that similar processes occur in a brain tumor stromal environment. We used murine cancer cell lines that express high or low levels of a peptide Ag, SIYRYYGL (SIY), recognized by transgenic 2C CD8(+) T cells. The two cell lines are killed with equivalent efficiency by 2C T cells in vitro. Following adoptive transfer of 2C T cells into mice with established SIY-Hi or SIY-Lo brain tumors, tumors of both types regressed, but low-Ag-expressing tumors recurred. High-Ag-expressing tumors contained CD11b(+) cells cross-presenting SIY peptide and were completely eliminated by 2C T cells. To further test the role of cross-presentation, RAG1(-/-) H-2(b) mice were infused with H-2(k) tumor cells expressing high levels of SIY peptide. Adoptively transferred 2C T cells are able to kill cross-presenting H-2(b) stromal cells but not H-2(k) tumor cells. In peripheral models, this paradigm led to a small static tumor. In the brain, activated 2C T cells were able to kill cross-presenting CD11b(+) cells and completely eliminate the H-2(k) tumors in most mice. Targeting brain tumor stroma or increasing Ag shedding from tumor cells to enhance cross-presentation may improve the clinical success of T cell adoptive therapies.)</td>\n",
       "      <td>(Microscale technologies have emerged as a powerful tool for studying and manipulating biological systems and miniaturizing experiments. However, the lack of software complementing these techniques has made it difficult to apply them for many high-throughput experiments. This work establishes Arraycount, an approach to automatically count cells in microwell arrays. The procedure consists of fluorescent microscope imaging of cells that are seeded in microwells of a microarray system and then analyzing images via computer to recognize the array and count cells inside each microwell. To start counting, green and red fluorescent images (representing live and dead cells, respectively) are extracted from the original image and processed separately. A template-matching algorithm is proposed in which pre-defined well and cell templates are matched against the red and green images to locate microwells and cells. Subsequently, local maxima in the correlation maps are determined and local maxima maps are thresholded. At the end, the software records the cell counts for each detected microwell on the original image in high-throughput. The automated counting was shown to be accurate compared with manual counting, with a difference of approximately 1-2 cells per microwell: based on cell concentration, the absolute difference between manual and automatic counting measurements was 2.5-13%., Current yeast interactome network maps contain several hundred molecular complexes with limited and somewhat controversial representation of direct binary interactions. We carried out a comparative quality assessment of current yeast interactome data sets, demonstrating that high-throughput yeast two-hybrid (Y2H) screening provides high-quality binary interaction information. Because a large fraction of the yeast binary interactome remains to be mapped, we developed an empirically controlled mapping framework to produce a &amp;quot;second-generation&amp;quot; high-quality, high-throughput Y2H data set covering approximately 20% of all yeast binary interactions. Both Y2H and affinity purification followed by mass spectrometry (AP/MS) data are of equally high quality but of a fundamentally different and complementary nature, resulting in networks with different topological and biological properties. Compared to co-complex interactome models, this binary map is enriched for transient signaling interactions and intercomplex connections with a highly significant clustering between essential proteins. Rather than correlating with essentiality, protein connectivity correlates with genetic pleiotropy.)</td>\n",
       "      <td>(Microscale technologies have emerged as a powerful tool for studying and manipulating biological systems and miniaturizing experiments. However, the lack of software complementing these techniques has made it difficult to apply them for many high-throughput experiments. This work establishes Arraycount, an approach to automatically count cells in microwell arrays. The procedure consists of fluorescent microscope imaging of cells that are seeded in microwells of a microarray system and then analyzing images via computer to recognize the array and count cells inside each microwell. To start counting, green and red fluorescent images (representing live and dead cells, respectively) are extracted from the original image and processed separately. A template-matching algorithm is proposed in which pre-defined well and cell templates are matched against the red and green images to locate microwells and cells. Subsequently, local maxima in the correlation maps are determined and local maxima maps are thresholded. At the end, the software records the cell counts for each detected microwell on the original image in high-throughput. The automated counting was shown to be accurate compared with manual counting, with a difference of approximately 1-2 cells per microwell: based on cell concentration, the absolute difference between manual and automatic counting measurements was 2.5-13%., BACKGROUND: Flow cytometry technology is widely used in both health care and research. The rapid expansion of flow cytometry applications has outpaced the development of data storage and analysis tools. Collaborative efforts being taken to eliminate this gap include building common vocabularies and ontologies, designing generic data models, and defining data exchange formats. The Minimum Information about a Flow Cytometry Experiment (MIFlowCyt) standard was recently adopted by the International Society for Advancement of Cytometry. This standard guides researchers on the information that should be included in peer reviewed publications, but it is insufficient for data exchange and integration between computational systems. The Functional Genomics Experiment (FuGE) formalizes common aspects of comprehensive and high throughput experiments across different biological technologies. We have extended FuGE object model to accommodate flow cytometry data and metadata. METHODS: We used the MagicDraw modelling tool to design a UML model (Flow-OM) according to the FuGE extension guidelines and the AndroMDA toolkit to transform the model to a markup language (Flow-ML). We mapped each MIFlowCyt term to either an existing FuGE class or to a new FuGEFlow class. The development environment was validated by comparing the official FuGE XSD to the schema we generated from the FuGE object model using our configuration. After the Flow-OM model was completed, the final version of the Flow-ML was generated and validated against an example MIFlowCyt compliant experiment description. RESULTS: The extension of FuGE for flow cytometry has resulted in a generic FuGE-compliant data model (FuGEFlow), which accommodates and links together all information required by MIFlowCyt. The FuGEFlow model can be used to build software and databases using FuGE software toolkits to facilitate automated exchange and manipulation of potentially large flow cytometry experimental data sets. Additional project documentation, including reusable design patterns and a guide for setting up a development environment, was contributed back to the FuGE project. CONCLUSION: We have shown that an extension of FuGE can be used to transform minimum information requirements in natural language to markup language in XML. Extending FuGE required significant effort, but in our experiences the benefits outweighed the costs. The FuGEFlow is expected to play a central role in describing flow cytometry experiments and ultimately facilitating data exchange including public flow cytometry repositories currently under development.)</td>\n",
       "      <td>BC3ACT</td>\n",
       "      <td>CLS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unigram    Bigram   Trigram  \\\n",
       "0  24.053512  3.221897  0.000000   \n",
       "1  22.905071  3.087832  1.346595   \n",
       "2  27.548932  4.184284  0.924027   \n",
       "3  18.624808  7.035975  2.722033   \n",
       "4  30.773328  3.112864  0.590549   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Unigram_detail  \\\n",
       "0  (A genetic tool to introduce marker-free deletions is essential for multiple manipulations of genomes. We report a simple and efficient method to create marker-free deletion mutants of Bacillus subtilis through transformation with recombinant PCR products, using the Escherichia coli mazF gene encoding an endoribonuclease that cleaves free mRNAs as a counter-selection tool. Our method will be applicable to any bacterium in which introduction of the mazF cassette into the genome by double crossover homologous recombination is possible., The next-generation sequencing technology coupled with the growing number of genome sequences opens the opportunity to redesign genotyping strategies for more effective genetic mapping and genome analysis. We have developed a high-throughput method for genotyping recombinant populations utilizing whole-genome resequencing data generated by the Illumina Genome Analyzer. A sliding window approach is designed to collectively examine genome-wide single nucleotide polymorphisms for genotype calling and recombination breakpoint determination. Using this method, we constructed a genetic map for 150 rice recombinant inbred lines with an expected genotype calling accuracy of 99.94% and a resolution of recombination breakpoints within an average of 40 kb. In comparison to the genetic map constructed with 287 PCR-based markers for the rice population, the sequencing-based method was approximately 20x faster in data collection and 35x more precise in recombination breakpoint determination. Using the sequencing-based genetic map, we located a quantitative trait locus of large effect on plant height in a 100-kb region containing the rice &quot;green revolution&quot; gene. Through computer simulation, we demonstrate that the method is robust for different types of mapping populations derived from organisms with variable quality of genome sequences and is feasible for organisms with large genome sizes and low polymorphisms. With continuous advances in sequencing technologies, this genome-based method may replace the conventional marker-based genotyping approach to provide a powerful tool for large-scale gene discovery and for addressing a wide range of biological questions.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
       "1  (Mitochondrial (mt) heteroplasmy in the control region (CR) of the black-faced spoonbill was investigated using LA-PCR. To avoid amplification of transpositioned nuclear genome fragment from mtDNA (numt), PCR product of the almost-complete mitochondrial genome was amplified using primers designed to anneal on the COIII gene. Then nested LA-PCR product was amplified between the cyt b and 12S rRNA genes using the almost-complete mitochondrial genome PCR product as a template. Nucleotide sequencing revealed tandem duplication composed of two units. The first contains cyt b-1, tRNA(Thr)-1, tRNA(Pro)-1, ND6-1, tRNA(Glu)-1 and CR1, and the second consists of cyt b-2, tRNA(Thr)-2, tRNA(Pro)-2, ND6-2, tRNA(Glu)-2 and CR2, followed by tRNA(Phe) and 12S rRNA. The duplicated cyt b-2 sequence coincided with 499 bp at the 3' end of cyt b-1. With the exception of the CR, the other genes in the duplicated sequence were identical to the original corresponding gene. Even though both CR1 and CR2 contain functional blocks, such as a poly-C site, a goose hairpin and a TAS structure in Domain I, the 3' end of CR1 was followed by a 112 bp sequence (non-coding region) that was not found in CR2 or in sequence homology analysis of similar genes. Meanwhile, CR2 ended in a complicated repeat sequence. The 5' franking region in the Domain I (Region A) and the 3' franking region in the Domain I (Region B) of the two CRs evolve in quite different manners: Region A was highly variable between CR1 and CR2 in the same individuals, while Region B was almost identical between them, which indicates concerted evolution., Selenocysteine is the only genetically encoded amino acid in humans whose biosynthesis occurs on its cognate transfer RNA (tRNA). O-Phosphoseryl-tRNA:selenocysteinyl-tRNA synthase (SepSecS) catalyzes the final step of selenocysteine formation by a poorly understood tRNA-dependent mechanism. The crystal structure of human tRNA(Sec) in complex with SepSecS, phosphoserine, and thiophosphate, together with in vivo and in vitro enzyme assays, supports a pyridoxal phosphate-dependent mechanism of Sec-tRNA(Sec) formation. Two tRNA(Sec) molecules, with a fold distinct from other canonical tRNAs, bind to each SepSecS tetramer through their 13-base pair acceptor-TPsiC arm (where Psi indicates pseudouridine). The tRNA binding is likely to induce a conformational change in the enzyme's active site that allows a phosphoserine covalently attached to tRNA(Sec), but not free phosphoserine, to be oriented properly for the reaction to occur.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
       "2  (In mangrove species the past geomorphic changes in coastal regions and reproductive systems are important factors of their distribution and genetic structure of populations. However, very little is known about the levels of genetic variation of Rhiozophora species in Southeast Asia. In this study, we surveyed levels and patterns of genetic variation as well as population structure of two sympatric mangrove species, Rhizophora apiculata and R. mucronata in Thailand, using five nuclear genes and two cpDNA regions. In all investigated DNA regions, nucleotide variation within species was low, while nucleotide divergence between the two species was considerable. The nuclear genes evolved 10 times faster than the cpDNA regions. In both R. apiculata and R. mucronata, significant positive F(IS) values were found, indicating deviation from Hardy-Weinberg proportions and a deficiency of heterozygotes. In both species, we found significant genetic differentiation between populations. However, the pattern of population differentiation (F(ST)) of R. apiculata differed from that of R. mucronata. Our results suggest that the two investigated species have different demographic history, even though they are sympatric and have similar reproductive systems., Whether or not bacteria form coherent evolutionary groups via means of genetic exchange and, hence, elicit distinct species boundaries remains an unsettled issue. A recent report implied that not only may the former be true but also, in fact, the clearly distinct Campylobacter jejuni and Campylobacter coli species may be converging as a consequence of increased interspecies gene flow fostered, presumably, by the recent invasion of an overlapping ecological niche (S. K. Sheppard, N. D. McCarthy, D. Falush, and M. C. Maiden, Science 320:237-239, 2008). We have reanalyzed the Campylobacter multilocus sequence typing database used in the previous study and found that the number of interspecies gene transfer events may actually be too infrequent to account, unequivocally, for species convergence. For instance, only 1 to 2% of the 4,507 Campylobacter isolates examined appeared to have imported gene alleles from another Campylobacter species. Furthermore, by analyzing the available Campylobacter genomic sequences, we show that although there seems to be a slightly higher number of exchanged genes between C. jejuni and C. coli relative to other comparable species ( approximately 10% versus 2 to 3% of the total genes in the genome, respectively), the function and spatial distribution in the genome of the exchanged genes are far from random, and hence, inconsistent with the species convergence hypothesis. In fact, the exchanged genes appear to be limited to a few environmentally selected cellular functions. Accordingly, these genes may represent important pathogenic determinants of pathogenic Campylobacter, and convergence of (any) two bacterial species remains to be seen.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "3  (The cell envelope is the target for many antibiotics. In Gram-positive bacteria, membrane alterations and dysfunction caused by antibiotics are sensed mainly by two classes of signal transduction systems: the ECF sigma factors and the two-component signal transduction systems (TCSs). Enduracidin is an antibiotic that inhibits the transglycosylation step of peptidoglycan biosynthesis, and is an attractive target for further antibiotic development studies. We assessed transcriptional responses to enduracidin in Bacillus subtilis cells using a high-density tiling chip, and compared the results with responses to bacitracin, which inhibits the lipid II cycle of peptidoglycan synthesis. We exploited the quantitative advantage of the tiling chip to introduce a new criterion, an increase in transcriptional level, in addition to the conventional induction ratio, in order to distinguish genes of biological significance from those with lower induction ratios. Our results indicate that introduction of the new criterion led to unambiguous identification of core transcriptional responses to antibiotics, with a reduction in the number of possible background genes, compared to previous results obtained using gene arrays. We identified 129 genes that were significantly upregulated by enduracidin and/or bacitracin. Notably, we found that inactivation of the LiaRS TCS, which was the system most strongly induced by the two antibiotics, resulted in increased sensitivity to enduracidin, probably through a failure to induce LiaIH proteins. We noted that 33 genes belonging to the SigM regulon were induced by both antibiotics. Consistent with stronger induction of the SigM regulon in enduracidin-treated cells, inactivation of sigM resulted in increased sensitivity to enduracidin. In addition, and for the first time, we found that the Spx regulon was induced in cells challenged by enduracidin and bacitracin, suggesting that thiol-oxidative stress occurred in cells treated with antibiotics. These findings contribute to further our understanding of the molecular nature of genetic systems involved in antibiotic resistance., Increased expression and activity of proteins driving cell cycle progression as well as inactivation of endogenous inhibitors of cyclin-dependent kinases (CDKs) enhance the proliferative potential of cells. Escape of cells during malignant transformation from the proper cell cycle control rendering them independent from growth factors provides rationale for therapeutic targeting of CDKs. Exposure of rapidly growing human MCF-7 breast cancer and HeLa cervix cancer cells to roscovitine (ROSC), a selective inhibitor of CDKs, inhibits their proliferation by induction of cell cycle arrest and/or apoptosis. The outcome strongly depends on the intrinsic traits of the tumor cells, on their cell cycle status prior to the onset of treatment and also on ROSC concentration. At lower dose ROSC primarily inhibits the cell cycle-related CDKs resulting in a strong cell cycle arrest. Interestingly, ROSC arrests asynchronously growing cells at the G(2)/M transition irrespective of the status of their restriction checkpoint. However, the exposure of cancer cells synchronized after serum starvation in the late G(1) phase results in a transient G(1) arrest only in cells displaying the intact G(1)/S checkpoint. At higher dosage ROSC triggers apoptosis. In HeLa cells inhibition of the activity of CDK7 and, in consequence, that of RNA polymerase II is a major event that facilitates the initiation of caspase-dependent apoptosis. In contrast, in the caspase-3-deficient MCF-7 breast cancer cells ROSC induces apoptosis by a p53-dependent pathway. HIPK2-mediated activation of the p53 transcription factor by phosphorylation at Ser46 results in upregulation of p53AIP1 protein. This protein after de novo synthesis and translocation into the mitochondria promotes depolarization of the mitochondrial membrane.)   \n",
       "4  (Microscale technologies have emerged as a powerful tool for studying and manipulating biological systems and miniaturizing experiments. However, the lack of software complementing these techniques has made it difficult to apply them for many high-throughput experiments. This work establishes Arraycount, an approach to automatically count cells in microwell arrays. The procedure consists of fluorescent microscope imaging of cells that are seeded in microwells of a microarray system and then analyzing images via computer to recognize the array and count cells inside each microwell. To start counting, green and red fluorescent images (representing live and dead cells, respectively) are extracted from the original image and processed separately. A template-matching algorithm is proposed in which pre-defined well and cell templates are matched against the red and green images to locate microwells and cells. Subsequently, local maxima in the correlation maps are determined and local maxima maps are thresholded. At the end, the software records the cell counts for each detected microwell on the original image in high-throughput. The automated counting was shown to be accurate compared with manual counting, with a difference of approximately 1-2 cells per microwell: based on cell concentration, the absolute difference between manual and automatic counting measurements was 2.5-13%., Elimination of peripheral tumors by adoptively transferred tumor-specific T cells may require killing of cancer cells and tumor stromal cells. Tumor Ags are cross-presented on stromal cells, resulting in direct cytotoxic T cell (CTL) killing of both Ag-expressing cancer cells and stromal cells. Indirect killing of Ag loss variant cells also occurs. We show here that similar processes occur in a brain tumor stromal environment. We used murine cancer cell lines that express high or low levels of a peptide Ag, SIYRYYGL (SIY), recognized by transgenic 2C CD8(+) T cells. The two cell lines are killed with equivalent efficiency by 2C T cells in vitro. Following adoptive transfer of 2C T cells into mice with established SIY-Hi or SIY-Lo brain tumors, tumors of both types regressed, but low-Ag-expressing tumors recurred. High-Ag-expressing tumors contained CD11b(+) cells cross-presenting SIY peptide and were completely eliminated by 2C T cells. To further test the role of cross-presentation, RAG1(-/-) H-2(b) mice were infused with H-2(k) tumor cells expressing high levels of SIY peptide. Adoptively transferred 2C T cells are able to kill cross-presenting H-2(b) stromal cells but not H-2(k) tumor cells. In peripheral models, this paradigm led to a small static tumor. In the brain, activated 2C T cells were able to kill cross-presenting CD11b(+) cells and completely eliminate the H-2(k) tumors in most mice. Targeting brain tumor stroma or increasing Ag shedding from tumor cells to enhance cross-presentation may improve the clinical success of T cell adoptive therapies.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Bigram_detail  \\\n",
       "0  (A genetic tool to introduce marker-free deletions is essential for multiple manipulations of genomes. We report a simple and efficient method to create marker-free deletion mutants of Bacillus subtilis through transformation with recombinant PCR products, using the Escherichia coli mazF gene encoding an endoribonuclease that cleaves free mRNAs as a counter-selection tool. Our method will be applicable to any bacterium in which introduction of the mazF cassette into the genome by double crossover homologous recombination is possible., Rickettsia prowazekii, the causative agent of epidemic typhus, is an obligately intracytoplasmic bacterium, a lifestyle that imposes significant barriers to genetic manipulation. The key to understanding how this unique bacterium evades host immunity is the mutagenesis of selected genes hypothesized to be involved in virulence. The R. prowazekii pld gene, encoding a protein with phospholipase D activity, has been associated with phagosomal escape. To demonstrate the feasibility of site-directed knockout mutagenesis of rickettsial genes and to generate a nonrevertible vaccine strain, we utilized homologous recombination to generate a pld mutant of the virulent R. prowazekii strain Madrid Evir. Using linear DNA for transformation, a double-crossover event resulted in the replacement of the rickettsial wild-type gene with a partially deleted pld gene. Linear DNA was used to prevent potentially revertible single-crossover events resulting in plasmid insertion. Southern blot and PCR analyses were used to confirm the presence of the desired mutation and to demonstrate clonality. While no phenotypic differences were observed between the mutant and wild-type strains when grown in tissue culture, the pld mutant exhibited attenuated virulence in the guinea pig model. In addition, animals immunized with the mutant strain were protected against subsequent challenge with the virulent Breinl strain, suggesting that this transformant could serve as a nonrevertible, attenuated vaccine strain. This study demonstrates the feasibility of generating site-directed rickettsial gene mutants, providing a new tool for understanding rickettsial biology and furthering advances in the prevention of epidemic typhus.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "1  (Mitochondrial (mt) heteroplasmy in the control region (CR) of the black-faced spoonbill was investigated using LA-PCR. To avoid amplification of transpositioned nuclear genome fragment from mtDNA (numt), PCR product of the almost-complete mitochondrial genome was amplified using primers designed to anneal on the COIII gene. Then nested LA-PCR product was amplified between the cyt b and 12S rRNA genes using the almost-complete mitochondrial genome PCR product as a template. Nucleotide sequencing revealed tandem duplication composed of two units. The first contains cyt b-1, tRNA(Thr)-1, tRNA(Pro)-1, ND6-1, tRNA(Glu)-1 and CR1, and the second consists of cyt b-2, tRNA(Thr)-2, tRNA(Pro)-2, ND6-2, tRNA(Glu)-2 and CR2, followed by tRNA(Phe) and 12S rRNA. The duplicated cyt b-2 sequence coincided with 499 bp at the 3' end of cyt b-1. With the exception of the CR, the other genes in the duplicated sequence were identical to the original corresponding gene. Even though both CR1 and CR2 contain functional blocks, such as a poly-C site, a goose hairpin and a TAS structure in Domain I, the 3' end of CR1 was followed by a 112 bp sequence (non-coding region) that was not found in CR2 or in sequence homology analysis of similar genes. Meanwhile, CR2 ended in a complicated repeat sequence. The 5' franking region in the Domain I (Region A) and the 3' franking region in the Domain I (Region B) of the two CRs evolve in quite different manners: Region A was highly variable between CR1 and CR2 in the same individuals, while Region B was almost identical between them, which indicates concerted evolution., Plasmodium falciparum is the major human malaria agent responsible for 200 to 300 million infections and one to three million deaths annually, mainly among African infants. The origin and evolution of this pathogen within the human lineage is still unresolved. A single species, P. reichenowi, which infects chimpanzees, is known to be a close sister lineage of P. falciparum. Here we report the discovery of a new Plasmodium species infecting Hominids. This new species has been isolated in two chimpanzees (Pan troglodytes) kept as pets by villagers in Gabon (Africa). Analysis of its complete mitochondrial genome (5529 nucleotides including Cyt b, Cox I and Cox III genes) reveals an older divergence of this lineage from the clade that includes P. falciparum and P. reichenowi (approximately 21+/-9 Myrs ago using Bayesian methods and considering that the divergence between P. falciparum and P. reichenowi occurred 4 to 7 million years ago as generally considered in the literature). This time frame would be congruent with the radiation of hominoids, suggesting that this Plasmodium lineage might have been present in early hominoids and that they may both have experienced a simultaneous diversification. Investigation of the nuclear genome of this new species will further the understanding of the genetic adaptations of P. falciparum to humans. The risk of transfer and emergence of this new species in humans must be now seriously considered given that it was found in two chimpanzees living in contact with humans and its close relatedness to the most virulent agent of malaria.)                                                                                                                                       \n",
       "2  (In mangrove species the past geomorphic changes in coastal regions and reproductive systems are important factors of their distribution and genetic structure of populations. However, very little is known about the levels of genetic variation of Rhiozophora species in Southeast Asia. In this study, we surveyed levels and patterns of genetic variation as well as population structure of two sympatric mangrove species, Rhizophora apiculata and R. mucronata in Thailand, using five nuclear genes and two cpDNA regions. In all investigated DNA regions, nucleotide variation within species was low, while nucleotide divergence between the two species was considerable. The nuclear genes evolved 10 times faster than the cpDNA regions. In both R. apiculata and R. mucronata, significant positive F(IS) values were found, indicating deviation from Hardy-Weinberg proportions and a deficiency of heterozygotes. In both species, we found significant genetic differentiation between populations. However, the pattern of population differentiation (F(ST)) of R. apiculata differed from that of R. mucronata. Our results suggest that the two investigated species have different demographic history, even though they are sympatric and have similar reproductive systems., Mexico is developing the basis for genomic medicine to improve healthcare of its population. The extensive study of genetic diversity and linkage disequilibrium structure of different populations has made it possible to develop tagging and imputation strategies to comprehensively analyze common genetic variation in association studies of complex diseases. We assessed the benefit of a Mexican haplotype map to improve identification of genes related to common diseases in the Mexican population. We evaluated genetic diversity, linkage disequilibrium patterns, and extent of haplotype sharing using genomewide data from Mexican Mestizos from regions with different histories of admixture and particular population dynamics. Ancestry was evaluated by including 1 Mexican Amerindian group and data from the HapMap. Our results provide evidence of genetic differences between Mexican subpopulations that should be considered in the design and analysis of association studies of complex diseases. In addition, these results support the notion that a haplotype map of the Mexican Mestizo population can reduce the number of tag SNPs required to characterize common genetic variation in this population. This is one of the first genomewide genotyping efforts of a recently admixed population in Latin America.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "3  (The cell envelope is the target for many antibiotics. In Gram-positive bacteria, membrane alterations and dysfunction caused by antibiotics are sensed mainly by two classes of signal transduction systems: the ECF sigma factors and the two-component signal transduction systems (TCSs). Enduracidin is an antibiotic that inhibits the transglycosylation step of peptidoglycan biosynthesis, and is an attractive target for further antibiotic development studies. We assessed transcriptional responses to enduracidin in Bacillus subtilis cells using a high-density tiling chip, and compared the results with responses to bacitracin, which inhibits the lipid II cycle of peptidoglycan synthesis. We exploited the quantitative advantage of the tiling chip to introduce a new criterion, an increase in transcriptional level, in addition to the conventional induction ratio, in order to distinguish genes of biological significance from those with lower induction ratios. Our results indicate that introduction of the new criterion led to unambiguous identification of core transcriptional responses to antibiotics, with a reduction in the number of possible background genes, compared to previous results obtained using gene arrays. We identified 129 genes that were significantly upregulated by enduracidin and/or bacitracin. Notably, we found that inactivation of the LiaRS TCS, which was the system most strongly induced by the two antibiotics, resulted in increased sensitivity to enduracidin, probably through a failure to induce LiaIH proteins. We noted that 33 genes belonging to the SigM regulon were induced by both antibiotics. Consistent with stronger induction of the SigM regulon in enduracidin-treated cells, inactivation of sigM resulted in increased sensitivity to enduracidin. In addition, and for the first time, we found that the Spx regulon was induced in cells challenged by enduracidin and bacitracin, suggesting that thiol-oxidative stress occurred in cells treated with antibiotics. These findings contribute to further our understanding of the molecular nature of genetic systems involved in antibiotic resistance., Two-component signal-transduction systems (TCSs) of bacteria are considered to form an intricate signal network to cope with various environmental stresses. One example of such a network in Escherichia coli is the signal transduction cascade from the EvgS/EvgA system to the PhoQ/PhoP system, where activation of the EvgS/EvgA system promotes expression of PhoP-activated genes. As a factor connecting this signal transduction cascade, we have identified a small inner membrane protein (65 aa), B1500. Expression of the b1500 gene is directly regulated by the EvgS/EvgA system, and b1500 expression from a heterologous promoter simultaneously activated the expression of mgtA and other PhoP regulon genes. This activation was PhoQ/PhoP-dependent and EvgS/EvgA-independent. Furthermore, deletion of b1500 from an EvgS-activated strain suppressed mgtA expression. B1500 is localized in the inner membrane, and bacterial two-hybrid data showed that B1500 formed a complex with the sensor PhoQ. These results indicate that the small membrane protein, B1500, connected the signal transduction between EvgS/EvgA and PhoQ/PhoP systems by directly interacting with PhoQ, thus activating the PhoQ/PhoP system.)   \n",
       "4  (Microscale technologies have emerged as a powerful tool for studying and manipulating biological systems and miniaturizing experiments. However, the lack of software complementing these techniques has made it difficult to apply them for many high-throughput experiments. This work establishes Arraycount, an approach to automatically count cells in microwell arrays. The procedure consists of fluorescent microscope imaging of cells that are seeded in microwells of a microarray system and then analyzing images via computer to recognize the array and count cells inside each microwell. To start counting, green and red fluorescent images (representing live and dead cells, respectively) are extracted from the original image and processed separately. A template-matching algorithm is proposed in which pre-defined well and cell templates are matched against the red and green images to locate microwells and cells. Subsequently, local maxima in the correlation maps are determined and local maxima maps are thresholded. At the end, the software records the cell counts for each detected microwell on the original image in high-throughput. The automated counting was shown to be accurate compared with manual counting, with a difference of approximately 1-2 cells per microwell: based on cell concentration, the absolute difference between manual and automatic counting measurements was 2.5-13%., Current yeast interactome network maps contain several hundred molecular complexes with limited and somewhat controversial representation of direct binary interactions. We carried out a comparative quality assessment of current yeast interactome data sets, demonstrating that high-throughput yeast two-hybrid (Y2H) screening provides high-quality binary interaction information. Because a large fraction of the yeast binary interactome remains to be mapped, we developed an empirically controlled mapping framework to produce a &quot;second-generation&quot; high-quality, high-throughput Y2H data set covering approximately 20% of all yeast binary interactions. Both Y2H and affinity purification followed by mass spectrometry (AP/MS) data are of equally high quality but of a fundamentally different and complementary nature, resulting in networks with different topological and biological properties. Compared to co-complex interactome models, this binary map is enriched for transient signaling interactions and intercomplex connections with a highly significant clustering between essential proteins. Rather than correlating with essentiality, protein connectivity correlates with genetic pleiotropy.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Trigram_detail  \\\n",
       "0  (A genetic tool to introduce marker-free deletions is essential for multiple manipulations of genomes. We report a simple and efficient method to create marker-free deletion mutants of Bacillus subtilis through transformation with recombinant PCR products, using the Escherichia coli mazF gene encoding an endoribonuclease that cleaves free mRNAs as a counter-selection tool. Our method will be applicable to any bacterium in which introduction of the mazF cassette into the genome by double crossover homologous recombination is possible., Programmed cell death (PCD) is a genetically-controlled disassembly of the cell. In animal systems, the central core execution switch for apoptotic PCD is the activation of caspases (Cysteine-containing Aspartate-specific proteases). Accumulating evidence in recent years suggests the existence of caspase-like activity in plants and its functional involvement in various types of plant PCD, although no functional homologs of animal caspases were identified in plant genome. In this mini-review, we will cover the recent results on the existence of plant caspase-like proteases and introduce major technologies used in detecting the activation of caspase-like proteases during plant PCD.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "1  (Mitochondrial (mt) heteroplasmy in the control region (CR) of the black-faced spoonbill was investigated using LA-PCR. To avoid amplification of transpositioned nuclear genome fragment from mtDNA (numt), PCR product of the almost-complete mitochondrial genome was amplified using primers designed to anneal on the COIII gene. Then nested LA-PCR product was amplified between the cyt b and 12S rRNA genes using the almost-complete mitochondrial genome PCR product as a template. Nucleotide sequencing revealed tandem duplication composed of two units. The first contains cyt b-1, tRNA(Thr)-1, tRNA(Pro)-1, ND6-1, tRNA(Glu)-1 and CR1, and the second consists of cyt b-2, tRNA(Thr)-2, tRNA(Pro)-2, ND6-2, tRNA(Glu)-2 and CR2, followed by tRNA(Phe) and 12S rRNA. The duplicated cyt b-2 sequence coincided with 499 bp at the 3' end of cyt b-1. With the exception of the CR, the other genes in the duplicated sequence were identical to the original corresponding gene. Even though both CR1 and CR2 contain functional blocks, such as a poly-C site, a goose hairpin and a TAS structure in Domain I, the 3' end of CR1 was followed by a 112 bp sequence (non-coding region) that was not found in CR2 or in sequence homology analysis of similar genes. Meanwhile, CR2 ended in a complicated repeat sequence. The 5' franking region in the Domain I (Region A) and the 3' franking region in the Domain I (Region B) of the two CRs evolve in quite different manners: Region A was highly variable between CR1 and CR2 in the same individuals, while Region B was almost identical between them, which indicates concerted evolution., Plasmodium falciparum is the major human malaria agent responsible for 200 to 300 million infections and one to three million deaths annually, mainly among African infants. The origin and evolution of this pathogen within the human lineage is still unresolved. A single species, P. reichenowi, which infects chimpanzees, is known to be a close sister lineage of P. falciparum. Here we report the discovery of a new Plasmodium species infecting Hominids. This new species has been isolated in two chimpanzees (Pan troglodytes) kept as pets by villagers in Gabon (Africa). Analysis of its complete mitochondrial genome (5529 nucleotides including Cyt b, Cox I and Cox III genes) reveals an older divergence of this lineage from the clade that includes P. falciparum and P. reichenowi (approximately 21+/-9 Myrs ago using Bayesian methods and considering that the divergence between P. falciparum and P. reichenowi occurred 4 to 7 million years ago as generally considered in the literature). This time frame would be congruent with the radiation of hominoids, suggesting that this Plasmodium lineage might have been present in early hominoids and that they may both have experienced a simultaneous diversification. Investigation of the nuclear genome of this new species will further the understanding of the genetic adaptations of P. falciparum to humans. The risk of transfer and emergence of this new species in humans must be now seriously considered given that it was found in two chimpanzees living in contact with humans and its close relatedness to the most virulent agent of malaria.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "2  (In mangrove species the past geomorphic changes in coastal regions and reproductive systems are important factors of their distribution and genetic structure of populations. However, very little is known about the levels of genetic variation of Rhiozophora species in Southeast Asia. In this study, we surveyed levels and patterns of genetic variation as well as population structure of two sympatric mangrove species, Rhizophora apiculata and R. mucronata in Thailand, using five nuclear genes and two cpDNA regions. In all investigated DNA regions, nucleotide variation within species was low, while nucleotide divergence between the two species was considerable. The nuclear genes evolved 10 times faster than the cpDNA regions. In both R. apiculata and R. mucronata, significant positive F(IS) values were found, indicating deviation from Hardy-Weinberg proportions and a deficiency of heterozygotes. In both species, we found significant genetic differentiation between populations. However, the pattern of population differentiation (F(ST)) of R. apiculata differed from that of R. mucronata. Our results suggest that the two investigated species have different demographic history, even though they are sympatric and have similar reproductive systems., Mexico is developing the basis for genomic medicine to improve healthcare of its population. The extensive study of genetic diversity and linkage disequilibrium structure of different populations has made it possible to develop tagging and imputation strategies to comprehensively analyze common genetic variation in association studies of complex diseases. We assessed the benefit of a Mexican haplotype map to improve identification of genes related to common diseases in the Mexican population. We evaluated genetic diversity, linkage disequilibrium patterns, and extent of haplotype sharing using genomewide data from Mexican Mestizos from regions with different histories of admixture and particular population dynamics. Ancestry was evaluated by including 1 Mexican Amerindian group and data from the HapMap. Our results provide evidence of genetic differences between Mexican subpopulations that should be considered in the design and analysis of association studies of complex diseases. In addition, these results support the notion that a haplotype map of the Mexican Mestizo population can reduce the number of tag SNPs required to characterize common genetic variation in this population. This is one of the first genomewide genotyping efforts of a recently admixed population in Latin America.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "3  (The cell envelope is the target for many antibiotics. In Gram-positive bacteria, membrane alterations and dysfunction caused by antibiotics are sensed mainly by two classes of signal transduction systems: the ECF sigma factors and the two-component signal transduction systems (TCSs). Enduracidin is an antibiotic that inhibits the transglycosylation step of peptidoglycan biosynthesis, and is an attractive target for further antibiotic development studies. We assessed transcriptional responses to enduracidin in Bacillus subtilis cells using a high-density tiling chip, and compared the results with responses to bacitracin, which inhibits the lipid II cycle of peptidoglycan synthesis. We exploited the quantitative advantage of the tiling chip to introduce a new criterion, an increase in transcriptional level, in addition to the conventional induction ratio, in order to distinguish genes of biological significance from those with lower induction ratios. Our results indicate that introduction of the new criterion led to unambiguous identification of core transcriptional responses to antibiotics, with a reduction in the number of possible background genes, compared to previous results obtained using gene arrays. We identified 129 genes that were significantly upregulated by enduracidin and/or bacitracin. Notably, we found that inactivation of the LiaRS TCS, which was the system most strongly induced by the two antibiotics, resulted in increased sensitivity to enduracidin, probably through a failure to induce LiaIH proteins. We noted that 33 genes belonging to the SigM regulon were induced by both antibiotics. Consistent with stronger induction of the SigM regulon in enduracidin-treated cells, inactivation of sigM resulted in increased sensitivity to enduracidin. In addition, and for the first time, we found that the Spx regulon was induced in cells challenged by enduracidin and bacitracin, suggesting that thiol-oxidative stress occurred in cells treated with antibiotics. These findings contribute to further our understanding of the molecular nature of genetic systems involved in antibiotic resistance., Two-component signal-transduction systems (TCSs) of bacteria are considered to form an intricate signal network to cope with various environmental stresses. One example of such a network in Escherichia coli is the signal transduction cascade from the EvgS/EvgA system to the PhoQ/PhoP system, where activation of the EvgS/EvgA system promotes expression of PhoP-activated genes. As a factor connecting this signal transduction cascade, we have identified a small inner membrane protein (65 aa), B1500. Expression of the b1500 gene is directly regulated by the EvgS/EvgA system, and b1500 expression from a heterologous promoter simultaneously activated the expression of mgtA and other PhoP regulon genes. This activation was PhoQ/PhoP-dependent and EvgS/EvgA-independent. Furthermore, deletion of b1500 from an EvgS-activated strain suppressed mgtA expression. B1500 is localized in the inner membrane, and bacterial two-hybrid data showed that B1500 formed a complex with the sensor PhoQ. These results indicate that the small membrane protein, B1500, connected the signal transduction between EvgS/EvgA and PhoQ/PhoP systems by directly interacting with PhoQ, thus activating the PhoQ/PhoP system.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "4  (Microscale technologies have emerged as a powerful tool for studying and manipulating biological systems and miniaturizing experiments. However, the lack of software complementing these techniques has made it difficult to apply them for many high-throughput experiments. This work establishes Arraycount, an approach to automatically count cells in microwell arrays. The procedure consists of fluorescent microscope imaging of cells that are seeded in microwells of a microarray system and then analyzing images via computer to recognize the array and count cells inside each microwell. To start counting, green and red fluorescent images (representing live and dead cells, respectively) are extracted from the original image and processed separately. A template-matching algorithm is proposed in which pre-defined well and cell templates are matched against the red and green images to locate microwells and cells. Subsequently, local maxima in the correlation maps are determined and local maxima maps are thresholded. At the end, the software records the cell counts for each detected microwell on the original image in high-throughput. The automated counting was shown to be accurate compared with manual counting, with a difference of approximately 1-2 cells per microwell: based on cell concentration, the absolute difference between manual and automatic counting measurements was 2.5-13%., BACKGROUND: Flow cytometry technology is widely used in both health care and research. The rapid expansion of flow cytometry applications has outpaced the development of data storage and analysis tools. Collaborative efforts being taken to eliminate this gap include building common vocabularies and ontologies, designing generic data models, and defining data exchange formats. The Minimum Information about a Flow Cytometry Experiment (MIFlowCyt) standard was recently adopted by the International Society for Advancement of Cytometry. This standard guides researchers on the information that should be included in peer reviewed publications, but it is insufficient for data exchange and integration between computational systems. The Functional Genomics Experiment (FuGE) formalizes common aspects of comprehensive and high throughput experiments across different biological technologies. We have extended FuGE object model to accommodate flow cytometry data and metadata. METHODS: We used the MagicDraw modelling tool to design a UML model (Flow-OM) according to the FuGE extension guidelines and the AndroMDA toolkit to transform the model to a markup language (Flow-ML). We mapped each MIFlowCyt term to either an existing FuGE class or to a new FuGEFlow class. The development environment was validated by comparing the official FuGE XSD to the schema we generated from the FuGE object model using our configuration. After the Flow-OM model was completed, the final version of the Flow-ML was generated and validated against an example MIFlowCyt compliant experiment description. RESULTS: The extension of FuGE for flow cytometry has resulted in a generic FuGE-compliant data model (FuGEFlow), which accommodates and links together all information required by MIFlowCyt. The FuGEFlow model can be used to build software and databases using FuGE software toolkits to facilitate automated exchange and manipulation of potentially large flow cytometry experimental data sets. Additional project documentation, including reusable design patterns and a guide for setting up a development environment, was contributed back to the FuGE project. CONCLUSION: We have shown that an extension of FuGE can be used to transform minimum information requirements in natural language to markup language in XML. Extending FuGE required significant effort, but in our experiences the benefits outweighed the costs. The FuGEFlow is expected to play a central role in describing flow cytometry experiments and ultimately facilitating data exchange including public flow cytometry repositories currently under development.)   \n",
       "\n",
       "  data_set task_type  \n",
       "0  BC3ACT   CLS       \n",
       "1  BC3ACT   CLS       \n",
       "2  BC3ACT   CLS       \n",
       "3  BC3ACT   CLS       \n",
       "4  BC3ACT   CLS       "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc3_act.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "bc3_act_test_df = bc3actrun.load(bc3_act_test_file)\n",
    "df_bc3_act[\"test_label\"] =bc3actrun.get_labels(bc3_act_test_df, bc3_act_test_eval_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert bc3_act_test_df[\"abstract\"].tolist()==df_bc3_act[\"Unigram_detail\"].apply(lambda x: x[0]).tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>test_label</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">0</th>\n",
       "      <th>count</th>\n",
       "      <td>5090.000000</td>\n",
       "      <td>5090.000000</td>\n",
       "      <td>5090.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>26.310835</td>\n",
       "      <td>6.696011</td>\n",
       "      <td>1.725549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>9.248763</td>\n",
       "      <td>5.347394</td>\n",
       "      <td>1.716761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>6.277610</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>19.703812</td>\n",
       "      <td>3.289165</td>\n",
       "      <td>0.787847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>25.162518</td>\n",
       "      <td>5.073287</td>\n",
       "      <td>1.389157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>31.525002</td>\n",
       "      <td>8.285161</td>\n",
       "      <td>2.268703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>75.011317</td>\n",
       "      <td>41.754463</td>\n",
       "      <td>18.708287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">1</th>\n",
       "      <th>count</th>\n",
       "      <td>910.000000</td>\n",
       "      <td>910.000000</td>\n",
       "      <td>910.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>29.274148</td>\n",
       "      <td>8.092841</td>\n",
       "      <td>2.263471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>9.356049</td>\n",
       "      <td>6.003147</td>\n",
       "      <td>1.725236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>11.140520</td>\n",
       "      <td>1.515848</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>22.694002</td>\n",
       "      <td>4.506764</td>\n",
       "      <td>1.172250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>28.306547</td>\n",
       "      <td>6.254243</td>\n",
       "      <td>1.879282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>34.315445</td>\n",
       "      <td>9.384924</td>\n",
       "      <td>2.837078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>74.011540</td>\n",
       "      <td>51.197969</td>\n",
       "      <td>18.973666</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Unigram       Bigram      Trigram\n",
       "test_label                                             \n",
       "0          count  5090.000000  5090.000000  5090.000000\n",
       "           mean   26.310835    6.696011     1.725549   \n",
       "           std    9.248763     5.347394     1.716761   \n",
       "           min    6.277610     0.000000     0.000000   \n",
       "           25%    19.703812    3.289165     0.787847   \n",
       "           50%    25.162518    5.073287     1.389157   \n",
       "           75%    31.525002    8.285161     2.268703   \n",
       "           max    75.011317    41.754463    18.708287  \n",
       "1          count  910.000000   910.000000   910.000000 \n",
       "           mean   29.274148    8.092841     2.263471   \n",
       "           std    9.356049     6.003147     1.725236   \n",
       "           min    11.140520    1.515848     0.000000   \n",
       "           25%    22.694002    4.506764     1.172250   \n",
       "           50%    28.306547    6.254243     1.879282   \n",
       "           75%    34.315445    9.384924     2.837078   \n",
       "           max    74.011540    51.197969    18.973666  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bc3_act.groupby(\"test_label\").describe().stack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{llrrr}\n",
      "\\toprule\n",
      "  &     &  Unigram &  Bigram &  Trigram \\\\\n",
      "test\\_label & {} &          &         &          \\\\\n",
      "\\midrule\n",
      "0 & count & 5090.00 & 5090.00 & 5090.00 \\\\\n",
      "  & mean & 26.31 & 6.70 & 1.73 \\\\\n",
      "  & std & 9.25 & 5.35 & 1.72 \\\\\n",
      "  & min & 6.28 & 0.00 & 0.00 \\\\\n",
      "  & 25\\% & 19.70 & 3.29 & 0.79 \\\\\n",
      "  & 50\\% & 25.16 & 5.07 & 1.39 \\\\\n",
      "  & 75\\% & 31.53 & 8.29 & 2.27 \\\\\n",
      "  & max & 75.01 & 41.75 & 18.71 \\\\\n",
      "1 & count & 910.00 & 910.00 & 910.00 \\\\\n",
      "  & mean & 29.27 & 8.09 & 2.26 \\\\\n",
      "  & std & 9.36 & 6.00 & 1.73 \\\\\n",
      "  & min & 11.14 & 1.52 & 0.00 \\\\\n",
      "  & 25\\% & 22.69 & 4.51 & 1.17 \\\\\n",
      "  & 50\\% & 28.31 & 6.25 & 1.88 \\\\\n",
      "  & 75\\% & 34.32 & 9.38 & 2.84 \\\\\n",
      "  & max & 74.01 & 51.20 & 18.97 \\\\\n",
      "\\bottomrule\n",
      "\\end{tabular}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print( df_bc3_act.groupby(\"test_label\").describe().stack().\n",
    "      to_latex(float_format=lambda x: \"{:.2f}\".format(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "bc3_act_train_df = bc3actrun.load(bc3_act_train_file)\n",
    "bc3_act_train_df[\"train_labels\"] = bc3actrun.get_labels(bc3_act_train_df, bc3_act_train_eval_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>train_labels</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1140</td>\n",
       "      <td>1140</td>\n",
       "      <td>1140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1140</td>\n",
       "      <td>1140</td>\n",
       "      <td>1140</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              title  abstract    id\n",
       "train_labels                       \n",
       "0             1140   1140      1140\n",
       "1             1140   1140      1140"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bc3_act_train_df.groupby([\"train_labels\"]).count()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Chemu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "chemu_train_dir = os.path.join(\"tmp\", \"chemu\", \"train\")\n",
    "chemu_test_dir = os.path.join(\"tmp\", \"chemu\", \"test\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "from chemu_gene_mention import ChemuGeneMention"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "chemurun = ChemuGeneMention()\n",
    "result_score, result_detail = chemurun.run_similarity_comparer( chemu_train_dir,  chemu_test_dir, \"text\")\n",
    "df_chemu_text = scores_to_df(result_score,result_detail, \"Chemu\", \"text\", \"NER\")\n",
    "\n",
    "result_score, result_detail = chemurun.run_similarity_comparer( chemu_train_dir,  chemu_test_dir, \"entity\")\n",
    "df_chemu_eval = scores_to_df(result_score,result_detail, \"Chemu\", \"anno\", \"NER\")\n",
    "\n",
    "df_chemu = pd.concat([df_chemu_text, df_chemu_eval])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "      <th>Unigram_detail</th>\n",
       "      <th>Bigram_detail</th>\n",
       "      <th>Trigram_detail</th>\n",
       "      <th>data_set</th>\n",
       "      <th>task_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>88.160570</td>\n",
       "      <td>73.775566</td>\n",
       "      <td>66.167284</td>\n",
       "      <td>(Example 55\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopentyl-4-methyl-1,3-thiazole-5-carboxylic acid (0.095 g, 0.448 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed tube was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopentyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)thiazole-5-carboxamide (0.143 g, 80%) as a beige solid., Example 53\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopropyl-4-methyl-1,3-oxazole-5-carboxylic acid (0.072 g, 0.428 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopropyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)oxazole-5-carboxamide (0.143 g, 89%) as a beige solid.)</td>\n",
       "      <td>(Example 55\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopentyl-4-methyl-1,3-thiazole-5-carboxylic acid (0.095 g, 0.448 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed tube was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopentyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)thiazole-5-carboxamide (0.143 g, 80%) as a beige solid., Example 53\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopropyl-4-methyl-1,3-oxazole-5-carboxylic acid (0.072 g, 0.428 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopropyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)oxazole-5-carboxamide (0.143 g, 89%) as a beige solid.)</td>\n",
       "      <td>(Example 55\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopentyl-4-methyl-1,3-thiazole-5-carboxylic acid (0.095 g, 0.448 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed tube was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopentyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)thiazole-5-carboxamide (0.143 g, 80%) as a beige solid., Example 53\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopropyl-4-methyl-1,3-oxazole-5-carboxylic acid (0.072 g, 0.428 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopropyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)oxazole-5-carboxamide (0.143 g, 89%) as a beige solid.)</td>\n",
       "      <td>Chemu text</td>\n",
       "      <td>NER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>59.438230</td>\n",
       "      <td>33.936262</td>\n",
       "      <td>19.367281</td>\n",
       "      <td>(Step 9: tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate\\nTert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-hydroxyltetrahydro-2H-pyran-3-ylcarbamate (2.33 g, 7.08 mmol) was dissolved in a mixed solution of 24 mL acetonitrile, 4 mL water and 4 mL acetic acid. Thereto was added an aqueous solution (4 mL) of ruthenium chloride hydrate (3.7 mg, 0.0142 mmol), and cooled to 0 °C. Sodium bromate (535 mg, 3.54 mmol) was added, and stirred at low temperature for about 1.5 hours until the raw materials were completely reacted. To the reaction solution was added 120 mL water, stirred at 0 °C overnight, and extracted with dichloromethane. The organic phase was washed with water, dried and concentrated, and then the residue was purified by silica gel column chromatography (petroleum ether/ethyl acetate, 10:1) to give the intermediate 1 tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate (1.71 g) as a white solid in 74% yield., Example 1: (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine -1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-amine\\nStep1: tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2] thiazine-1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate\\nTo N,N-dimethylacetamide (2 mL) were added tert-butyl (2R,3S)-2-(2,5-di-fluorophenyl) -5-oxotetrahydro-2H-pyran-3-ylcarbamate (48 mg, 0.146 mmol), 2,3,4,5,6,7-hexahydropyrrolo [3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide p-toluenesulfonate (48 mg, 0.13 mmol) and triethylamine (10 mg, 0.1 mmol), and stirred at room temperature for 3 hours. After cooling in an ice bath, sodium triacetoxyborohydride (87 mg, 0.39 mmol) was added, slowly warmed to room temperature and stirred overnight. Saturated sodium bicarbonate solution was added, and the resulting mixture was extracted with dichloromethane, washed with saturated brine, dried, concentrated and then purified by silica gel column chromatography to give tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide-6-yl) -2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate (48 mg). Yield: 71%.)</td>\n",
       "      <td>(Step 9: tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate\\nTert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-hydroxyltetrahydro-2H-pyran-3-ylcarbamate (2.33 g, 7.08 mmol) was dissolved in a mixed solution of 24 mL acetonitrile, 4 mL water and 4 mL acetic acid. Thereto was added an aqueous solution (4 mL) of ruthenium chloride hydrate (3.7 mg, 0.0142 mmol), and cooled to 0 °C. Sodium bromate (535 mg, 3.54 mmol) was added, and stirred at low temperature for about 1.5 hours until the raw materials were completely reacted. To the reaction solution was added 120 mL water, stirred at 0 °C overnight, and extracted with dichloromethane. The organic phase was washed with water, dried and concentrated, and then the residue was purified by silica gel column chromatography (petroleum ether/ethyl acetate, 10:1) to give the intermediate 1 tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate (1.71 g) as a white solid in 74% yield., Example 1: (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine -1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-amine\\nStep1: tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2] thiazine-1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate\\nTo N,N-dimethylacetamide (2 mL) were added tert-butyl (2R,3S)-2-(2,5-di-fluorophenyl) -5-oxotetrahydro-2H-pyran-3-ylcarbamate (48 mg, 0.146 mmol), 2,3,4,5,6,7-hexahydropyrrolo [3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide p-toluenesulfonate (48 mg, 0.13 mmol) and triethylamine (10 mg, 0.1 mmol), and stirred at room temperature for 3 hours. After cooling in an ice bath, sodium triacetoxyborohydride (87 mg, 0.39 mmol) was added, slowly warmed to room temperature and stirred overnight. Saturated sodium bicarbonate solution was added, and the resulting mixture was extracted with dichloromethane, washed with saturated brine, dried, concentrated and then purified by silica gel column chromatography to give tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide-6-yl) -2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate (48 mg). Yield: 71%.)</td>\n",
       "      <td>(Step 9: tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate\\nTert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-hydroxyltetrahydro-2H-pyran-3-ylcarbamate (2.33 g, 7.08 mmol) was dissolved in a mixed solution of 24 mL acetonitrile, 4 mL water and 4 mL acetic acid. Thereto was added an aqueous solution (4 mL) of ruthenium chloride hydrate (3.7 mg, 0.0142 mmol), and cooled to 0 °C. Sodium bromate (535 mg, 3.54 mmol) was added, and stirred at low temperature for about 1.5 hours until the raw materials were completely reacted. To the reaction solution was added 120 mL water, stirred at 0 °C overnight, and extracted with dichloromethane. The organic phase was washed with water, dried and concentrated, and then the residue was purified by silica gel column chromatography (petroleum ether/ethyl acetate, 10:1) to give the intermediate 1 tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate (1.71 g) as a white solid in 74% yield., Example 1: (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine -1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-amine\\nStep1: tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2] thiazine-1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate\\nTo N,N-dimethylacetamide (2 mL) were added tert-butyl (2R,3S)-2-(2,5-di-fluorophenyl) -5-oxotetrahydro-2H-pyran-3-ylcarbamate (48 mg, 0.146 mmol), 2,3,4,5,6,7-hexahydropyrrolo [3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide p-toluenesulfonate (48 mg, 0.13 mmol) and triethylamine (10 mg, 0.1 mmol), and stirred at room temperature for 3 hours. After cooling in an ice bath, sodium triacetoxyborohydride (87 mg, 0.39 mmol) was added, slowly warmed to room temperature and stirred overnight. Saturated sodium bicarbonate solution was added, and the resulting mixture was extracted with dichloromethane, washed with saturated brine, dried, concentrated and then purified by silica gel column chromatography to give tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide-6-yl) -2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate (48 mg). Yield: 71%.)</td>\n",
       "      <td>Chemu text</td>\n",
       "      <td>NER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>79.425707</td>\n",
       "      <td>52.160029</td>\n",
       "      <td>46.756433</td>\n",
       "      <td>(Example 9\\n5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine\\nDMF (1 mL) and DIPEA (0.022 ml, 0.13 mmol) were added sequentially to a flask charged with N-(azetidin-3-yl)-5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-4-methoxypyridin-2-amine (22 mg, 0.050 mmol). 1-Fluoro-3-iodopropane (9 mg, 0.05 mmol) in DMF (0.1 mL) was then added. After 2 hours, the reaction was diluted with saturated aqueous sodium chloride, and the mixture was extracted in EtOAC (3×). The combined organic layers were washed with water and dried over sodium sulfate, filtered and concentrated under reduced pressure. The resulting crude film was purified by flash silica chromatography, elution gradient 2 to 10% MeOH in DCM, to give 5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine (9.0 mg, 36%) as a dry film., Example 19\\nPreparation of N-(4-((6S,8R)-7-(2,2-difluoropropyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine\\nDMF (2 mL) and DIPEA (0.074 mL, 0.42 mmol) were added sequentially to a flask charged with N-(4-((6S,8R)-7-(2,2-difluoropropyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)azetidin-3-amine (75 mg, 0.17 mmol). 1-Fluoro-3-iodopropane (31.9 mg, 0.17 mmol) in DMF (0.1 mL) was then added, and stirring was continued for 2 hours. The reaction was stopped, diluted with saturated aqueous sodium chloride and the compound was extracted in EtOAC (×3). The combined extracts were washed with water and dried over sodium sulfate, filtered and concentrated under reduced pressure to afford a film. This material was purified by flash silica chromatography, eluting with 2 to 10% (methanol containing 1% ammonium hydroxide) in DCM to afford N-(4-((6S,8R)-7-(2,2-difluoropropyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine (43 mg, 51%).)</td>\n",
       "      <td>(Example 9\\n5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine\\nDMF (1 mL) and DIPEA (0.022 ml, 0.13 mmol) were added sequentially to a flask charged with N-(azetidin-3-yl)-5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-4-methoxypyridin-2-amine (22 mg, 0.050 mmol). 1-Fluoro-3-iodopropane (9 mg, 0.05 mmol) in DMF (0.1 mL) was then added. After 2 hours, the reaction was diluted with saturated aqueous sodium chloride, and the mixture was extracted in EtOAC (3×). The combined organic layers were washed with water and dried over sodium sulfate, filtered and concentrated under reduced pressure. The resulting crude film was purified by flash silica chromatography, elution gradient 2 to 10% MeOH in DCM, to give 5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine (9.0 mg, 36%) as a dry film., Example 6\\nN-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine\\n1-Fluoro-3-iodopropane (40.9 μL, 0.39 mmol) was added to a solution of N-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)azetidin-3-amine (169 mg, 0.39 mmol) and DIPEA (203 μL, 1.16 mmol) in NMP (1.7 mL) at room temperature. After 18 hours, the reaction was concentrated under reduced pressure. The resulting residue was purified by reverse phase flash chromatography (C18), eluting with decreasingly polar mixtures of water (containing 0.2% ammonium hydroxide) and MeCN as eluents. Product fractions were combined and lyopholized to afford N-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine (75 mg, 39%) as clear residue.)</td>\n",
       "      <td>(Example 9\\n5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine\\nDMF (1 mL) and DIPEA (0.022 ml, 0.13 mmol) were added sequentially to a flask charged with N-(azetidin-3-yl)-5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-4-methoxypyridin-2-amine (22 mg, 0.050 mmol). 1-Fluoro-3-iodopropane (9 mg, 0.05 mmol) in DMF (0.1 mL) was then added. After 2 hours, the reaction was diluted with saturated aqueous sodium chloride, and the mixture was extracted in EtOAC (3×). The combined organic layers were washed with water and dried over sodium sulfate, filtered and concentrated under reduced pressure. The resulting crude film was purified by flash silica chromatography, elution gradient 2 to 10% MeOH in DCM, to give 5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine (9.0 mg, 36%) as a dry film., Example 6\\nN-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine\\n1-Fluoro-3-iodopropane (40.9 μL, 0.39 mmol) was added to a solution of N-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)azetidin-3-amine (169 mg, 0.39 mmol) and DIPEA (203 μL, 1.16 mmol) in NMP (1.7 mL) at room temperature. After 18 hours, the reaction was concentrated under reduced pressure. The resulting residue was purified by reverse phase flash chromatography (C18), eluting with decreasingly polar mixtures of water (containing 0.2% ammonium hydroxide) and MeCN as eluents. Product fractions were combined and lyopholized to afford N-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine (75 mg, 39%) as clear residue.)</td>\n",
       "      <td>Chemu text</td>\n",
       "      <td>NER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>50.937163</td>\n",
       "      <td>13.844644</td>\n",
       "      <td>7.317617</td>\n",
       "      <td>(3,5-Dichloro-4-(5-fluoro-2-methyl-4-oxo-1,7-naphthyridin-1 (4H)-yl)benzonitrile (Intermediate 12)\\nStep 1: 3,5-Difluoro-N-methoxy-N-methylisonicotinamide\\n3,5-difluoroisonicotinic acid (18.78 g, 118 mmol), N,O-dimethylhydroxylamine hydrochloride (12.09 g, 124 mmol), HATU (47.1 g, 124 mmol) and DIPEA (61.9 ml, 354 mmol) were suspened in DCM (236 ml). The mixture was allowed to stir at room temperature for 24 hours. The mixture was concentrated under reduced pressure. The residue was extracted with EtOAc washing with saturated aqueous NH4Cl solution (2×), water, saturated NaHCO3 and brine. The organic layer was dried over sodium sulfate and concentrated. The crude was purified by chromatography on silica gel using 0-75% EtOAc in heptanes as eluent to give the title compound (18.9 g, 79% yield)., The second step\\nTo a solution of Compound 36 (1.14g, 1.97mmol) in DMF (12 mL) were added cesium carbonate (1.29 g, 3.95 mmol) and 1,2-dibromoethane (0.255 mL, 2.96 mmol), and the mixture was stirred at room temperature for 18.5 hours. The reaction mixture was extracted with ethyl acetate after addition of water. The organic layer was washed with water and saturated brine, dried over anhydrous magnesium sulfate, and concentrated under reduced pressure. The resulting residue was purified by silica gel chromatography (hexane-ethyl acetate) to yield Compound 37 (1.13g, yield 95%).)</td>\n",
       "      <td>(3,5-Dichloro-4-(5-fluoro-2-methyl-4-oxo-1,7-naphthyridin-1 (4H)-yl)benzonitrile (Intermediate 12)\\nStep 1: 3,5-Difluoro-N-methoxy-N-methylisonicotinamide\\n3,5-difluoroisonicotinic acid (18.78 g, 118 mmol), N,O-dimethylhydroxylamine hydrochloride (12.09 g, 124 mmol), HATU (47.1 g, 124 mmol) and DIPEA (61.9 ml, 354 mmol) were suspened in DCM (236 ml). The mixture was allowed to stir at room temperature for 24 hours. The mixture was concentrated under reduced pressure. The residue was extracted with EtOAc washing with saturated aqueous NH4Cl solution (2×), water, saturated NaHCO3 and brine. The organic layer was dried over sodium sulfate and concentrated. The crude was purified by chromatography on silica gel using 0-75% EtOAc in heptanes as eluent to give the title compound (18.9 g, 79% yield)., (Example 12)\\nThe first step\\nTo a solution of Compound 58 (850mg, 1.91mmol) and 3-bromoprop-1-yne (455mg, 3.82mmol) in DMF (5mL) was added 60% sodium hydride (153mg, 3.82mmol) under ice-cooling, and the mixture was stirred for 1 hour at 0 °C. After the reaction mixture was diluted with ethyl acetate, water was added thereto, and the mixture was extracted with ethyl acetate. The organic layer was washed with water, dried over anhydrous sodium sulfate, and concentrated under reduced pressure. The resulting residue was dissolved in ethanol (20 mL) and water (3 mL), ammonium chloride (1.02 g, 19.1 mmol) and iron powder (1.07 g, 19.1 mmol) was added thereto, and the mixture was stirred for 4 hours at 90 °C. After the insoluble materials were removed by filtration, and the filtrate was concentrated under reduced pressure. The resulting residue was extracted with chloroform after addition of chloroform and water. The organic layer was dried over anhydrous sodium sulfate, and concentrated under reduced pressure. The resulting residue was purified by silica gel chromatography (hexane-ethyl acetate) to yield Compound 59 (251 mg, yield 29%).)</td>\n",
       "      <td>(3,5-Dichloro-4-(5-fluoro-2-methyl-4-oxo-1,7-naphthyridin-1 (4H)-yl)benzonitrile (Intermediate 12)\\nStep 1: 3,5-Difluoro-N-methoxy-N-methylisonicotinamide\\n3,5-difluoroisonicotinic acid (18.78 g, 118 mmol), N,O-dimethylhydroxylamine hydrochloride (12.09 g, 124 mmol), HATU (47.1 g, 124 mmol) and DIPEA (61.9 ml, 354 mmol) were suspened in DCM (236 ml). The mixture was allowed to stir at room temperature for 24 hours. The mixture was concentrated under reduced pressure. The residue was extracted with EtOAc washing with saturated aqueous NH4Cl solution (2×), water, saturated NaHCO3 and brine. The organic layer was dried over sodium sulfate and concentrated. The crude was purified by chromatography on silica gel using 0-75% EtOAc in heptanes as eluent to give the title compound (18.9 g, 79% yield)., Synthesis of Compound 13\\nTo a mixture of compound 12 (0.45 g, 4.2 mol) and Et3N (8 mL, 0.06 mol) in THF (15 mL), a solution of 2-(tert-Butoxycarbonyloxyimino)-2-phenylacetonitrile (2.1 g, 8.3 mol) in THF (30 mL) was added dropwise at 0° C. Following complete addition, the solution was allowed to warm to room temperature and left to stir for 4 hours. The reaction mixture was concentrated to oil under reduced pressure and CH2Cl2 (50 mL) was added. The mixture was washed with sodium hydroxide (5%, 30 mL) and brine (30 mL). The organic layer was dried over sodium sulfate and concentrated under reduced pressure. The crude product was purified by column chromatography (silica gel, MeOH:CH2Cl2=1:10, v/v) to give compound 13 as a yellow oil (803.1 mg, 60.7%). LC-MS m/z (ES+), 304.22 (M+H)+.)</td>\n",
       "      <td>Chemu text</td>\n",
       "      <td>NER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>57.138872</td>\n",
       "      <td>18.352924</td>\n",
       "      <td>5.902813</td>\n",
       "      <td>(Example 73\\nPreparation of 2-chloro-3-fluoro-5-nitrobenzoic acid (C206)\\nTo a suspension of 2-chloro-3-fluorobenzoic acid (10 g, 57.5 mmol) in concentrated sulfuric acid (62 mL, 1149.5 mmol) was added concentrated nitric acid (4 mL, 86.2 mmol) dropwise at −10° C. and the reaction mixture was stirred between −10° C. and 0° C. for 3 hours. The reaction mixture was slowly poured into a beaker of crushed ice (˜1 L), and the resulting precipitated solid was filtered and washed with water (100 mL). The crude product was recrystallized from hot water to afford the title compound as an off-white solid (7.5 g, 60%): mp 163-165° C.;, Step D: 7-Chloro-6-(difluoromethyl)-4-methyl-2-(trifluoromethyl)pyrazolo[1,5-c]pyridine-3-carboxylic acid. To a solution of ethyl 7-chloro-6-(difluoromethyl)-4-methyl-2-(trifluoromethyl)pyrazolo[1,5-a]pyridine-3-carboxylate (2.6 g, 7.29 mmol) in a mixture of 1,4-dioxane (30 mL) and water (6 mL) was added lithium hydroxide monohydrate (459 mg, 10.9 mmol) and the reaction mixture was stirred at 60° C. for 4 h. To the reaction mixture was added lithium hydroxide monohydrate (459 mg, 10.9 mmol) and the reaction mixture was stirred at 60° C. for 18 h. To the reaction mixture was added lithium hydroxide monohydrate (459 mg, 10.9 mmol) and the reaction mixture was stirred at 60° C. for 2 h. To the reaction mixture was added a fourth portion of lithium hydroxide monohydrate (459 mg, 10.9 mmol) and the reaction mixture was stirred at 60° C. for 2 h. The reaction mixture was concentrated to 6 mL under vacuum. The residue was diluted with water (20 mL) and the mixture was washed with chloroform (1×10 mL). The aqueous layer was acidified to pH 4 by addition of 1 M hydrochloric acid. The precipitate was collected and the solid was washed with water (2×20 mL) to give the title compound (1.80 g, 5.48 mmol, 75%) as a white crystalline solid.)</td>\n",
       "      <td>(Example 73\\nPreparation of 2-chloro-3-fluoro-5-nitrobenzoic acid (C206)\\nTo a suspension of 2-chloro-3-fluorobenzoic acid (10 g, 57.5 mmol) in concentrated sulfuric acid (62 mL, 1149.5 mmol) was added concentrated nitric acid (4 mL, 86.2 mmol) dropwise at −10° C. and the reaction mixture was stirred between −10° C. and 0° C. for 3 hours. The reaction mixture was slowly poured into a beaker of crushed ice (˜1 L), and the resulting precipitated solid was filtered and washed with water (100 mL). The crude product was recrystallized from hot water to afford the title compound as an off-white solid (7.5 g, 60%): mp 163-165° C.;, Synthesis of Compound A-1\\nA solution of N-phenyl-[1,1'-biphenyl]-2-amine (0.64 g, 2.61 mmol) in 1,2-dichlorobenzene (5 mL) was bubbled with nitrogen gas for 30 mins, boron tribromide (1M in dichloromethane, 4 mL, 4.00 mmol) was added and the reaction mixture was heated to 180° C. for 3 hours. After cooling the reaction mixture to room temperature (22° C.), diphenylzine (1.27 g, 5.78 mmol) was added to the reaction mixture inside the glovebox. The reaction mixture was heated to 120° C. for 6 hours. The reaction mixture was cooled to room temperature (22° C.), water was added and the reaction mixture was extracted with diethyl ether (×3). The combined organic layer was washed with water and brine solution in this sequence, dried over MgSO4 and concentrated in vacuo. The residue was purified by flash column chromatography with dichloromethane-hexane (1:9 to 2:8) to give Compound A-1 (0.68 g, 80% yield) as white solid.)</td>\n",
       "      <td>(Example 73\\nPreparation of 2-chloro-3-fluoro-5-nitrobenzoic acid (C206)\\nTo a suspension of 2-chloro-3-fluorobenzoic acid (10 g, 57.5 mmol) in concentrated sulfuric acid (62 mL, 1149.5 mmol) was added concentrated nitric acid (4 mL, 86.2 mmol) dropwise at −10° C. and the reaction mixture was stirred between −10° C. and 0° C. for 3 hours. The reaction mixture was slowly poured into a beaker of crushed ice (˜1 L), and the resulting precipitated solid was filtered and washed with water (100 mL). The crude product was recrystallized from hot water to afford the title compound as an off-white solid (7.5 g, 60%): mp 163-165° C.;, Intermediate 1: 5-Bromo-7-chloroindolin-2-one\\nTo a cooled (0° C.) solution of 7-chloroindolin-2-one (1.0 g, 6.0 mmol) in TFA (11 mL) was added N-bromosuccinimide (1.0 g, 6.0 mmol) in portions. The reaction mixture was stirred at 0° C. for 6 h. The solvent was removed in vacuo and the residue was diluted with DCM (25 mL) and concentrated, followed by a similar sequence with EtOAc. The crude product was triturated with ethanol to provide the title compound as a white solid (861 mg, 58% yield).)</td>\n",
       "      <td>Chemu text</td>\n",
       "      <td>NER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unigram     Bigram    Trigram  \\\n",
       "0  88.160570  73.775566  66.167284   \n",
       "1  59.438230  33.936262  19.367281   \n",
       "2  79.425707  52.160029  46.756433   \n",
       "3  50.937163  13.844644  7.317617    \n",
       "4  57.138872  18.352924  5.902813    \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Unigram_detail  \\\n",
       "0  (Example 55\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopentyl-4-methyl-1,3-thiazole-5-carboxylic acid (0.095 g, 0.448 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed tube was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopentyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)thiazole-5-carboxamide (0.143 g, 80%) as a beige solid., Example 53\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopropyl-4-methyl-1,3-oxazole-5-carboxylic acid (0.072 g, 0.428 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopropyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)oxazole-5-carboxamide (0.143 g, 89%) as a beige solid.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "1  (Step 9: tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate\\nTert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-hydroxyltetrahydro-2H-pyran-3-ylcarbamate (2.33 g, 7.08 mmol) was dissolved in a mixed solution of 24 mL acetonitrile, 4 mL water and 4 mL acetic acid. Thereto was added an aqueous solution (4 mL) of ruthenium chloride hydrate (3.7 mg, 0.0142 mmol), and cooled to 0 °C. Sodium bromate (535 mg, 3.54 mmol) was added, and stirred at low temperature for about 1.5 hours until the raw materials were completely reacted. To the reaction solution was added 120 mL water, stirred at 0 °C overnight, and extracted with dichloromethane. The organic phase was washed with water, dried and concentrated, and then the residue was purified by silica gel column chromatography (petroleum ether/ethyl acetate, 10:1) to give the intermediate 1 tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate (1.71 g) as a white solid in 74% yield., Example 1: (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine -1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-amine\\nStep1: tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2] thiazine-1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate\\nTo N,N-dimethylacetamide (2 mL) were added tert-butyl (2R,3S)-2-(2,5-di-fluorophenyl) -5-oxotetrahydro-2H-pyran-3-ylcarbamate (48 mg, 0.146 mmol), 2,3,4,5,6,7-hexahydropyrrolo [3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide p-toluenesulfonate (48 mg, 0.13 mmol) and triethylamine (10 mg, 0.1 mmol), and stirred at room temperature for 3 hours. After cooling in an ice bath, sodium triacetoxyborohydride (87 mg, 0.39 mmol) was added, slowly warmed to room temperature and stirred overnight. Saturated sodium bicarbonate solution was added, and the resulting mixture was extracted with dichloromethane, washed with saturated brine, dried, concentrated and then purified by silica gel column chromatography to give tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide-6-yl) -2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate (48 mg). Yield: 71%.)   \n",
       "2  (Example 9\\n5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine\\nDMF (1 mL) and DIPEA (0.022 ml, 0.13 mmol) were added sequentially to a flask charged with N-(azetidin-3-yl)-5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-4-methoxypyridin-2-amine (22 mg, 0.050 mmol). 1-Fluoro-3-iodopropane (9 mg, 0.05 mmol) in DMF (0.1 mL) was then added. After 2 hours, the reaction was diluted with saturated aqueous sodium chloride, and the mixture was extracted in EtOAC (3×). The combined organic layers were washed with water and dried over sodium sulfate, filtered and concentrated under reduced pressure. The resulting crude film was purified by flash silica chromatography, elution gradient 2 to 10% MeOH in DCM, to give 5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine (9.0 mg, 36%) as a dry film., Example 19\\nPreparation of N-(4-((6S,8R)-7-(2,2-difluoropropyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine\\nDMF (2 mL) and DIPEA (0.074 mL, 0.42 mmol) were added sequentially to a flask charged with N-(4-((6S,8R)-7-(2,2-difluoropropyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)azetidin-3-amine (75 mg, 0.17 mmol). 1-Fluoro-3-iodopropane (31.9 mg, 0.17 mmol) in DMF (0.1 mL) was then added, and stirring was continued for 2 hours. The reaction was stopped, diluted with saturated aqueous sodium chloride and the compound was extracted in EtOAC (×3). The combined extracts were washed with water and dried over sodium sulfate, filtered and concentrated under reduced pressure to afford a film. This material was purified by flash silica chromatography, eluting with 2 to 10% (methanol containing 1% ammonium hydroxide) in DCM to afford N-(4-((6S,8R)-7-(2,2-difluoropropyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine (43 mg, 51%).)                               \n",
       "3  (3,5-Dichloro-4-(5-fluoro-2-methyl-4-oxo-1,7-naphthyridin-1 (4H)-yl)benzonitrile (Intermediate 12)\\nStep 1: 3,5-Difluoro-N-methoxy-N-methylisonicotinamide\\n3,5-difluoroisonicotinic acid (18.78 g, 118 mmol), N,O-dimethylhydroxylamine hydrochloride (12.09 g, 124 mmol), HATU (47.1 g, 124 mmol) and DIPEA (61.9 ml, 354 mmol) were suspened in DCM (236 ml). The mixture was allowed to stir at room temperature for 24 hours. The mixture was concentrated under reduced pressure. The residue was extracted with EtOAc washing with saturated aqueous NH4Cl solution (2×), water, saturated NaHCO3 and brine. The organic layer was dried over sodium sulfate and concentrated. The crude was purified by chromatography on silica gel using 0-75% EtOAc in heptanes as eluent to give the title compound (18.9 g, 79% yield)., The second step\\nTo a solution of Compound 36 (1.14g, 1.97mmol) in DMF (12 mL) were added cesium carbonate (1.29 g, 3.95 mmol) and 1,2-dibromoethane (0.255 mL, 2.96 mmol), and the mixture was stirred at room temperature for 18.5 hours. The reaction mixture was extracted with ethyl acetate after addition of water. The organic layer was washed with water and saturated brine, dried over anhydrous magnesium sulfate, and concentrated under reduced pressure. The resulting residue was purified by silica gel chromatography (hexane-ethyl acetate) to yield Compound 37 (1.13g, yield 95%).)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "4  (Example 73\\nPreparation of 2-chloro-3-fluoro-5-nitrobenzoic acid (C206)\\nTo a suspension of 2-chloro-3-fluorobenzoic acid (10 g, 57.5 mmol) in concentrated sulfuric acid (62 mL, 1149.5 mmol) was added concentrated nitric acid (4 mL, 86.2 mmol) dropwise at −10° C. and the reaction mixture was stirred between −10° C. and 0° C. for 3 hours. The reaction mixture was slowly poured into a beaker of crushed ice (˜1 L), and the resulting precipitated solid was filtered and washed with water (100 mL). The crude product was recrystallized from hot water to afford the title compound as an off-white solid (7.5 g, 60%): mp 163-165° C.;, Step D: 7-Chloro-6-(difluoromethyl)-4-methyl-2-(trifluoromethyl)pyrazolo[1,5-c]pyridine-3-carboxylic acid. To a solution of ethyl 7-chloro-6-(difluoromethyl)-4-methyl-2-(trifluoromethyl)pyrazolo[1,5-a]pyridine-3-carboxylate (2.6 g, 7.29 mmol) in a mixture of 1,4-dioxane (30 mL) and water (6 mL) was added lithium hydroxide monohydrate (459 mg, 10.9 mmol) and the reaction mixture was stirred at 60° C. for 4 h. To the reaction mixture was added lithium hydroxide monohydrate (459 mg, 10.9 mmol) and the reaction mixture was stirred at 60° C. for 18 h. To the reaction mixture was added lithium hydroxide monohydrate (459 mg, 10.9 mmol) and the reaction mixture was stirred at 60° C. for 2 h. To the reaction mixture was added a fourth portion of lithium hydroxide monohydrate (459 mg, 10.9 mmol) and the reaction mixture was stirred at 60° C. for 2 h. The reaction mixture was concentrated to 6 mL under vacuum. The residue was diluted with water (20 mL) and the mixture was washed with chloroform (1×10 mL). The aqueous layer was acidified to pH 4 by addition of 1 M hydrochloric acid. The precipitate was collected and the solid was washed with water (2×20 mL) to give the title compound (1.80 g, 5.48 mmol, 75%) as a white crystalline solid.)                                                                                                                                                                                                                                                                                                                                                                 \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Bigram_detail  \\\n",
       "0  (Example 55\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopentyl-4-methyl-1,3-thiazole-5-carboxylic acid (0.095 g, 0.448 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed tube was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopentyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)thiazole-5-carboxamide (0.143 g, 80%) as a beige solid., Example 53\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopropyl-4-methyl-1,3-oxazole-5-carboxylic acid (0.072 g, 0.428 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopropyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)oxazole-5-carboxamide (0.143 g, 89%) as a beige solid.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "1  (Step 9: tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate\\nTert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-hydroxyltetrahydro-2H-pyran-3-ylcarbamate (2.33 g, 7.08 mmol) was dissolved in a mixed solution of 24 mL acetonitrile, 4 mL water and 4 mL acetic acid. Thereto was added an aqueous solution (4 mL) of ruthenium chloride hydrate (3.7 mg, 0.0142 mmol), and cooled to 0 °C. Sodium bromate (535 mg, 3.54 mmol) was added, and stirred at low temperature for about 1.5 hours until the raw materials were completely reacted. To the reaction solution was added 120 mL water, stirred at 0 °C overnight, and extracted with dichloromethane. The organic phase was washed with water, dried and concentrated, and then the residue was purified by silica gel column chromatography (petroleum ether/ethyl acetate, 10:1) to give the intermediate 1 tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate (1.71 g) as a white solid in 74% yield., Example 1: (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine -1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-amine\\nStep1: tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2] thiazine-1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate\\nTo N,N-dimethylacetamide (2 mL) were added tert-butyl (2R,3S)-2-(2,5-di-fluorophenyl) -5-oxotetrahydro-2H-pyran-3-ylcarbamate (48 mg, 0.146 mmol), 2,3,4,5,6,7-hexahydropyrrolo [3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide p-toluenesulfonate (48 mg, 0.13 mmol) and triethylamine (10 mg, 0.1 mmol), and stirred at room temperature for 3 hours. After cooling in an ice bath, sodium triacetoxyborohydride (87 mg, 0.39 mmol) was added, slowly warmed to room temperature and stirred overnight. Saturated sodium bicarbonate solution was added, and the resulting mixture was extracted with dichloromethane, washed with saturated brine, dried, concentrated and then purified by silica gel column chromatography to give tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide-6-yl) -2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate (48 mg). Yield: 71%.)   \n",
       "2  (Example 9\\n5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine\\nDMF (1 mL) and DIPEA (0.022 ml, 0.13 mmol) were added sequentially to a flask charged with N-(azetidin-3-yl)-5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-4-methoxypyridin-2-amine (22 mg, 0.050 mmol). 1-Fluoro-3-iodopropane (9 mg, 0.05 mmol) in DMF (0.1 mL) was then added. After 2 hours, the reaction was diluted with saturated aqueous sodium chloride, and the mixture was extracted in EtOAC (3×). The combined organic layers were washed with water and dried over sodium sulfate, filtered and concentrated under reduced pressure. The resulting crude film was purified by flash silica chromatography, elution gradient 2 to 10% MeOH in DCM, to give 5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine (9.0 mg, 36%) as a dry film., Example 6\\nN-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine\\n1-Fluoro-3-iodopropane (40.9 μL, 0.39 mmol) was added to a solution of N-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)azetidin-3-amine (169 mg, 0.39 mmol) and DIPEA (203 μL, 1.16 mmol) in NMP (1.7 mL) at room temperature. After 18 hours, the reaction was concentrated under reduced pressure. The resulting residue was purified by reverse phase flash chromatography (C18), eluting with decreasingly polar mixtures of water (containing 0.2% ammonium hydroxide) and MeCN as eluents. Product fractions were combined and lyopholized to afford N-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine (75 mg, 39%) as clear residue.)                                                                                                                                                                \n",
       "3  (3,5-Dichloro-4-(5-fluoro-2-methyl-4-oxo-1,7-naphthyridin-1 (4H)-yl)benzonitrile (Intermediate 12)\\nStep 1: 3,5-Difluoro-N-methoxy-N-methylisonicotinamide\\n3,5-difluoroisonicotinic acid (18.78 g, 118 mmol), N,O-dimethylhydroxylamine hydrochloride (12.09 g, 124 mmol), HATU (47.1 g, 124 mmol) and DIPEA (61.9 ml, 354 mmol) were suspened in DCM (236 ml). The mixture was allowed to stir at room temperature for 24 hours. The mixture was concentrated under reduced pressure. The residue was extracted with EtOAc washing with saturated aqueous NH4Cl solution (2×), water, saturated NaHCO3 and brine. The organic layer was dried over sodium sulfate and concentrated. The crude was purified by chromatography on silica gel using 0-75% EtOAc in heptanes as eluent to give the title compound (18.9 g, 79% yield)., (Example 12)\\nThe first step\\nTo a solution of Compound 58 (850mg, 1.91mmol) and 3-bromoprop-1-yne (455mg, 3.82mmol) in DMF (5mL) was added 60% sodium hydride (153mg, 3.82mmol) under ice-cooling, and the mixture was stirred for 1 hour at 0 °C. After the reaction mixture was diluted with ethyl acetate, water was added thereto, and the mixture was extracted with ethyl acetate. The organic layer was washed with water, dried over anhydrous sodium sulfate, and concentrated under reduced pressure. The resulting residue was dissolved in ethanol (20 mL) and water (3 mL), ammonium chloride (1.02 g, 19.1 mmol) and iron powder (1.07 g, 19.1 mmol) was added thereto, and the mixture was stirred for 4 hours at 90 °C. After the insoluble materials were removed by filtration, and the filtrate was concentrated under reduced pressure. The resulting residue was extracted with chloroform after addition of chloroform and water. The organic layer was dried over anhydrous sodium sulfate, and concentrated under reduced pressure. The resulting residue was purified by silica gel chromatography (hexane-ethyl acetate) to yield Compound 59 (251 mg, yield 29%).)                                                                                                                                                                                                                                                                                    \n",
       "4  (Example 73\\nPreparation of 2-chloro-3-fluoro-5-nitrobenzoic acid (C206)\\nTo a suspension of 2-chloro-3-fluorobenzoic acid (10 g, 57.5 mmol) in concentrated sulfuric acid (62 mL, 1149.5 mmol) was added concentrated nitric acid (4 mL, 86.2 mmol) dropwise at −10° C. and the reaction mixture was stirred between −10° C. and 0° C. for 3 hours. The reaction mixture was slowly poured into a beaker of crushed ice (˜1 L), and the resulting precipitated solid was filtered and washed with water (100 mL). The crude product was recrystallized from hot water to afford the title compound as an off-white solid (7.5 g, 60%): mp 163-165° C.;, Synthesis of Compound A-1\\nA solution of N-phenyl-[1,1'-biphenyl]-2-amine (0.64 g, 2.61 mmol) in 1,2-dichlorobenzene (5 mL) was bubbled with nitrogen gas for 30 mins, boron tribromide (1M in dichloromethane, 4 mL, 4.00 mmol) was added and the reaction mixture was heated to 180° C. for 3 hours. After cooling the reaction mixture to room temperature (22° C.), diphenylzine (1.27 g, 5.78 mmol) was added to the reaction mixture inside the glovebox. The reaction mixture was heated to 120° C. for 6 hours. The reaction mixture was cooled to room temperature (22° C.), water was added and the reaction mixture was extracted with diethyl ether (×3). The combined organic layer was washed with water and brine solution in this sequence, dried over MgSO4 and concentrated in vacuo. The residue was purified by flash column chromatography with dichloromethane-hexane (1:9 to 2:8) to give Compound A-1 (0.68 g, 80% yield) as white solid.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Trigram_detail  \\\n",
       "0  (Example 55\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopentyl-4-methyl-1,3-thiazole-5-carboxylic acid (0.095 g, 0.448 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed tube was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopentyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)thiazole-5-carboxamide (0.143 g, 80%) as a beige solid., Example 53\\nTo a suspension of 6-(2-amino-5-chlorothiazol-4-yl)-3,4-dihydroquinolin-2(1H)-one hydrochloride (0.100 g, 0.408 mmol) and 2-cyclopropyl-4-methyl-1,3-oxazole-5-carboxylic acid (0.072 g, 0.428 mmol), and pyridine (0.15 mL, 1.831 mmol) in acetonitrile (4 mL) in a sealed was added propylphosphonic anhydride solution (50 wt % in ethyl acetate, 0.85 mL, 1.431 mmol). The sealed tube was heated to 50° C. for 5 days and the precipitation formed. After cooling, the precipitate was collected by filtration and washed with cold 1:1 acetonitrile/water to give 2-cyclopropyl-4-methyl-N-(4-(2-oxo-1,2,3,4-tetrahydroquinolin-6-yl)thiazol-2-yl)oxazole-5-carboxamide (0.143 g, 89%) as a beige solid.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "1  (Step 9: tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate\\nTert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-hydroxyltetrahydro-2H-pyran-3-ylcarbamate (2.33 g, 7.08 mmol) was dissolved in a mixed solution of 24 mL acetonitrile, 4 mL water and 4 mL acetic acid. Thereto was added an aqueous solution (4 mL) of ruthenium chloride hydrate (3.7 mg, 0.0142 mmol), and cooled to 0 °C. Sodium bromate (535 mg, 3.54 mmol) was added, and stirred at low temperature for about 1.5 hours until the raw materials were completely reacted. To the reaction solution was added 120 mL water, stirred at 0 °C overnight, and extracted with dichloromethane. The organic phase was washed with water, dried and concentrated, and then the residue was purified by silica gel column chromatography (petroleum ether/ethyl acetate, 10:1) to give the intermediate 1 tert-butyl (2R,3S)-2-(2,5-difluorophenyl)-5-oxotetrahydro-2H-pyran-3-ylcarbamate (1.71 g) as a white solid in 74% yield., Example 1: (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine -1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-amine\\nStep1: tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2] thiazine-1,1-dioxide-6-yl)-2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate\\nTo N,N-dimethylacetamide (2 mL) were added tert-butyl (2R,3S)-2-(2,5-di-fluorophenyl) -5-oxotetrahydro-2H-pyran-3-ylcarbamate (48 mg, 0.146 mmol), 2,3,4,5,6,7-hexahydropyrrolo [3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide p-toluenesulfonate (48 mg, 0.13 mmol) and triethylamine (10 mg, 0.1 mmol), and stirred at room temperature for 3 hours. After cooling in an ice bath, sodium triacetoxyborohydride (87 mg, 0.39 mmol) was added, slowly warmed to room temperature and stirred overnight. Saturated sodium bicarbonate solution was added, and the resulting mixture was extracted with dichloromethane, washed with saturated brine, dried, concentrated and then purified by silica gel column chromatography to give tert-butyl (2R,3S,5R)-5-(2,3,4,5,6,7-hexahydropyrrolo[3',4':3,4]pyrazolo[1,5-b][1,2]thiazine-1,1-dioxide-6-yl) -2-(2,5-difluorophenyl)tetrahydro-2H-pyran-3-ylcarbamate (48 mg). Yield: 71%.)   \n",
       "2  (Example 9\\n5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine\\nDMF (1 mL) and DIPEA (0.022 ml, 0.13 mmol) were added sequentially to a flask charged with N-(azetidin-3-yl)-5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-4-methoxypyridin-2-amine (22 mg, 0.050 mmol). 1-Fluoro-3-iodopropane (9 mg, 0.05 mmol) in DMF (0.1 mL) was then added. After 2 hours, the reaction was diluted with saturated aqueous sodium chloride, and the mixture was extracted in EtOAC (3×). The combined organic layers were washed with water and dried over sodium sulfate, filtered and concentrated under reduced pressure. The resulting crude film was purified by flash silica chromatography, elution gradient 2 to 10% MeOH in DCM, to give 5-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-N-(1-(3-fluoropropyl)azetidin-3-yl)-4-methoxypyridin-2-amine (9.0 mg, 36%) as a dry film., Example 6\\nN-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine\\n1-Fluoro-3-iodopropane (40.9 μL, 0.39 mmol) was added to a solution of N-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)azetidin-3-amine (169 mg, 0.39 mmol) and DIPEA (203 μL, 1.16 mmol) in NMP (1.7 mL) at room temperature. After 18 hours, the reaction was concentrated under reduced pressure. The resulting residue was purified by reverse phase flash chromatography (C18), eluting with decreasingly polar mixtures of water (containing 0.2% ammonium hydroxide) and MeCN as eluents. Product fractions were combined and lyopholized to afford N-(4-((6S,8R)-7-((1-fluorocyclopropyl)methyl)-8-methyl-6,7,8,9-tetrahydro-3H-pyrazolo[4,3-f]isoquinolin-6-yl)-3-methoxyphenyl)-1-(3-fluoropropyl)azetidin-3-amine (75 mg, 39%) as clear residue.)                                                                                                                                                                \n",
       "3  (3,5-Dichloro-4-(5-fluoro-2-methyl-4-oxo-1,7-naphthyridin-1 (4H)-yl)benzonitrile (Intermediate 12)\\nStep 1: 3,5-Difluoro-N-methoxy-N-methylisonicotinamide\\n3,5-difluoroisonicotinic acid (18.78 g, 118 mmol), N,O-dimethylhydroxylamine hydrochloride (12.09 g, 124 mmol), HATU (47.1 g, 124 mmol) and DIPEA (61.9 ml, 354 mmol) were suspened in DCM (236 ml). The mixture was allowed to stir at room temperature for 24 hours. The mixture was concentrated under reduced pressure. The residue was extracted with EtOAc washing with saturated aqueous NH4Cl solution (2×), water, saturated NaHCO3 and brine. The organic layer was dried over sodium sulfate and concentrated. The crude was purified by chromatography on silica gel using 0-75% EtOAc in heptanes as eluent to give the title compound (18.9 g, 79% yield)., Synthesis of Compound 13\\nTo a mixture of compound 12 (0.45 g, 4.2 mol) and Et3N (8 mL, 0.06 mol) in THF (15 mL), a solution of 2-(tert-Butoxycarbonyloxyimino)-2-phenylacetonitrile (2.1 g, 8.3 mol) in THF (30 mL) was added dropwise at 0° C. Following complete addition, the solution was allowed to warm to room temperature and left to stir for 4 hours. The reaction mixture was concentrated to oil under reduced pressure and CH2Cl2 (50 mL) was added. The mixture was washed with sodium hydroxide (5%, 30 mL) and brine (30 mL). The organic layer was dried over sodium sulfate and concentrated under reduced pressure. The crude product was purified by column chromatography (silica gel, MeOH:CH2Cl2=1:10, v/v) to give compound 13 as a yellow oil (803.1 mg, 60.7%). LC-MS m/z (ES+), 304.22 (M+H)+.)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "4  (Example 73\\nPreparation of 2-chloro-3-fluoro-5-nitrobenzoic acid (C206)\\nTo a suspension of 2-chloro-3-fluorobenzoic acid (10 g, 57.5 mmol) in concentrated sulfuric acid (62 mL, 1149.5 mmol) was added concentrated nitric acid (4 mL, 86.2 mmol) dropwise at −10° C. and the reaction mixture was stirred between −10° C. and 0° C. for 3 hours. The reaction mixture was slowly poured into a beaker of crushed ice (˜1 L), and the resulting precipitated solid was filtered and washed with water (100 mL). The crude product was recrystallized from hot water to afford the title compound as an off-white solid (7.5 g, 60%): mp 163-165° C.;, Intermediate 1: 5-Bromo-7-chloroindolin-2-one\\nTo a cooled (0° C.) solution of 7-chloroindolin-2-one (1.0 g, 6.0 mmol) in TFA (11 mL) was added N-bromosuccinimide (1.0 g, 6.0 mmol) in portions. The reaction mixture was stirred at 0° C. for 6 h. The solvent was removed in vacuo and the residue was diluted with DCM (25 mL) and concentrated, followed by a similar sequence with EtOAc. The crude product was triturated with ethanol to provide the title compound as a white solid (861 mg, 58% yield).)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
       "\n",
       "     data_set task_type  \n",
       "0  Chemu text  NER       \n",
       "1  Chemu text  NER       \n",
       "2  Chemu text  NER       \n",
       "3  Chemu text  NER       \n",
       "4  Chemu text  NER       "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_chemu.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "To a solution of 5-Bromo-4-methoxy-pyridin-2-ylamine (53 g, 0.261 mol, 1.0 eq.) in EtOH:H2O=4:1 (500 mL) is added chloro-acetaldehyde (24.589 g, 0.313 mol, 1.2 eq.), then NaHCO3 (26.3 g, 0.313 mol, 1.2 eq.) is added. The resultant mixture is heated to 90° C. for 4 h. After cooling to r.t., the organic solvent is evaporated. The residue is extracted with DCM (200 mL×3). The organic layer are combined, dried over Na2SO4, filtered and concentrated. The crude product is purified by silica gel chromatography (DCM:MeOH=50:1) to afford compound 6-Bromo-7-methoxy-imidazo[1,2-a]pyridine (39 g, 66%) as a brown solid.\n",
      "\n",
      "Step 1. N-(5-bromo-2-fluoropyridin-3-yl)ethanesulfonamide\n",
      "To a solution of 5-bromo-2-fluoropyridin-3-amine (3 g, 20 mmol) in pyridine (10 mL) and DCM (20 mL) at room temperature was added ethanesulfonyl chloride (2.2 mL, 24 mmol). After stirring for 30 min, the solvent was evaporated. The resulting residue was diluted with MeOH (4 mL) and partitioned between EtOAc and brine. The organic layer was separated, and the aq. layer was extracted with EtOAc. The combined organic layers were dried over Na2SO4, filtered, and concentrated. Purification via silica gel chromatography afforded the title compound (2 g, 40%). LCMS for C7H9BrFN2O2S (M+H)+: calculated m/z=283.0, 285.0; found 283.0, 284.9.\n"
     ]
    }
   ],
   "source": [
    "print(*df_chemu_text.sort_values(by=[\"Unigram\"], ascending=True).head(n=10)[\"Unigram_detail\"].iloc[5], sep='\\n\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EXAMPLE 6\n",
      "5-(3-methoxyphenyl)-3-methyl-1-{2-oxo-2-[4-(2-oxo-1,4-dihydro-2H-quinazolin-3-yl)-piperidin-1-yl]-ethyl}-1H-pyrimidine-2,4-dione (reaction scheme 1, compound 6)\n",
      "Similarly to example 1.8, starting from 71 mg (0.5 mmol) of 3-methoxyphenylboronic acid and 150 mg (0.3 mmol) of 5-bromo-3-methyl-1-{2-oxo-2-[4-(2-oxo-1,4-dihydro-2H-quinazolin-3-yl)-piperidin-1-yl]-ethyl}-1H-pyrimidine-2,4-dione (prepared as described in example 2.5), 120 mg (77%) of 5-(3-methoxyphenyl)-3-methyl-1-{2-oxo-2-[4-(2-oxo-1,4-dihydro-2H-quinazolin-3-yl)-piperidin-1-yl]-ethyl}-1H-pyrimidine-2,4-dione is obtained in the form of a white solid with a melting point of 251° C.\n",
      "\n",
      "EXAMPLE 4\n",
      "5-(3-fluorophenyl)-3-methyl-1-{2-oxo-2-[4-(2-oxo-1,4-dihydro-2H-quinazolin-3-yl)-piperidin-1-yl]-ethyl}-1H-pyrimidine-2,4-dione (reaction scheme 1, compound 4)\n",
      "Similarly to example 1.8, starting from 66 mg (0.5 mmol) of 3-fluorophenylboronic acid and 150 mg (0.3 mmol) of 5-bromo-3-methyl-1-{2-oxo-2-[4-(2-oxo-1,4-dihydro-2H-quinazolin-3-yl)-piperidin-1-yl]-ethyl}-1H-pyrimidine-2,4-dione (prepared as described in example 2.5), 30 mg (20%) of 5-(3-fluorophenyl)-3-methyl-1-{2-oxo-2-[4-(2-oxo-1,4-dihydro-2H-quinazolin-3-yl)-piperidin-1-yl]-ethyl}-1H-pyrimidine-2,4-dione is obtained in the form of a white solid with a melting point of 240° C.\n"
     ]
    }
   ],
   "source": [
    "print(*df_chemu_text.sort_values(by=[\"Unigram\"], ascending=False).head(n=10)[\"Unigram_detail\"].iloc[5], sep='\\n\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_summary = pd.concat([df_sst2, df_aimed_unique,df_aimed_random, df_bc2_gm, df_bc3_act,df_chemu ])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "      <th>test_label</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>data_set</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">AIMED (R)</th>\n",
       "      <th>count</th>\n",
       "      <td>584.000000</td>\n",
       "      <td>584.000000</td>\n",
       "      <td>584.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>96.947936</td>\n",
       "      <td>82.294414</td>\n",
       "      <td>73.146194</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>11.921198</td>\n",
       "      <td>18.519621</td>\n",
       "      <td>21.285885</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>27.498597</td>\n",
       "      <td>8.247861</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>77.777778</td>\n",
       "      <td>66.666667</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">SST2</th>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>36.514837</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>43.643578</td>\n",
       "      <td>16.666667</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>53.452248</td>\n",
       "      <td>30.151134</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>70.710678</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Unigram      Bigram     Trigram  test_label\n",
       "data_set                                                        \n",
       "AIMED (R)  count  584.000000  584.000000  584.000000  0.0       \n",
       "           mean   96.947936   82.294414   73.146194  NaN        \n",
       "           std    11.921198   18.519621   21.285885  NaN        \n",
       "           min    27.498597   8.247861    0.000000   NaN        \n",
       "           25%    100.000000  77.777778   66.666667  NaN        \n",
       "...                     ...        ...         ...    ..        \n",
       "SST2       min    0.000000    0.000000    0.000000   NaN        \n",
       "           25%    36.514837   0.000000    0.000000   NaN        \n",
       "           50%    43.643578   16.666667   0.000000   NaN        \n",
       "           75%    53.452248   30.151134   0.000000   NaN        \n",
       "           max    100.000000  100.000000  70.710678  NaN        \n",
       "\n",
       "[64 rows x 4 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_summary.groupby([\"data_set\"]).describe().stack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Unigram</th>\n",
       "      <th>Bigram</th>\n",
       "      <th>Trigram</th>\n",
       "      <th>test_label</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>task_type</th>\n",
       "      <th>data_set</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">CLS</th>\n",
       "      <th>BC3ACT</th>\n",
       "      <td>26.760271</td>\n",
       "      <td>6.907863</td>\n",
       "      <td>1.807134</td>\n",
       "      <td>0.151667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SST2</th>\n",
       "      <td>46.055815</td>\n",
       "      <td>17.378379</td>\n",
       "      <td>1.388049</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">NER</th>\n",
       "      <th>BC2GM anno</th>\n",
       "      <td>70.766285</td>\n",
       "      <td>19.552406</td>\n",
       "      <td>5.411720</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BC2GM text</th>\n",
       "      <td>33.191640</td>\n",
       "      <td>13.122176</td>\n",
       "      <td>4.200074</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chemu anno</th>\n",
       "      <td>84.286901</td>\n",
       "      <td>30.670974</td>\n",
       "      <td>6.832106</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chemu text</th>\n",
       "      <td>68.446277</td>\n",
       "      <td>42.391903</td>\n",
       "      <td>31.631991</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">REL</th>\n",
       "      <th>AIMED (R)</th>\n",
       "      <td>96.947936</td>\n",
       "      <td>82.294414</td>\n",
       "      <td>73.146194</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AIMED (U)</th>\n",
       "      <td>67.137026</td>\n",
       "      <td>36.067096</td>\n",
       "      <td>20.770509</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Unigram     Bigram    Trigram  test_label\n",
       "task_type data_set                                               \n",
       "CLS       BC3ACT      26.760271  6.907863   1.807134   0.151667  \n",
       "          SST2        46.055815  17.378379  1.388049  NaN        \n",
       "NER       BC2GM anno  70.766285  19.552406  5.411720  NaN        \n",
       "          BC2GM text  33.191640  13.122176  4.200074  NaN        \n",
       "          Chemu anno  84.286901  30.670974  6.832106  NaN        \n",
       "          Chemu text  68.446277  42.391903  31.631991 NaN        \n",
       "REL       AIMED (R)   96.947936  82.294414  73.146194 NaN        \n",
       "          AIMED (U)   67.137026  36.067096  20.770509 NaN        "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_summary.groupby([\"task_type\",\"data_set\"]).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{llrrrr}\n",
      "\\toprule\n",
      "      &     &  Unigram &  Bigram &  Trigram &  test\\_label \\\\\n",
      "data\\_set & task\\_type &          &         &          &             \\\\\n",
      "\\midrule\n",
      "AIMED (R)  & REL & 96.95 & 82.29 & 73.15 & nan \\\\\n",
      "AIMED (U)  & REL & 67.14 & 36.07 & 20.77 & nan \\\\\n",
      "BC2GM anno & NER & 70.77 & 19.55 & 5.41 & nan \\\\\n",
      "BC2GM text & NER & 33.19 & 13.12 & 4.20 & nan \\\\\n",
      "BC3ACT  & CLS & 26.76 & 6.91 & 1.81 & 0.15 \\\\\n",
      "Chemu anno & NER & 84.29 & 30.67 & 6.83 & nan \\\\\n",
      "Chemu text & NER & 68.45 & 42.39 & 31.63 & nan \\\\\n",
      "SST2  & CLS & 46.06 & 17.38 & 1.39 & nan \\\\\n",
      "\\bottomrule\n",
      "\\end{tabular}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(df_summary.groupby([ \"data_set\",\"task_type\"]).mean().\n",
    "      to_latex(float_format=lambda x: \"{:.2f}\".format(x)))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
